\documentclass[phd,tocprelim]{cornell}
%% Added to avoid the \ifpdf name clash error
\let\ifpdf\relax
%
% tocprelim option must be included to put the roman numeral pages in the
% table of contents
%
% The cornellheadings option will make headings completely consistent with
% guidelines.
%
% This sample document was originally provided by Blake Jacquot, and
% fixed up by Andrew Myers.
%
%Some possible packages to include
\usepackage{amsmath}
\usepackage{adjustbox}
\usepackage{graphicx,pstricks}
\usepackage{graphics}
\usepackage{moreverb}
\usepackage{subfigure}
\usepackage{epsfig}
\usepackage{subfigure}
% \usepackage{caption}
\usepackage{hangcaption}
\usepackage{txfonts}
\usepackage{palatino}
\usepackage[LGR,T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage[greek,english]{babel}
\usepackage[hidelinks]{hyperref}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{multirow}
\usepackage{multicol}
\usepackage{tabu}
\usepackage{textcomp} % defines \textmu, which is now what inputenx seems to use for Î¼ - probably due inpmath.. also \textdegree... but not \textrho
\usepackage{textgreek}


% JaBbA document dependency
\usepackage{enumitem,amssymb}
\usepackage{courier}
\newlist{todolist}{itemize}{2}
\setlist[todolist]{label=$\square$}
\usepackage{pifont}
\usepackage{algorithm}
\usepackage{algpseudocode}
\usepackage{algorithmicx}

\usepackage{mathtools, nccmath}
\usepackage{siunitx}
\DeclarePairedDelimiter\ceil{\lceil}{\rceil}
\DeclarePairedDelimiter\floor{\lfloor}{\rfloor}
\DeclarePairedDelimiter{\nint}\lfloor\rceil
\DeclarePairedDelimiter{\abs}\lvert\rvert
\newcommand{\cmark}{\ding{51}}%
\newcommand{\xmark}{\ding{55}}%
\newcommand{\done}{\rlap{$\square$}{\raisebox{2pt}{\large\hspace{1pt}\cmark}}%
\hspace{-2.5pt}}
\newcommand{\wontfix}{\rlap{$\square$}{\large\hspace{1pt}\xmark}}
% -------
\newcommand{\norm}[1]{\left\lVert#1\right\rVert}
\newcommand{\R}{\mathbb R}
\newcommand{\Q}{\mathbb Q}
\newcommand{\ttt}[1]{\texttt{#1}}
\setlength{\parskip}{0em}
\setlength{\parindent}{1em}


\newtheorem{theorem}{Theorem}[section]
\newtheorem{corollary}{Corollary}[theorem]
\newtheorem{lemma}[theorem]{Lemma}

%if you're having problems with overfull boxes, you may need to increase
%the tolerance to 9999
\tolerance=9999
\widowpenalty=10000

% making footnotes
\usepackage[symbol]{footmisc}
\renewcommand{\thefootnote}{\fnsymbol{footnote}}

%\bibliographystyle{plain}
\bibliographystyle{unsrt}
%\bibliographystyle{IEEEbib}
%\captionsetup{justification=raggedright,singlelinecheck=false}
\renewcommand{\caption}[1]{\singlespacing\hangcaption{#1}\normalspacing}
\renewcommand{\topfraction}{0.85}
\renewcommand{\textfraction}{0.1}
\renewcommand{\floatpagefraction}{0.75}

\renewcommand{\contentsname}{Table of Contents}
\addto\captionsenglish{
  \renewcommand{\contentsname}%
    {Table of Contents}%
}

% SO solution to resizing table within page
\newsavebox{\mybox}
\newenvironment{mytabularwrap}{\begin{lrbox}{\mybox}}
  {\end{lrbox}%
  \setbox0\hbox{\usebox\mybox}%
  \ifdim\wd0<\textwidth
      \usebox\mybox%
  \else
      \resizebox{\textwidth}{!}{\usebox\mybox}%
  \fi
}

% Final edits:
% done Write a good acknowledgment
% done Remove Sally text and figures and just refer to them
% include the clinical report system in the Conclusions
% rinse once to fix all citations

% done 1. Confirm conferral date with Registrar; update title page if necessary.
% done 2. Please change title of "Contents" to "Table of Contents"
% 3. Text in figure is too small; all lettering must be readable even when reduced 25%, including axes and legends. Please enlarge the text/figure to make the text readable; if the figure or caption exceed one page, please start the next page with "Figure X (continued)". Alternatively, you can separate and enlarge portions of the figures, rotate them 90 degrees, or adjust the background color for better contrast. (Figures 1.4, 4.1, 5.22, 5.28, 6.3)
% done 4. When chapters are based on published materials, it must be noted appropriately. Please see guidelines for more information.
% 5. While not necessary, it would alleviate confusion to include more of the last paragraph on page 79 with its last line appearing on page 81, as well as "6.11" appearing on page 162.
% done 5. Captions for tables must appear above the table. (Table B.1)

% Revision
% (done) Date on title page must be month of conferral. Confirm with program coordinator and/or Registrar. Month must be written in full, IE "December".
% (done, need to redo after other things) Please adjust widowed lines at bottom of page 6, etc, and orphaned line at top of page 10, etc.
% (done) When noting citation for a published chapter, include the asterisk after chapter title and citation text at bottom of first chapter page. See formatting guidelines for more details.

% (done) Text in figure is too small; all lettering must be readable even when reduced 25%, including axes and legends. Please enlarge the text/figure to make the text readable. Alternatively, you can separate and enlarge portions of the figures, rotate them 90 degrees, or adjust the background color for better contrast. Please see guidelines if the figure takes up or exceeds a single page. (Figures 2.3 (done), 2.11 (done), 3.20 (done), 4.1(done), 4.2(removed), 4.12 (done), 5.3 (done), 5.4 (removed), 5.22()
% (done) When a figure/table is rotated, any text appearing on that page (EG. the caption) must be rotated in the same orientation. (Tables 2.1, 3.2, 4.2, Figure 4.11, 4.13

% (done) When a figure/table takes up or exceeds one page, the caption must be placed alone on the preceding page. Any subsequent pages of the figure must begin with begin with "Figure/Table X (Continued)". (Table 3.1

% (done) Fix margins for Figure 3.17, some of the figure is off the page.
% (done) Caption overlaps with page number (caption to Figures 3.21 on page 93 and 4.14 on page 124). Please revise.
% (done) Caption for Figure 4.3 extends over the bottom margin, as does Figure on page 167. Please revise.
% (done) Text on page 194 extends over right margin, please revise.

% (done) While not required, it is advised to either include the figure number and title in the caption paragraph or single-space the caption paragraph so as to avoid confusion. Please be sure all captions have consistent justification, either center or left.

% Further revisions
% 1. Please adjust widowed line at bottom of page 60. 
% 2. What figures are you trying to reference here on page 71? "wide range of tumor purities (Figures) as it utilizes" etc 
% 3. Please remove blank area/page break at bottom of pages 73, 125. 
% 4. Chapters can't end in a figure or table. Please revise. (Chapter 3) 
% 5. What figures are you referring to on page 128? "Figure 2E" does not conform to numbering convention, please revise. 

%% \title {Structural variants in cancer genomes: representation, inference, and applications}
\title{A genome graph paradigm for cancer genome structure}
\author {Xiaotong Yao}
\conferraldate {December}{2021}
\degreefield {Ph.D. Computational Biology}
\copyrightholder{Xiaotong Yao}
\copyrightyear{2021}

\begin{document}

\maketitle
\makecopyright

\begin{abstract}
    % 350 words
    % characterizes SVs in the form of copy number aberrations (CNA) and junctions.
    % SVs inherently alter the coordinate system of the reference genome, thus the interpretation of any junction is dependent on other overlapping junctions. Plus, with short-read WGS, we can only rely on relatively local sequence readouts, making it challenging to reconstruct the long range linear structures of the DNA.
    % which treats genomic sequences as directed graphs, where vertices are single-stranded DNA and edges are 3'-5' phosphodiester bonds joining adjacent vertices. 
    % I show that JaBbA not only achieves more accurate CN estimation, more complete genome graphs than other methods,
    Cancer genomes harbor structural variants (SV), producing various driver alterations and reflecting the signatures of mutagenesis. With the fast accumulating whole genome sequencing (WGS) data, many complex patterns of SVs have been discovered. Yet, one of the major obstacles to better SV analysis is the lack of flexible and general framework to represent the complexity of SVs. Here, I present genome graph as a general data structure to represent rearranged genomes, and give a mixed-integer quadratic programming algorithm, Junction Balance Analysis (JaBbA), to infer coherent and accurate integer copy numbers (CN) simultaneously for both DNA segments and junctions from the WGS of a tumor sample. With JaBbA, I discover distinct classes of complex rearrangements from 2778 pan-cancer genome graphs, including \textit{tyfonas}, a massive complex amplification pattern different from previously known double minutes or breakage-fusion-bridge cycles. Tyfonas are more effective at producing highly-expressed fusion genes and are specifically enriched in acral not cutaneous melanoma. Next, I use the genome graph framework to reconstruct the evolving SV events in diverging lineages of human fibroblast cells surviving natural telomere crisis and show that telomere crisis can instigate a wide range of chromosomal aberrations, not limited to BFBs and chromothripsis. Lastly, by integrating genome graph in the analysis of 85 whole genomes of lung adenocarcinoma lacking alterations in the RTK/RAS/RAF pathway, I characterized simple and complex SV events abolishing the expression of various tumor suppressors or enhance that of oncogenes. In sum, I show that genome graph and JaBbA is a general, robust analysis paradigm that elucidates the complexity of SVs in cancer genomes, provides a basis for SV evolutionary trajectory inference, and empowers whole genome analysis of cancers.
    % Genome instability is a hallmark of cancer partially reflected by abundant and diverse large-scale structural variants (SV). Whole genome sequencing enables hollistically discovery of the repertoire of SVs. Extracting distinct patterns from observed SVs in cancer genomes can help trace back to responsible mutagenesis processes, or precisely define consequences of known mutagenesis processes in controlled experiments. To date, many complex patterns of SVs have been discovered in various cancer types. Yet, there still lacks a general conceptual framework and practical toolset to account for the complexity of SVs, stagnating the systematic discovery of the etiology of SVs. 
    
    % Here, I present genome graph as a general data structure to represent rearranged genomes, and give a mixed-integer quadratic programming algorithm, Junction Balance Analysis (JaBbA), to infer coherent and accurate integer copy numbers (CN) simultaneously for both DNA segments and junctions from the WGS of a tumor sample. With JaBbA inferred genome graphs in a pan-cancer cohort of 2778 samples, I discover distinct classes of complex rearrangements including \textit{tyfonas}, a massive, complex amplification pattern different from previously known double minutes (DM) or breakage-fusion-bridge (BFB) cycles. Tyfonas is specifically enriched in acral not cutaneous melanoma and generates more highly-expressed fusion genes that potentially confers its significant response to immunotherapies in spite of low mutation burdens. Taking together all 13 simple to complex SV patterns, I show that the patient cohort form clusters that shows different enrichment in tumor types, DNA repair defects, and overall survival. Next, to define the genomic consequences of a crucial mutagenesis process, telomere crisis, I reconstruct the evolving SV patterns in diverging lineages of cells escaping natural telomere crisis by inducing telomerase expression. The results show that telomere crisis can instigate a wide range of simple and complex chromosomal aberrations, not limited to the previously proposed BFBs and chromothripsis. Lastly, by integrating the genome graph paradigm into the analysis of 85 whole genomes of lung adenocarcinoma lacking known alterations in the RTK/RAS/RAF pathway by whole exome sequencing (WES), I characterized simple and complex SV events abolishing the expression of various tumor suppressors or enhance that of oncogenes. In sum, I show that genome graph with the JaBbA is a general, robust analysis paradigm that elucidates the complexity of SVs in cancer genomes, provides a basis for SV evolutionary trajectory inference, and empowers whole genome analysis of cancers.
\end{abstract}

\begin{biosketch}
    Xiaotong Yao has had strong interest in biology from an early age and later developed an appreciation for using computational methods to answer biological questions. His interest for computational biology started with systems biology and synthetic biology by participating in the 2011 International Genetically Engineered Machine Competition. Subsequently, he curated lipid metabolic pathways in the \textit{Streptomyces avermitilis} metabolic network. To get formally trained in computational biology, he joined the Master's program in Bioinformatics and Systems Biology at the Biology Department of New York University, where he built predictive models for protein sumoylation from public protein function databases. Following a passion for precision medicine, in 2015, he became a PhD student in the Tri-institutional Program in Computational Biology and Medicine at Weill Cornell Medicine, focusing on cancer genomics, and in 2016 he joined Dr. Marcin Imielinski's lab to pursue his dissertation research in using genome graphs to model complex Structural variants in cancer whole genomes. He aspires to continue the research in cancer genomics in the future towards making a more direct impact on the patient outcomes.
\end{biosketch}

\begin{dedication}
    To Rosalind Yao and Shan Huang.
\end{dedication}

\begin{acknowledgements}
    This thesis is founded by my mentor Dr. Marcin Imielinski. In 2016, I was pondering whether a particular recurrent SV event is just a variant calling artifact, when Marcin generously broke me free from the existing framework of counting junctions like mutations, by revealing a possibility where all SVs in a genome can be unified in one graph to reveal their complexity. The elegance of this formulation immediately captured my interest, but it was not until after numerous afternoons sitting down with Marcin, browsing through thousands of genome graphs, improving our models over and over, that I finally felt I started to grasp the significance of this work. During the challenging times of preparing for manuscript submission, responses to reviewers, Marcin's astuteness has been the constant guidance I held onto. Beyond imparting his unique combination of expertise in computational sciences and medicine, working with Marcin instilled in me a swift yet rigorous approach to data analysis and scientific investigation in general, that can best be summarized by a quote from the famous investor Naval Ravikant -- \textit{Impatience with actions, patience with results.} With new data at hand there should be immediate list of things burning to be found out; always ask yourself the next question; fail quick and iterate; battle-test your code; motivate every sentence in a presentation or manuscript with a question -- are just a few of the teachings that were cast into my tenets. Besides extraordinary professional and academic mentoring, Marcin actively cared for my physical and mental health. For that, I will be forever grateful for the chance to have done this work with Marcin.
    
    Next, I want to thank my wonderful teammates and fellow graduate students, Kevin Hadi, Julie Behr, Aditya Deshpande, who have been offering support and encouragement when things got tricky, critic and reflection when there is room for improvement. It is a privilege to have friends who are also role models. I also thank the rest of the Imielinski lab members for feedback on all the projects and unconditional support throughout this period.

    In true spirit of science as a collective effort, I want to thank all the collaborators whose names I cannot list in an exhausting manner. Working with these brightest minds has been an honor and I will carry the gifts of this experience onwards. Special thanks to Drs. Sally Dewhurst and Titia de Lange for giving me the opportunity to making discoveries on the SV outcomes of telomere crisis; Dr. Jian Carrot-Zhang, for characterizing the whole genome of lung adenocarcinomas; Dr. Nicolas Robine for many projects together and continuous supportive conversations.

    Last but not least, I want to thank the rest of my thesis committee Drs. Christina Leslie (chairperson), Simon Powell, and Iman Hajirasouliha for helping me through the academic training.
\end{acknowledgements}

\contentspage
\tablelistpage
\figurelistpage
%\abbrlist
%SV -- structural variant
%SNV -- single nucleotide variant
%INDEL -- insertion and deletion
%WGS -- whole genome sequencing
%CN -- copy number
%CNA -- copy number aberration
%JaBbA -- Junction Balance Analysis
%\symlist


\normalspacing \setcounter{page}{1} \pagenumbering{arabic}
\pagestyle{cornell} \addtolength{\parskip}{0.5\baselineskip}

% "Defining fundamental biological and/or technical analytic questions"
% gGnome -- a framework that can represent complex SVs
% JaBbA -- a reliable method to infer JBGGs from WGS
% Distinct patterns -- a distinct complex amplicon emerged from analyzing high JCN clusters
% TC -- detailed characterization of chr changes after crisis
% LUAD -- alternative SV drivers for LUAD lacking known SNV/fusion drivers

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Chapter 1: introduction
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter{Introduction}

\section{From patterns to mechanisms to etiology} \label{sec:philo}
Genomic instability is a hallmark of cancer \cite{Hanahan2011-ni} and genomic DNA sequencing of tumor tissues have proved the ubiquity of genomic variants \cite{pcawg_marker2020-yi}. Even though the idea that chromosomal abnormalities as an intrinsic property of cancer has been proposed \cite{Boveri2008-rl} for more than a century, the advent of massively parallel sequencing in the past decade has been revolutionary in expanding the known repertoire of somatic variants with unprecedented detail including single nucleotide variants (SNV), small insertions and deletions (INDEL), and structural variants (SV) \cite{Meyerson2010-xc}.

% Stratton 2009; % The points of studying variant patterns at all
% rephrase from recent Nik-Zainal review?
The somatic variants in a cancer genome are result of mutagenesis, DNA repair, and evolution \cite{Stratton2009-ty}. The goal of studying their patterns is to categorize variant into classes, then facotorize the counts of variant of classes among a population into \textit{signatures} (more discussed below), and eventually infer the indogeneous and exogeneous mechanisms from which specific signatures arise. 

Such studies commonly combine various features of observed genomic variants, including substitution type, spatial distribution (strand-coordinated clustered mutations by APOBEC \cite{Shi2017-cd}), multi-scale sequence context (tri-nucleotide contexts \cite{alexandrov2013}; mesoscale palindromes preference by APOBEC3A \cite{Buisson2019-ue}), replication timing and transcription level (higher frequency in late replication region and less transcribed regions \cite{Lawrence2013-zx}), 3-dimensional chromatin structures (mutation rate change associated with topological association domain boundaries \cite{Akdemir2020-hn}; lamina associated domains accumulate more UV-related mutations \cite{Garcia-Nieto2017-ox}), tissue specificity (lineage-specific INDELs in lung adenocarcinoma \cite{imielinski2017}, negative correlation between frequency and cell of origin-specific expression \cite{polak2015}), to elucidate the relationships between the genesis and repair of DNA damages in soma (\cite{Zou2021-je,Volkova2020-zg,Vohringer2021-bv}). They are creating wide-reaching impacts on at least three fronts.

% THE SIGNIFICANCE TO STUDY VARIANT PATTERNS!!!!!!!!!!!!!!!
% 1, biology of genome maintanence
First of all, these studies offer insights into the basic biology of genome maintenance and DNA repair. For example, one of the most critical and extensively studied DNA repair pathways, homologous recombination (HR), was associated with excessive accumulation of SNV mutation signature 3 (will discuss the definition of mutation signatures in later sections), tandem duplications, and microhomology-directed deletions \cite{Nik-Zainal2016-bz}. By adapting an \textit{Escherichia coli} replication fork barrier in human cells, Willis et al. proposed that \textit{BRCA1} not \textit{BRCA2} is responsible for suppressing the replication restart-bypass mechanism that result in microhomology-mediated tandem duplications \cite{Willis2017-bw}.

% 2, better background frequencies
% help driver discovery, can come later
% With better understanding of mutation propensity of different parts of the genome in different tissues, environments, and genetic backgrounds, we can also better approximate the null distribution of mutation density and more precisely pinpoint the variants undergoing positive selection (drivers) (cite FishHook, MutSigCV, dndscv).
Second, more textured understanding of the heterogeneous background somatic mutation rate of the genome in various tissue origins and environments improves the detection of positive selection during the evolution of tumors. As one of, if not the most important goals of cancer genomics, determining the exact set of alterations (or \textit{drivers}) out of a vast sea of somatic variants that are responsible for malignant phenotypes of cells has been heavily guided by the analysis of significantly recurrently mutated sites, regions, or genes. Very early on, the data clearly showed that without a well-calibrated background mutation distribution, an enormous amount of false positives will render cancer genome sequencing efforts futile \cite{Getz2007-wf}. Consequently, many statistical models are used to correct for variations in background mutation frequencies (MutSigCV \cite{lawrence2013}, FishHook \cite{imielinski2017}, dNdScv \cite{Martincorena2017-yt}. As the patterns of variants are linked to their mutagenesis mechanisms and are not homogeneous among different tumors, they can contribute to a more accurate estimate of background mutation rates along the genome and further reduce Type I error of driver discovery.

% 3, clinical benefit
Last but not least, they are bringing novel and profound clinical impact to the prevention, diagnosis, and treatment of cancer. Using the positive association between smoking behavior of the donors and the observed SNV signature 4 burden, Alexandrov et al. \cite{Alexandrov2016-zs} estimated each pack of cigarette smoked per day for a year roughly contribute to 150 somatic mutations accumulated in lung tissue. This confirms and complements decades of public health and epidemiology conclusions about the causal link between smoking and elevated cancer risks. In diagnostics, mutation signatures combined with sophisticated machine learning techniques are shown to predict the tissue of origin with mutation patterns alone \cite{Jiao2020-le}. Homologous recombination defects of breast and ovarian tumors can be predicted by combining several feature of SNVs, INDELs, and SVs, and guide the prescription of PARP-inhibitor therapy \cite{Davies:2017642}.

% Two philosophical routes
\label{sec}
Philosophically, the studies of variant patterns take two main routes. One is to crystallize latent signatures of mutagenesis pathways out of observed counts of distinct classes of variants in a collection of unrelated genomes. One of the earliest landmark examples is the extraction of \textit{mutation signatures} from a feature space of tri-nucleotide context (the immediate 5' and 3' nucleotide of the mutated nucleotide) of SNVs \cite{Alexandrov2013-zv}. The general idea is to count the number of variants in each patient that fall into one of the 96 categories (4 possible choices on 5' and 3' each times 6 substitution types) and compile a matrix from a large patient cohort. The factorization of this matrix will produce linearly independent vectors in the feature space and the top of the list represent the more common 'themes' in the cohort. Many of the original mutation signatures have been associated with specific mutagenesis pathways since discovered. For instance, mutation signature 1 has attributed to the spontaneous or enzymatic deamination of 5-methylcytosine to thymine resulting in C>T transitions, and it is correlated with the age of the patient, analogous to a molecular clock for cell divisions \cite{alexandrov2013}. So far this methodology and the resulting signature repertoire have been instrumental in revealing the active mutational processes in many different cancers \cite{Choi2021-pu} and have been revised multiple times to reflect the expanding datasets \cite{Alexandrov2020-by}. As exciting as it is, mutation signatures are simply mathematical abstractions of observed data which always requires further molecular evidence to trace back to the responsible mutational processes, which remains challenging. Plus, the repertoire of the signatures is not yet finalized with respect to source data and algorithmic improvements. Last but not least, such approach ideally requires a non-overlapping and complete taxonomy of variant classes, which is easier for small, local variants (SNVs, small INDELs), and still lacking for complex SVs. I will further expand on this point in the next section.

The other study design is to create controlled mutagenesis experiments, for instance in human cell lines, organoids, mouse models, where a particular mutagen is introduced or a particular DNA repair pathway component is disabled for a sustained period, then identify the variants acquired during, to attribute specific types of variants to the corresponding processes. There are successful stories of evaluating the SNV/INDEL mutation signatures induced by exposure to external mutagens in cell models \cite{Zou2021-je}. Specifically for SVs, Umbreit et al. characterized the consequences of mechanically broken dicentric chromosome bridge \cite{Umbreit2020-kr}. Shoshani et al. captured the continuous instability in extra-chromosomal circular DNA (eccDNA) in response to therapeutic pressure \cite{Shoshani2020-zj}. Ly et al. constructed an elegant model of inactivated centromere that led to missegregation, which caused various complex SVs including chromothripsis and eccDNA \cite{Ly2019-bt}. Apparently, such studies are more powerful in establishing the causal links between mutational processes and signatures, but significantly harder to design, especially in the face of intricate interactions between DNA damage and repair mechanisms, bias introduced by positive or negative selection, and the essentiality of many DNA repair genes. After all, it also relies on an adequate framework to describe the variants.

Despite these challenges, the combination of the these two approaches has rapidly and permanently reshaped this field and will be both indispensable. In this thesis, I will demonstrate the data driven approach with my study discovering complex SV patterns in genome graphs with Chapter \ref*{chap:complex_events}, and the controlled experimental approach in Chapter \ref*{chap:tel_crisis} as I portray the evolution of SVs after telomere crisis.

\section{Complex structural variants discovered in cancers}
Among somatic variants, structural variants (SV) are of particular interest as they are expected to have a bigger phenotypic impact on cells, more complex etiology, and have been linked to various mechanisms of therapy resistance and metastasis \cite{Wedge2018-nf,Priestley:20196a6}. In recent years, many complex patterns of SVs have been discovered.
%% do I need this? compare WES and WGS?
% Enriching specific regions of the genome for sequencing, including gene panels and whole exome sequencing (WES), are done routinely in many clinical settings (cite exome paper from GDAN bibliography) to pair patients with targeted therapy informed by the driver alterations. Yet, they are blind to the vast un-captured genome territory (e.g. non-coding, intergenic) harboring more variants. In contrast, while currently less widely used in clinic due to higher cost and analytically more challenging (cite Olivier review recently), whole genome sequencing (WGS) provides a hollistic profile of all genomic variants, thanks to the unbiased sampling of DNA fragments over the whole genome. Wielding the high-throughput sequencing technologies

% THE PAST: what SV patterns have been discovered
In \cite{stephens2011}, Stephens et al. observed the phenomenon and coined the term \textit{chromothripsis}, theorized to originate from shattering a chromosome arm into pieces and erroneously rejoined a subset of them in random order. Such processes can generate up to hundreds of junctions at random locations and with random orientations, while leaving an oscillating CN profile due to loss of interspersed segments. Since its initial definition, there have been many claims of its prevalence in different tumor types, even non-cancerous somatic tissues. One recent pan-cancer whole genome analyses estimated chromothripsis-like events exist in as many as 29\% of pan-cancer genomes \cite{Cortes-Ciriano2020-fx}.

In prostate carcinoma genomes, Baca et al. \cite{baca2013} discovered chained, long-range, reciprocal junctions that are formed through simultaneous multi-loci (>2) translocations. Even though it is also a complex SV event with clustered junctions, its pattern is qualitatively distinct from chromothripsis, with usually smaller footprint, long-range chains, sometimes involving many chromosomes (>=3), and less loss of genetic materials limited to the \textit{deletion bridges} between adjacent junctions along the chain. Furthermore, there are also experimental evidence that androgen receptors are co-recruited with topoisomerase II beta (\textit{TOP2B}) to sites of TMPRSS-ERG fusion and lead to double strand breaks (DSB) \cite{Haffner2010-cy}, indicating that chromoplexy may have a transcription-related origin than a chromothripsis.

As a prime example of using WGS to refresh and upgrade the understanding of an old chromosomal aberration, double minutes (DM) or extrachromosomal circular DNAs (eccDNA), recently have been investigated in a series of studies \cite{Helmsauer2020-do,Wu2019-ap,Kim2020-cu} presenting its high-resolution, complex structure along with histone markers and 3-D chromatin conformations. In subsets of glioblastoma multiforme (GBM) and other central nervous system cancers, certain cancer genes like epidermal growth factor receptor (\textit{EGFR}) can detach from its original chromosome location and exist on eccDNAs, which replicates with other nucleus DNA yet inherited by daughter cells in a non-symmetric stochastic process \cite{Gaubatz1990-da}. 

% (Describe PCAWG, in more details)
In a more systematic fashion, the Pan-cancer Analysis of Whole Genomes (PCAWG) consortium catalogued WGS variants across >2,500 cases spanning 38 tumor types\cite{pcawg_marker2020-yi} to identify novel classes of complex SVs and cluster these into signatures, mirroring previous work in the categorization of single nucleotide variants (SNVs) into distinct mutational processes\cite{Li2020-ds,Alexandrov2013-zv,Nik-Zainal2012-dc,Menghi2018-er}. It also builds a taxonomy of basic repertoire of CNA outcomes given a sequence of junctions, and nominated a new event type \textit{templated insertion cycle/chain} (TIC) attributed to template switching of stalled replication forks.

Despite these extraordinary advancements, the discoveries of SV patterns have been outpaced by that of smaller variants. There are several reasons behind the current status. First, the field has not yet converged to a single algorithmic framework to identify and systematically compare the full spectrum of these patterns in a tumor whole genome sequence. For instance, it is unclear whether some of the clustered rearrangement patterns commonly observed in cancer represent variations on known events (e.g. "amplified" chromothripsis) or as yet uncharacterized event classes \cite{pcawg_marker2020-yi, Li2020-ds}. Plus, even though some of these studies are prudent in interpreting SV patterns combining both junctions and CNA, they are not unified in a easily-replicable, data-driven way. Many example events shown in publications clearly contain copy number change points without a consistent junction, which are inconsistent with the fact that every \textit{mappable} copy number change point should have a junction "explaining" it (discussed in more details in \cite{Behr2021-gf}). Furthermore, various heuristics are often used to attribute CNA to nearby junctions, making it hard to evaluate performance or replicate results across labs, studies or datasets.

% Second, tumor clones may have accumulated SVs repeatedly throughout their life history, it is very often that multiple events overlap at the same locus, blurring the boundaries of individual events when mapped to a single linear reference.
% \clearpage
Second, though copy number is primarily a concept applied the dosage of genomic \textit{intervals}, a junction may also be present in one or more copies per cell, and thus be associated with a \textit{junction copy number} (JCN). Elevated JCN might occur through the focal (e.g. extrachromosomal) amplification \cite{Verhaak2019,Wu2019-ap} or whole chromosome (or genome) duplication of an already rearranged allele (Figure \ref{fig:jcn}). We argue the topology of JCNs are important pieces of evidence that all efforts hitherto tend to omit or use implicitly, due to the lack of reliable inference methods.

Third, unlike small variants which can be each counted as an irreducible edit, rearranged genome is not simply the sum of all junctions. When junctions overlap over the reference genome, their interpretations are interdependent. Two junctions can be on the same molecule (in \textit{cis}) or different (in \textit{trans}), and their multiplicity allow for different copies of a junction to have different localizations. To actually guess the final derivative sequence resulted from complex SVs, we need a general paradigm that can represent this complexity, and serve as an intermediate scaffold to integrate long-molecule profiling technologies.

As a result, the taxonomy of complex SV is still under-defined, stagnating the discovery of causal mutational processes, and hindering the determination of genomic consequences of known mutagenesis pathways. In the following section, I review the current common practices in identifying and analyzing SVs from WGS data and explain in more detail why it is not ideal for the complex SV patterns in cancer.

\section{Characterization of SVs with WGS data}
% OLD and DECOUPLED
Practically, SVs in cancer genomes have been routinely characterized in two separate ways: aberrant junctions and copy number aberration (CNA). On one hand, junctions are two loci associated with orientations (called \textit{breakends}) in the reference genome that are adjacent in the studied genome. When a junction is not connecting two breakends that are already adjacent in the reference genome, it is called an aberrant junction. For ease of discussion, we simply call it junction. There have been many methods and tools to identify junctions from WGS \cite{wala2018,Cameron2017-pz,Rausch2012-ly,Wang2011-lg,Layer2014-xq}, and the evidence they rely on fall into three main categories: discordant read pairs, split reads, and assembled contigs. Depending on the type or combination of types of evidence supporting a candidate junction, the precision of identification can be as good as single nucleotide level. For somatic junction identifications in cancer genomes, the general principle is to differentiate the junctions with supporting evidence in the tumor and not in the matching normal (usually peripheral blood) tissues.

On the other hand, the amount of reads mapped to a certain reference genomic region, termed \textit{coverage}, is a readout for the absolute number of copies of that DNA segment within an "average" cell of the studied sample. CNAs are parts of the reference genome with copy numbers (CN) deviated from 2 (for autosomes). They are detected as locations of shifting center of coverage data \cite{Ha2014-ai,Xi2011-oa,Carter2012-xo,Favero2015-hj,Van_Loo2010-ed,Shen2016-fi}, or called \textit{copy number change points}. Of course the variations in coverage over the reference genome are affected by a lot of confounding factors other than CN, for instance, GC-content of the reference genome (due to the GC-bias of short-read sequencers \cite{Benjamini_Speed_2012}), mappability (uniqueness of reference sequences affects false positive and false negative read alignments) \cite{Lee_Schatz_2012}. In cancer tissue sequencing, there are also mixture of stromal cells, which are assumed to be near diploid and do not possess the tumor-specific CNAs. Elucidated in \cite{Carter2012-xo}, the purity (proportion of cancer cells in a sample) and ploidy (DNA segment width-weighted average CN) form an affine transformation (preserved collinearity) from coverage to absolute CN, hence can compare across samples.

Despite being inferred orthogonally, junctions and CNAs are two facets of the same latent genome structure. Based on the structure of DNA, apart from whole chromosome, contig, or genome duplications or loss, any change in the copy number of a genomic region require at least one breakage and formation of the covalent 3'-5' phosphodiester bond, which, if mappable at both breakends, should be detected as a junction. Conversely, at any breakend that forms a junction in a rearranged genome, if no perfect reciprocal junction is connected to the other side of the breakage, then there is bound to be loss of some genetic material.

Conceptually, integrating the two procedures should improve both detections. On one hand, junctions provide much more precise localization for copy number change points, and consistent copy number changes with respect to the junction topology should increase the confidence in the junction calls. Indeed, there have been multiple methods that have made such attempts. A CNA inference method, CONSERTING\cite{Chen2015-sw}, iteratively matches candidate copy number change points with nearby junctions of the consistent orientations and achieved better segmentation than other methods using only coverage evidence. In terms of junction identification, Pedersen and Quinlan developed Duphold \cite{Pedersen2019-pp}, that annotates junction candidates with whether the coverage change at the breakpoint is convincingly consistent to greatly improve the specificity of SV calls within germline genomes.

% BUT THESE ARE ALL BEATING AROUND THE BUSH
However, these improvements are still confined within the variant detection realm and to truly couple these orthogonal readouts and approach complex rearranged sequences, a fundamentally new paradigm of thinking about SVs are desperately needed.

\section{Complexity of somatic SVs in cancer genomes}
For legacy reasons, most analyses on genomic structural variants have been represented with data structures designed for smaller, local variants (like SNVs, short INDELs, and simple duplication and deletion junctions \cite{Danecek2011-yi}) where a variant is represented as an sequence edit at one or a pair specific locations in the genome. It considers each junction or CNA as an independent edit of a reference locus \cite{Li2008-su,Li2009-fv}. This works fine most of the time for germline genomes, whose SVs are dominated by simple classes, namely tandem duplications, simple deletions, and inversions \cite{Collins2020-iv}. Tandem duplications and simple deletions are composed of only one junction while inversions and balanced translocations (exceedingly rare in germline but common in cancer) a pair of reciprocal junctions. Although more complex patterns in germline genomes have also been characterized \cite{Carvalho2011-lb,Collins2020-iv}, they are, as expected with larger phenotypic effect and lower rate of emergence, rarer than the simpler ones. This approach has apparent advantages including its simplicity, availability of robust tool set like VCF formats and VCFtools \cite{Danecek2011-yi}, and each junction or CNA can be analyzed as an independent unit, yet it quickly becomes inadequate when the SVs become complex in cancer genomes.

% includes tandem duplication junction burdens and also complex SVs, the SV signatures 6 extracted are a undifferentiated mixture of all clustered junctions 

Such school of thought to treat each junction as an independent unit of SV is also influencing the extraction of SV signatures. Many attempts have been published to represent each junction or CNA with a unified feature space like the 96-dimension tri-nucleotide context of SNVs or INDELs. While this approach is valid to the extent of local sequence context of the breakends, and most simple SVs, they are less suitable to describe the properties of complex events which can contain many junctions per event. For example, while HRDetect \cite{Davies:2017642} uses counts of several types of junctions to help predict HRD status, however in the final model only the simple junction counts contributing to the prediction (tandem duplications and deletions of various sizes and breakend microhomology status). The rearrangement signature six which was associated with clustered inversion-like junctions and deletions was not explored or utilized. In another attempt at integrated mutational signature inference, Funnel et al. \cite{Funnell2019-ti} adpated multi-model correlated topic models to avoid the pitfall of relative sparsity of SVs to SNVs and showed the combined signature achieve better stratification of ovarian cancer patients. One of their important SV features is characterized by fold-back inversions (FBI), a hallmark of BFB cycle-like processes. Nevertheless, the framework cannot provide further details on the exact events that are producing these junctions.

The limitation of this approach in dealing with complex SV events is visible even in the simplest form of complexity. Consider the toy model in Figure 2 of \cite{Maciejowski2016-nf}, where a unrearranged genomic contig \textit{ABCDE} underwent two nested tandem duplications. When mapped to the reference genome, the second junction and its assciated CNA appeared to be consistent with a deletion when only seen from the perspective of the reference genome. If we have perfect long-read sequencing for any rearranged allele as such, we may directly observe its structure and could have avoided this illusion. Yet, in practical short-read WGS, when the B/C/D segments are (easily) longer than the hundred basepairs reads, all we can rely on are the local junctions and the change of copy numbers. Thus, a more general data structure is called for to encapsulate all the observed junctions and CNAs without making any assumption about the actual underlying event.

\section{Representation of SVs with genome graphs}
Graphs have been widely used to model genomic sequences for decades, with \textit{de novo} assembly as one of the fields most heavily reliant on diverse yet related data structures like de Bruijn graphs \cite{Zerbino2008-cc}, string graphs \cite{Simpson2010-xe}, unitig graphs \cite{Li2012-ac}. Motivated by rapidly increasing resequencing data and pool of known variants, Garrison et al. developed \textit{variation graphs} (vg) \cite{Garrison2018-sj}, compressing all sequence variants found in a population, allowing for variant-aware read mapping for individual samples, and achieving more accurate variant calling and genotyping \cite{Hickey2020-cu,Siren2020-tf}. Vg creates 'bubbles' (alternative paths) to represent polymorphic sequences and efficiently maps sequencing reads by generalized compressed suffix arrays to avoid biases of aligning to a single linear reference genome. More recently, Li et al. \cite{Li2020-ai} developed \textit{minigraph}, the first method to construct reference pangenome graph from multiple genome sequences of the same species while retaining the coordinates of a linear reference. They showed that this approach can compactly annotate highly complex and polymorphic regions like the Human Leukocyte Antigen locus.

%high-fidelity long-molecule sequencing combined with other long- and short-read platforms, catalyzed the telomere-to-telomere \textit{de novo} assembly of a complete human genome \cite{Nurk2021-ro,Logsdon2020-vm,Miga2020-xo}. This major milestone was accompanied by 

One commonality of them is that vertices stores the sequence of a contiguous DNA segment (contig) and a path (or cycle) constitute longer sequence and eventually approach a genome. In resequencing studies, we map reads to locations in an existing reference genome, and identify variants based on the discrepancy between reads and the mapped reference sequence. Nevertheless, we can apply a similar idea of modeling, now replacing actual DNA sequences to the genomic intervals they map to in the reference.

% \clearpage
There have been several studies employing the concept of \textit{interval graph} or \textit{breakpoint graph} to model the somatic SVs in cancer genomes. Originally, breakpoint graphs are built to solve ancestral genome reconstruction problems based on synteny across species \cite{Alekseyev2009-fw,Avdeyev2016-un}. One of the earliest example of applying a reference genome-based genome graph was from Greenman et al. \cite{Greenman:2012eg}, where vertices represent allelic specific double-stranded segments and edges the adjacencies among them, while each vertex (segment) must have a left and a right boundary and an edge must specify which boundaries of the incident vertices are incident. In another early effort, Oesper et al. \cite{Oesper2012-vw} mark the left and right boundaries of each double stranded genomic intervals with two vertices and categorize the adjacencies between vertices as either constituting the interval (interval edge) or junction (reference and variant edges). Later, Dzamba et al. 2017 \cite{Dzamba2017-wo}, Deshpande et al. \cite{Deshpande2019-gs}, Aganezov et al. 2019 \cite{Aganezov2019-yh}, Lee and Lee 2021 \cite{Lee2021-rl}, also adopted this second implementation.
 
In our own publication \cite{Hadi2020-um} and more expansively Chapter \ref{chap:gg}, we have formulated a directed skew-symmetric graph data structure, called \textit{genome graph}, that is dual to the variations of interval graphs described above. Like the previous work, we will show that genome graphs can represent genomic rearrangements based on a reference genome and serve as a scaffold to inferring the derivative sequences. Though these implementations are functionally equivalent and only subtly different, a skew-symmetric directed graph formulation made the concept, in our opinions, much more intuitive and immediately compatible with numerous algorithms that are general to directed graphs.

To our knowledge, there has not been a published software package that facilitates the analysis of genome graphs. One that can freely construct any genome graphs, parse the results from the aforementioned methods, serves as the foundation to build useful operations (e.g. subgraph, walk, shortest paths, partition, max/min flow), and characterize biological meaningful events. In Chapter \ref{chap:gg}, we set out to build such an interface in R programming language using the state-of-the-art genomic interval operations \cite{Lawrence2013-lh} and graph computation (https://igraph.org/). We aim to provide a robust, practical, and versatile tool to popluarize the usage of genome graphs in the whole-genome analysis of cancers. As an example, Chapter \ref{chap:gdan} will show our collaboration with the Cancer Genome Atlas to characterize the whole genomes of lung adenocarcinomas lacking known alterations in the RTK/RAS/RAF pathway by whole exome sequencing (WES).

Another major obstacle between genome graphs and its wider adoption by the research community is the lack of proper visualization tools. Projection of data onto reference genome coordinates, or genome browsers, has been the quintessential tool of genomics \cite{Robinson2011-ra,kent2002-ke}. There have also been several tools designed to visualize SVs at different stages of the analysis. Samplot is a Python command line tool that generates static images of junction-supporting reads along with coverage data in the vicinity of the junction breakends \cite{Belyeu2021-tm}. Specifically for graph genomes like vg, sequence tube maps \cite{Beyer2019-nb} takes an creative turn and visualize reads aligned to graph genomes in a similar way to transport networks. Later, MoMI-G \cite{Yokoyama2019-xd} attempted to deliver a comprehensive suite for long-read data aligned at complex SV sites with sequence tube maps and circular plots as modular units.

However, for complex structural variants, they are too constrained within one or two regions at a time to convey the full picture. For example, chromoplexy can involve many different chromosomes or very distant loci. One particular tool, Ribbon\cite{Nattestad2016-dn,Nattestad2018-wl}, is designed with SV complexity in mind and is able to put more than two discontiguous windows in one image, yet it is specific for long-read alignments and cannot visualize general genome graphs. To this end, we built a GPU-powered Javascript application \textit{gGnome.js} to interactively browse any genome graph mapped to its reference genome.

\section{Reconstruction of junction-balanced genome graphs}
To represent coherent double stranded genomic DNA, or karyotype, we need to annotate genome graphs (both vertices and edges) with non-negative integer copy numbers, and because of the linear structure of DNA, we find that these copy numbers need to follow Eq. \ref{eq:junction_balance_constraint}. Such graphs are named junction-balanced genome graphs. In Chapter \ref{chap:gg}, we will show that junction balance arises naturally from compressing all the linear alleles.

Despite its clear conceptual advantage, genome graph has not been ubiquitously adopted as the mainstream framework for SV analysis. The main reason is that the numeric quality of the copy numbers in the reconstructed graphs have not shown high enough fidelity for the heterogeneous landscape of complex somatic SVs, as well as varying sample purity. 

There have been several approaches proposed to reconstruct junction-balance genome graphs with different objectives, underlying models, inference methods, and scopes (Table \ref{tab:jbgg_inference}, \cite{Oesper2012-vw,McPherson2017-ry,Li2016-qa,Dzamba2017-wo,Deshpande2019-gs,Li2020-ds,Lee2021-rl,Aganezov2019-yh}). In Chapter \ref{chap:gg}, I will present a mixed-integer quadratic programming method, Junction Balance Analysis (JaBbA) to infer coherent and optimal copy numbers of a genome graph from WGS data. Besides, I will show a comprehensive benchmarking of some of the aforementioned methods and discuss the potential reasons for their strengths and limitations. With recent improvements and proof of application in large pan-cancer WGS cohorts \cite{Hadi2020-um}, it shows a promising trend that genome graph-based SV analyses are becoming robust enough to power general SV analysis.

\section{SV pattern discovery guided by junction copy numbers}
% In WGS, CNAs are detected as change-points \cite{chiang2009} in sequencing read depth along the genome \mbox{(e.g. BIC-seq \cite{Xi2011-oa})} while rearrangements are nominated through the analysis of junction-spanning read pairs (e.g. SvABA \cite{wala2018}, GRIDSS \cite{Cameron2017-pz}). CNAs and junctions are, however, intrinsically coupled, since every copy of every non-telomeric segment must have a left (towards smaller coordinates) and a right (towards larger coordinates) neighbor, whether that neighbor is already adjacent on the reference (via a reference or REF junction) or introduced through rearrangement (via a variant or ALT junction) \cite{Medvedev:2010bm, Greenman:2012eg}. Furthermore, 

%% how to use genome graphs in SV pattern recognition, classification
The inference of JCN in WGS involves the fitting of a genome graph to read depth and junction data through the application of a junction balance constraint, which requires the CN of every genomic interval (i.e. vertex) to be in balance with the JCN of its neighboring junctions (i.e. edges) (Figure \ref{fig:ggraph_schematic}, \ref{fig:jabba_io})\cite{Medvedev:2010bm, Greenman:2012eg, Oesper2012-vw, Li2016-qa, Dzamba2017-wo, McPherson2017-ry}. We hypothesized that the topology and CN of both vertices and edges on junction-balanced genome graphs might reveal novel classes of complex SV events and mutational processes. 

In particular, high junction copy numbers (JCN) must arise through subsequent alterations that replicates existing junctions, and hence are an indication of complex amplification events. As touched upon before, there are well known mechanisms like double minutes and breakage-fusion-bridge cycles active in various cancers and WGS has painted detailed structures of them. Yet, the same mechanism can generate a wide range of complexity \cite{Deshpande2019-gs} and it is unknown if there are other amplification patterns. One such example is neochromosomes found in several liposarcoma cell lines that do not fit in either DM or BFB cycles \cite{garsed2014}. Hence, a systematic, data-driven classification of complex amplicons in a comprehensive dataset is crucial to answer these questions.

In the first half of Chapter \ref{chap:complex_events}, I will show the discovery of 3 stable classes arising from hierarchical clustering of amplified subgraphs from out pan-cancer genome graphs and show that tyfonas is a distinct amplification pattern.

\section{Stratifying pan-cancer patient cohort by burdens of SV events}
Using SV events as a biomarker has always been a top desired application of WGS. Early efforts includes using the burdens of tandem duplication and deletions of various sizes as a features to predict homologous recombination deficiency (HRD, \cite{Davies:2017642}). More recently, the existence of extrachromosomal circular DNA (eccDNA) has been linked to excessive replication stress in the cells. One necessary indication that SV events may possess such discriminating power is to show that non-random grouping patterns among patients can arise based solely on their burdens. For instance, like mentioned above, in \cite{baca2013}, almost half of studied prostate adenocarcinoma contain chromoplexy patterns, then if we can see a subgroup of patients differentiated by chromoplexy presence from the others, and it happens to enrich in prostate adenocarcinoma, it should be a positive indication that our patterns are confirming previous results and add to our confidence about new results.

\clearpage
In the second half of Chapter \ref{chap:complex_events}, I show our effort in using the 13 simple to complex SV event types to reveal 13 groups of patients in the pan-cancer cohort and their associations with tumor type, genotype, and overall survival. 

% Cancer genomes are shaped by mutational processes that create both simple and complex structural variants (SVs). 
\section{Evolution of SVs after telomere crisis}
The studies of the relationship between variant outcomes and mutational processes usually take one of two directions. Most of the studies including our efforts described above are searching for common patterns from (hopefully unbiased) observations about genomes, and trace back to their possible mechanisms. However, more traditionally, the genome maintenance and DNA repair research community has always taken the more hypothesis-driven approach to create controlled experiments of mutagenesis. However, there is significant gap between the most advanced variant profiling technology and pattern recognition methods and the delicately designed experimental systems where we can clearly attribute any DNA changes to a particular mutagenesis process. Furthermore, this gap can be complicated by the fact that the same "type" or "class" of variants can be generated by multiple mechanisms, and a mechanism leads to diverse outcomes. Therefore, the integration between advanced analysis and finely controlled experimental systems are the key moving forward.

At this point, we have demonstrated the power of our genome graph paradigm in delineating SV events from large observational pan-cancer WGS cohorts. Beginning the next chapter \ref{chap:tel_crisis}, I will use it to investigate the chromosome structural consequences of an crucial and (nearly) universal mutagenesis process, telomere crisis.


In Chapter \ref{chap:tel_crisis} I will dive deep into this detailed characterization of the structural outcomes of telomere crisis, and discuss about the implications of adopting our genome graph framework to resolve evolving genome structures among multiple related samples in the Concluding Remarks \ref{chap:conclude}.

\section{Genome graphs facilitates whole genome analysis of tumors}
The most important goal of sequencing cancer genomes is to pinpoint the somatic alterations responsible for the malignant phenotype, or \textit{drivers}. To achieve that, the main signal to rely on is the recurrent alteration of a particular region of the genome from independent patients tumors. To put simply, each patient's tumor can be regarded an independent clonal expansion, if a particular genomic region is altered in a cohort more than expected from a null distribution where the region is not positively selected, it can be nominated as selected in these cancers. In theory, whole genome sequencing (WGS) is a more ideal tool to profiling the cancer genomes for such purpose since it samples reads from the whole genome, while whole exome sequencing (WES) only mainly captures the known exonic sequences. Yet, limited by higher cost and difficulty to analyze, WGS was not the primary choice of earlier cancer genome profiling efforts. For example, the Cancer Genome Atlas (TCGA), as one of the earliest large-scale cancer sequencing projects, relied mainly on WES, while powerful at discovering recurrent coding mutations, also misses by design non-coding mutations, and the vast majority of SVs.

Having discussed how our genome graph paradigm can represent complex genome structures and be used to study SV patterns, I will take it a step further and show how it enables a more comprehensive whole genome analysis of tumors by better modeling of the rearrangements. More specifically, as part of the next phase of TCGA effort to push for whole genome characterizations, we focused on a subset lung adenocarcinomas (LUAD) in which WES and RNA-seq (for fusion transcripts) did not assign a clear driver. In the last Chapter \ref{chap:gdan}, 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Chapter 2: formalize genome graphs and JaBbA
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter{Junction-balanced genome graphs represent structurally altered genomes *} \label{chap:gg}
% I also establish the correspondance between a walk on a genome graph to the underlying karyotype and show genome graphs can unify a diverse array of putative functional events including fusion genes and enhancer hijiacking. 
In this chapter, I give formal definitions of a genome graph, a junction balanced genome graph, and Junction Balance Analysis to infer integer copy numbers from short-read whole genome sequencing data. Taking advantage of our novel interactive genome browser \textit{gGnome.js}, I show that our complete tool set can serve as an intuitive portal for researchers to incorporate genome graphs in WGS studies. 

Part of this chapter is described in our published article \cite{Hadi2020-um}*, led by Kevin Hadi, Julie Behr, Marcin Imielinski, and myself.

\textit{
    \textbf{Individual contributions:} Marcin Imielinski and I conceptualized, formulated, and implemented JaBbA for Junction Balance Analysis. Kevin Hadi and I collected and processed the WGS data from simulations and HCC1954, HCC1143 cell lines. I carried out the benchmarking experiments of JaBbA. Marcin Imielinski and I proposed and developed gGnome, and Kevin Hadi, Julie Behr, Zi-ning Choo, Alon Shaiber, Joe DeRose have contributed. Charalampos Xanthoupolakis, Marcin Imielinski, and I designed the gGnome.js genome browser for genome graphs.    
}

% \chapter[Evolution and phase of clustered deletions]{Evolution and phase of clustered deletions*}\label{chrig}
\footnotetext[1]{Hadi, K., Yao, X., Behr, J.M., et al. Distinct classes of complex structural variation uncovered across thousands of cancer genome graphs. Cell, 183(1):197â210, 2020.}

\newpage

%===========================%
\section{Genome graph as a general data structure to represent rearranged genome}
\subsection{Reference genome}
Let the reference genome $\mathcal{C}$ comprise $c$ pairs of strings, labeled $C^i$ and $C^{-i}, i \in 1,\dots,c$. Each string pair $C^{\pm{i}} = \{C^i, C^{-i}\}, i e\in 1,\dots, c$ is called a \textit{chromosome}, and each string in that pair is called a \textit{strand}. We use $C^i$ and $C^{-i}$ to refer to "positive" and "negative" strands of chromosome $i$, each having length $L_i \in \mathbb{N}$. We use brackets to denote substrings on these strands. For example, $C^{i}[q,r]$ refers to the substring of $C^{i}$ beginning at position $q$ and ending at position $r$ (inclusive) where $q \le r \in 1,\dots,L_i$. We also use $C^{i}_q$ as a shorthand for $C^{i}[q,q]$. In the remainder of this dissertation, we call $C^{i}[q,r], i \in 1, \dots, c, q \le r \in 1, \dots, L_i$ a signed \textit{genomic interval}. The width of interval $C^{-i}[q,r]$ is $r-q+1$.

Every signed genomic interval has a "reverse complement" $C^{-i}[q,r]$ and together they make a double stranded interval $C^{\pm{i}}[q,r]$, termed \textit{segment}. Two segments $C^{i_1}[q_1, r_1], C^{i_2}[q_2, r_2]$ are said to \textit{overlap} if $i_1 = i_2$ and either $q_2 \le r_1$ or $q_1 \le r_2$, and a set of intervals are said to be \textit{disjoint} if none of its members overlap. As a trivial example, one chromosome $C^{\pm{i}}$ is a segment of width $L_i$.

%% how to define a segmentation
% A segmentation $S$ of a segment $C^{\pm{i}}[q, r]$ is set of disjoint segments $\{C^{\pm{i}}[q,k_1], C^{\pm{i}}[k_1 + 1,k_2], ..., C^{\pm{i}}[k_n + 1,r]\}, k_1 \ge q, k_n <= r$
As is the convention in defining reference genomes, on the positive strand $C^i, i>0$, the coordinate increases from 5' end (left) to 3' end (right). In other words, $C^{i}_q$ is the 5' end of $C^{i}[q,r]$, and $C^{i}_r$ its 3' end. Conversely, $C^{-i}_r$ is the 5' end of $C^{-i}[q,r]$, and $C^{-i}_q$ the 3' end. We term the 3' end the outgoing end $b_o$ and the 5'e end receiving end $b_r$. Based on skew symmetry, $b_o (v)= \bar{b_r(\bar{v})}$ and $b_r(v) = \bar{b_o(\bar{v})}$. A pair of reverse complement interval ends of the opposite sides, for example, $b = \{b_r(v), b_o(\bar{v})\}$ or $b = \{b_o(v), b_r(\bar{v})\}$ is termed a \textit{breakend}. When marking the location of a breakend $b = \{b_r(v), b_o(\bar{v})\}, v = C^{i}[q,r], i>0$, we simply use a width 1 signed interval $b = C^{i}_{q}$, here the positive sign of $i$ indicates the breakend belongs to the interval continuing onto the right of the coordinate $q$, while for breakend $b = \{b_o(v), b_r(\bar{v})\}$, $b = C^{-i}_r$.

Taken the above definitions together, each segment $s = C^{\pm{i}}[q,r]$ has two breakends, left (smaller coordinate) $b_L(s) = \{b_r(v), b_o(\bar{v})\} = C^{i}_q$ and right (larger coordinate) $b_R(s) = \{b_o(v), b_r(\bar{v})\} = C^{-i}_{r}, v = C^{i}[q,r], i>0$.

\subsection{Genome graph}
Based on a reference genome $C$, we define genome graph $G = (V, E)$, where the vertices $V$ is a multi-set of signed intervals, and edges $E$ is a set of directed adjacencies representing the 3'-5' phosphodiester bonds between vertices. Since genomic DNA is a pair of reverse complement strands, we restrict $G$ to be skew-symmetric \cite{Goldberg1996-qm} with the mapping function "reverse complement" denoted with a bar over, such that for any vertex $v \in V$, there exists its \textit{symmetric} vertex $\bar{v} \in V$, and for any edge $e = (v_1, v_2) \in E$, $\bar{e} = (\bar{v_2}, \bar{v_1}) \in E$. Intuitively, each reverse complement pair of vertices represents a double stranded DNA segment. Since any vertex maps to a genomic interval, we mark its location in the reference genome with $v = C^{i}[q,r]$.

% definition in as plain language as possible
% need to define: reference genome, interval, signed interval, junction
\begin{figure}[h!]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/ggraph_schematic.png}
    \caption{\textbf{Schematic of a genome graph.}}{Grey and purple rectangles are positive and negative strand intervals. Links are edges of different types: black, reference; red, alternate or aberrant; blue, loose ends; dark blue, telomeric ends.}
    \label{fig:ggraph_schematic}
\end{figure}

Because the phosphodiester bonds are always connecting the outgoing (3') end of a source vertex $v_1 = C^{i_1}[q_1,r_1]$ and the receiving (5') end of a sink vertex $v_2 = C^{i_2}[q_2,r_2]$, we can denote the genomic location of an edge $e = (v_1, v_2)$ with a tuple of the connected breakends $e = (b_o(v_1), b_r(v_2))$. For a node $v$ we name the set of incoming edges to its 5' end $E_{-}(v)$ and the set of outgoing edges from its 3' end $E_{+}(v)$.

Following this definition, there are at least two types of edges. The first type, \textit{REF} edges are the ones that are connecting breakends adjacent in the reference genome hence always with location $e_r = (C^{i}_{q}, C^{i}_{q + 1})$, if $i>0$, or $(C^{i}_{q_1 + 1}, C^{i}_{q_1})$, if $i<0, q \in [1,L_i)$. In contrast, the second type \textit{ALT} edges are neo-adjacencies that are not present in the reference genome and resulting from rearrangements (\textbf{Figure \ref{fig:ggraph_schematic}}).

Like a reverse complement pair of vertices $\{v, \bar{v}\}$ compose a double-stranded DNA segment, a reverse complement pair of edges $\{e, \bar{e}\}$ compose a \textit{junction}. Trivially, $e$ and $\bar{e}$ will always be the same type of edge, so we also have two types of junctions, REF and ALT. Each junction $a = \{e, \bar{e}\}, v_1 = C^{i_1}[q_1,r_1], v_2 = C^{i_2}[q_2,r_2]$ can be represented with the two breakends it connects, namely $b_1 = \{b_o(v_1), b_r(\bar{v_1})\} = C^{i_1}_{q_1 [1-I(i_1)] + r_1 I(i_1)}$, and similarly $b_2 = C^{i_2}_{q_2 [1-I(i_2)] + r_2 I(i_2)}$, where indicator function $I:\mathbb{R}\rightarrow \{0, 1\}, I(x)=1, if x \ge 0, I(x) = 0, if x<0$.

\begin{figure}[h!]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/junction_orientation.png}
    \caption{\textbf{Four basic classes of junctions.}}{Dark gray indicates the \textit{cis} side and light gray the \textit{trans} side of a breakend.}
    \label{fig:junction_orientations}
\end{figure}

For a breakend location in the middle of a chromosome, only one of its two sides is incident to the junction, we call this side the \textit{cis} side for it is the part of the genome that will be on the derivative molecule after introducing the junction, and the other \textit{trans} side. Based on the relative locations and orientations of its two connected breakends, we can classify a junction into four basic classes (\textbf{Figure \ref{fig:junction_orientations}}): DUP-like (tandem duplication), DEL-like (deletion), INV-like (inversion), TRA-like (translocation), named after the most common and simplest SV events that can give rise to such junctions. In WGS studies, these junctions are the direct products of SV calling methods like SvABA \cite{Wala2018-qa}, GRIDSS \cite{Cameron2017-pz}, Delly \cite{Rausch2012-ly}, Lumpy \cite{Layer2014-xq}, and many more. In the following section we show a basic algorithm to build a genome graph from a set of breakends and junctions.

% TODO!!!
\subsection{Build genome graph from breakends and junctions}
To describe a rearranged and copy number altered reference genome, we partition $\mathcal{C}$ according to a collection of \textit{breakends} $\mathcal{B}$. We also define a set of \textit{junctions} $\mathcal{A}$ representing alternative adjacencies between a set of the breakends in $\mathcal{B}$.




% TODO!!!
Each $b^i = C^ \in \mathcal{B},\ i \in 1,\ldots,c$ is an ordered and unique sequence of integer coordinates $B^i = (B^i_k), 1 \le B^i_k \le L_i$ on chromosome $i$, where $B^i_1$ = 1 and $B^i_{|B^i|} = L^i$.  Each junction $A \in \mathcal{A}$ is a tuple $(i_1,r_1,i_2,r_2),\ r_1 \in B^{|i_1|}, r_2 \in B^{|i_2|}$, $|i_1|, |i_2| \in  1, \ldots, c$ representing a (3'-5' phosphodiester) bond between the position $r_1 + \frac{-sgn(i_1)+1}{2}$ on chromosome / strand $C^{i_1}$ and position $r_2+\frac{sgn(i_2)+1}{2}$ on chromosome / strand $C^{i_2}$.  For every adjacency $A = (i_1,r_1,i_2,r_2) \in \mathcal{A}$ we require $\mathcal{A}$ to contain the reverse complement adjacency $\bar{A} = (-i_2,r_2, -i_1,r_1)$. The adjacencies in $\mathcal{A}$ are "alternative" relative to a set of "reference adjacencies" $\mathcal{R}$ implied by $B^i$, comprising tuples $(i, B^i_k, i, B^i_k)$ and $(-i, B^i_k, -i, B^i_k)$ for each breakend $B^i_k, k\in 1,\dots,|B^i|$ in each chromosome $i \in 1,\ldots,c$.



\begin{algorithm}[h!] 
    \caption{BuildGraph} 
    \label{algo:buildgraph}  
    \begin{algorithmic}[1]
        \Procedure{BuildGraph}{$\mathcal{A}$, $\mathcal{B}$} \Comment{Returns genome graph $G = (V, E, \psi, \phi)$}
        \State $G \gets emptyGraph;\quad$ $\phi, \psi \gets emptyDictionary$
        \State $\mathcal{B} = \mathcal{B} \cup \ getBreakPoints(\mathcal{A})$ \Comment{Extract additional breakends from junctions in $\mathcal{A}$}
        \For{$B^i_k,\ k\in 1,\ldots,|B^i|-1,\ c \in 1,\ldots,|\mathcal{C}|/2$}
        \State $v, \bar{v} \gets newVertices(2, \psi, "I")$ \Comment{$newVertices$ updates $\psi$ to map $v$ and $\bar{v}$ to "I"}
        \State $\phi(v) = (i, B^i_k, B^i_{k+1}-1)$
        \State $\phi(\bar{v}) = (-i, B^i_k, B^i_{k+1}-1)$
        \State $l_1, \bar{l_1}, l_2, \bar{l_2} \gets newVertices(4, \psi, "L")$
        \State $e_1, \bar{e_1}, e_2, \bar{e_2} \gets newEdges(\{ (l_1, v), (\bar{v},\bar{l_1}), (v, l_2), (\bar{l_2},\bar{v}) \}, \psi, "L")$
        \State \algorithmicif\ $k = 1$ \ \algorithmicthen\ \ $\psi(\{l_1, e_1, \bar{l_1}, \bar{e_1} \}) \gets "T"$  \Comment{Label telomeric loose ends with "T"}
        \State \algorithmicif\ $k = |B^i|-1$ \ \algorithmicthen\ $\psi(\{l_2, e_2, \bar{l_2}, \bar{e_2} \}) \gets "T"$
        \State $V(G) \gets V(G) \cup  \{ v, \bar{v}, l_1, \bar{l_1}, l_2, \bar{l_2} \})$
        \State $E(G) \gets E(G) \cup  \{ e_1, \bar{e_1}, e_2, \bar{e_2} \})$
        \EndFor        
        \algstore{buildgraph}        
    \end{algorithmic}
\end{algorithm}

\newpage

\begin{algorithm}[h!]
    \centering
    Algorithm \ref{algo:buildgraph} (Continued)
    \begin{algorithmic}
        \algrestore{buildgraph}
        \State $\mathcal{R} = referenceAdjacencies(\mathcal{B})$  \Comment{Get implied reference adjacencies from breakends}
        % \State $\mathcal{A} = \mathcal{A} \cup \ reverseComplement(\mathcal{A})$  \Comment{Make sure $\mathcal{A}$ contains both forward and RC adjacencies} 
        \For{$A = (i_1,r_1,i_2,r_2) \in \mathcal{A} \cup \mathcal{R}$}
        \If{$sgn(i_1)>0$}
        \State $v_1 \gets getVertex(\{ \hat{v} \ | \ \phi(v) = (i_1, q, r_1+\frac{-sgn(i)+1}{2}), v \in V(G), q \in \mathbb{N}\}$)
        \Else
        \State $v_1 \gets getVertex(\{ \hat{v} \ | \ \phi(\hat{v}) = (i_1, r_1 + \frac{-sgn(i)+1}{2}, q), v \in V(G), q \in \mathbb{N}\})$
        \EndIf
        \If{$sgn(i_2)>0$}
        \State $v_2 \gets getVertex(\{ \hat{v} \ | \ \phi(v) = (i_2, r_2 + \frac{sgn(i_2)+1}{2}, q), v \in V(G), q \in \mathbb{N}\})$
        \Else
        \State $v_2 \gets getVertex(\{ \hat{v} \ | \ \phi(\hat{v}) = (i_2, q, r_2 + \frac{sgn(i_2)+1}{2}), v \in V(G), q \in \mathbb{N} \})$
        \EndIf
        \State \algorithmicif\ $A \in \mathcal{R}$ \algorithmicthen\ $e = newEdges((v_1, v_2), \psi, "R")$ \ \algorithmicelse \ $e = newEdges((v_1, v_2), \psi, "A")$
        \State $E(G) \gets E(G) \cup \{e\}$
        \EndFor
        \State \textbf{return} $(V(G), E(G), \psi, \phi)$
        \EndProcedure
    \end{algorithmic}
\end{algorithm}

\newpage

\subsection{Walks on a genome graph map to rearranged karyotype}
Analogous to an assembly graph, on genome graph $G$, traveling through a walk $h = (v_1, e_1, v_2, e_2, \dots, e_{k-1}, v_k), k \in \mathbb{N^+}$, where each edge $e_i = (v_i, v_{i+1}), i \in 1, 2, \dots, k-1$, is equivalent to generating a substring of DNA sequence, simply by concatenating the reference sequences of the vertices. If some of the edges along the walk belong to ALT edges $E_{A}$, then the walk represents a possible derivative sequence resulted from the SVs that gave rise to these edges. Since $G$ is skew-symmetric, every walk $h$ has a reverse complement walk $\bar{h} = (\bar{v_k}, \bar{e_{k-1}}, \dots, \bar{e_2}, \bar{v_2}, \bar{e_1}, \bar{v_1})$. We also denote the number of times a walk $h$ travel through a node $v$ or an edge $e$ using $\delta : (V \cup E, H) \rightarrow \mathbb{N}$. Based on this definition, we have two important observations.

First, given a tuple of genomic intervals $(v_1, v_2, \dots, v_k)$ whose order implies sequential covalent bonding, we can easily produce a genome graph and represent the tuple as a walk on that graph, by filling in edges between each consecutive pair of intervals based on their corresponding incident breakends, then add the same for the reverse complement walk. For example, the edge joining $(v_1, v_2)$ should be $(b_o(v_1), b_r(v_2))$. This observation implicated that any DNA sequence, as long as it can be constructed from pasting substrings of the reference genome, can be converted to a walk and a genome graph.

Second, every interval has exactly one upstream neighbor (incident to its 5' breakend, except for the first interval), and one downstream neighbor (incident to its 3' breakend, except for the last interval), as genomic DNA is known to be linear. Thus, we can associate a walk with a non-negative integer $\kappa(h) \in \mathbb{N}$ as the multiplicity of such molecule present in a genome. Consequently, the graph arise from these walks must posses be \textit{junction balanced}, a property we are going to define in  Subsection \ref{sub:JBGG}.

\subsection{Junction-balanced genome graph} \label{sub:JBGG}
We define a mapping $\kappa:\{V \cup E\}\rightarrow \mathbb{N}$ of non-negative integer copy number (CN) to vertices and edges of $G$, where $\kappa(v),v \in V$ and  $\kappa(e),e \in E$ represent the CN of vertex $v$ and edge $e$, respectively. For a graph $G = (V, E)$ that arise from the aforementioned procedure from a set of tuples of genomic intervals, the following junction balance constraint is always true:

\begin{equation} \label{eq:junction_balance_constraint}
    \kappa(v)= \sum_{e\in E^-(v)} \kappa(e) = \sum_{e\in E^+(v)} \kappa(e)
\end{equation}

This is because to travel into $v$ once one must travel through one of the edges in $E_{-}(v)$, and vice versa for the outgoing edges. In plain language, the principle of \textit{junction balance} constrains the CN of every vertex to be equal to the sum of its incoming edges and the sum of its outgoing edges. Formally,

\begin{equation} \label{eq:sum_of_walk}
    \begin{split}
        \forall v \in V, \kappa(v) = \sum_{h \in H}\kappa(h)\delta(h, v) \\
        \forall e \in E, \kappa(e) = \sum_{h \in H}\kappa(h)\delta(h, e)
    \end{split}
\end{equation}

in which $\delta: \{H, V \cup E\} \rightarrow \mathbb{N}$ is the number of times $h$ travels through a vertex or an edge $\sum_{j \in 1,\dots,k}I(v = v_j)$, with the only exception that if $h = (v_1, e_1, v_2, e_2, \dots, e_{k-1}, v_k), v_1=v_k, k \in \mathbb{N^+}$ is circular $\delta(h,v_k) = 0$, so that $v_k$ is not double counted.

In addition, since double-stranded DNA require both strands to have the same CN, we require the CN $\kappa$ to obey \textit{skew-symmetry}, which means that every vertex or edge must have the same copy number as its reverse complement.

\begin{equation} \label{eq:skew_symmetry}
    \kappa(v) = \kappa(\bar{v}),\ \forall_{v \in V} \quad \kappa(e) = \kappa(\bar{e}),\ \forall_{e \in E}
\end{equation}

We call the combination $(G,\kappa)$ for which $\kappa$ satisfies Eqs. \ref{eq:junction_balance_constraint}-\ref{eq:skew_symmetry} a \textit{junction-balanced genome graph} (JBGG).

% TODO!!!
% Flow decomposition:
% Why is a complete graph contain all the possible walks from source to sink?
\subsection{Decomposing a JBGG to walks}
In a perfect scenario where we have infinitely long sequencing reads, we can read out any sequence directly and fully (telomere to telomere), for any rearranged karyotype. However, in practice, sequencing reads from existing technologies are fragmented and limited in length, so the walks representing complete and true karyotypes are always latent. Despite current long-read sequencing provides us a better chance at obtaining the rearranged allele, it is still "local" relative to the scale of some of the most long-range complex SV events like tyfonas \cite{Hadi2020-um} (see also Chapter \ref{chap:complex_events}). Furthermore, due to the differences in cost, accuracy, throughput, and analytical robustness, the overwhelming majority of cancer WGS data is still based on massive parallel short-read sequencing, with which SV events are detected as copy number aberrations and junctions. Despite certain challenges, for instance with SVs in repetitive regions, short-read WGS has been found to approach the complete SV repertoire well \cite{Behr2021-gf}.

% TODO!!!: why a complete genome graph contains all possible karyotypes
Though we cannot directly read the derivative sequence after rearrangement, from junctions we can always build a genome graph (see Algorithm \ref{algo:buildgraph} in later sections). If we have the complete knowledge on all junctions, then this graph will encapsulate every possible karyotype that could result from them, hence must contain the set of walks representing the true sequence (proof not shown). The enumeration of all possible walks from all the source nodes to all the sink nodes is analogous to elementary flux mode of a stoichiometric matrix \cite{Maarleveld2013}, and produces a set of paths and cycles $H$ where the latter can be further embedded to form more complex walks.

However, such genome graphs only deal with the topology of junctions, and for a coherent double-stranded DNA model, the vertices and edges' dosages need also be constrained by Eq. \ref{eq:junction_balance_constraint} and Eq. \ref{eq:skew_symmetry}. With that we can define the reverse problem of Eq. \ref{eq:sum_of_walk},

\begin{equation} \label{eq:walk_decomposition}
    \begin{aligned}
        \text{Given } & G = (V, E), \kappa(G) \\
        \text{find } & \kappa(h):H \rightarrow \mathbb{N} \\
        s.t. & \forall v \in V, \kappa(v) = \sum_{h \in H}\delta(h,v)\kappa(h) \\
        & \forall e \in E, \kappa(e) = \sum_{h \in H}\delta(h,e)\kappa(h) \\
        & \kappa(h) = \kappa(\bar{h})
    \end{aligned}
\end{equation}

%Suppose we are given an input JBGG with its junction-balanced CN $G, \kappa(G)$, the walk decomposition problem is now to find a multiplicity of every walk $\kappa(h), h \in H$, such that Eq. \ref{eq:sum_of_walk} is satisfied. 
Since in its essence, this is equivalent to a flow decomposition problem, the junction balance constraints \ref{eq:junction_balance_constraint} guarantee that it is feasible, usually with a huge number of possible solutions. Upon this pool of possible karyotypes, we can further define different loss functions, such as parsimony (number of unique walks with $\kappa(h)>0$), to reach some optimal guesses at what the actual rearranged allele is that constitute the input JBGG. Taking the most naive formulation as an example, to minimize the total number of unique walks to reconstitute an input JBGG, formally the question is defined as,

\begin{equation} \label{eq:walk_decomposition_parsimony}
    \begin{aligned}
        \text{Given a JBGG } & G = (V, E), \kappa(G) \\
        \underset{\kappa(h):H \rightarrow \mathbb{N}}{\text{minimize}} & \sum_{h \in H} [\![ \kappa(h) ]\!] \\
        s.t. & \forall v \in V, \kappa(v) = \sum_{h \in H}\delta(h,v)\kappa(h) \\
        & \forall e \in E, \kappa(e) = \sum_{h \in H}\delta(h,e)\kappa(h) \\
        & \kappa(h) = \kappa(\bar{h})
    \end{aligned}
\end{equation}


We shall see an example of solving a variation of this problem in Chapter \ref{chap:tel_crisis}.

When combined with read coverage data, we can further infer JBGG by filling in an optimal set of CNs for all the vertices and edges. We call this problem the inference or \textit{reconstruction of JBGG}, to which we will present one solution in the next section.

% Intuitively, all SVs except for whole chromosome/genome duplications or episomal amplifications result in the breakage of DNA and forming breakends. When new phosphodiester bonds form, there must be two breakends .

%  For lack of better alternative, most early cancer SV studies hitherto treat these two facets of the same rearranged sequence separately or informally combine them to describe certain patterns \cite{stephens2011,baca2013}. The PCAWG consortium made fundamental effort to build up a pool of CNA and junction patterns for different classes of SV events, but still .... 

%===========================%
\clearpage
\section{Junction Balance Analysis infers copy numbers on genome graphs from short-read whole genome sequencing}

We developed an algorithm, JaBbA, to accurately infer junction-balanced genome graphs from WGS data. In the following three subsections, we will 1) formally define this algorithm \textit{junction balance analysis}, 2) describe the pipeline implementing the algorithm, and 3) comprehensively compare its features and performance to other methods of similar purpose.

\begin{figure}[!h]
    \centering
    \includegraphics[height=0.95\textheight]{figures/600ppi/jabba_io.png}
    \caption{\textbf{Inputs and outputs of Junction Balance Analysis}}
    \label{fig:jabba_io}
\end{figure}
\clearpage
% {Top left, read depth on genomic bins and junctions are the two required input data summarized from a WGS. Bottom left, an example locus showing read depth as scatter plots over reference coordinates and input junctions as red curves connecting the breakends. Top right, the network layout of the genome (sub)graph of the highlighted locus built from the junctions. Bottom right, the junction balanced genome graph (JBGG) output with integer CN for segments on the Y axis and junction CN implied.}

%Analogous to previous approaches, we define a (directed) genome graph of vertices representing strands of genomic segments and edges  representing a pair of 3' and 5' DNA ends that are adjacent in the reference genome (REF edge) or connected through rearrangement (ALT edge). A \textit{junction balanced genome-graph} assigns every vertex and edge in the graph an integer CN, while enforcing the constraint that the dosage of every interval (vertex) is equal to the sum of the JCN of the incoming (or similarly, outgoing) junctions (\textbf{Fig. \ref{fig:jabba_io}}).  The MIP optimization solved by JaBbA minimizes the residual between observed read depth and inferred interval dosage through joint assignment of CN to intervals and junctions (\textbf{Fig. {fig:Fig1}B, Fig. {fig:S1}C}, sec:methods).

\begin{figure}[!h]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{A schematic of the mixed integer quadratic program optimization problem that JaBbA solves.}
        \label{fig:jabba_formulation}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/jabba_formulation.png}
    \end{adjustbox}
\end{figure}
\clearpage

\subsection{Formulation of the mixed-integer quadratic programming problem} \label{sec:jabba_formulation}
% \subsubsection*{Inferring Junction Balanced Genome Graphs}
We infer JBGGs from a genome graph $G$ and binned, normalized, and purity / ploidy-transformed read depth data $x \in \mathbb{R}^n$ across $n$ genomic bins (see below for read depth transformation details)  through the solution of a mixed integer quadratic program (MIQP), which assigns an integer CN $\kappa:V_I \cup E \rightarrow \mathbb{N}$ to the vertices and edges of $G$.  The genome graph $G$ is generated, as above, from a set of breakends $\mathcal{B}_{seg}$ obtained from a preliminary segmentation of genome-wide read depth (i.e. via segmentation software such as CBS) and a set of junctions $\mathcal{A}$ (i.e. from a junction caller such as SvABA or DELLY) (Figure \ref{fig:jabba_io}). In Figure \ref{fig:jabba_io}: Top left, read depth on genomic bins and junctions are the two required input data summarized from a WGS. Bottom left, an example locus showing read depth as scatter plots over reference coordinates and input junctions as red curves connecting the breakends. Top right, the network layout of the genome (sub)graph of the highlighted locus built from the junctions. Bottom right, the junction balanced genome graph (JBGG) output with integer CN for segments on the Y axis and junction CN implied.

Each vertex $v \in V_I(G)$ is associated with a partition of bins $J(v) \subseteq \{1, \ldots, n\}$ (based on genomic coordinate overlap) and a mean bin value $\rho(v) = \frac{1}{|J(v)|}\sum_{j \in J(v)} x_j$.  We model each bin subset $x_{J(v)}$ as an i.i.d. sample from a Gaussian distribution with standard deviation $\sigma(v)$ and mean $\kappa(v)$.  The log likelihood is

\begin{equation} \label{eq:nodeloglikelihood}
    log P(x_{J(v)} | \kappa(v), \sigma(v)) =  \sum_{j \in J(v)} log \ \mathcal{N}(x_j | \kappa(v), \sigma(v)^2) = - \mathcal{V}(v, \kappa, x, J) + Const(\kappa)
\end{equation}

where $\mathcal{N}(\mu, \sigma^2)$ is the Gaussian probability density function with mean $\mu$ and variance $\sigma^2$ and $\mathcal{V}(v, \kappa, x, J) =  \frac{|J(v)|}{2\sigma(v)^2}(\rho(v)- \kappa(v))^2$ is the \textit{read depth residual} of vertex $v$. The variance $\sigma^2(v)$ is a $\kappa$-independent parameter that models read depth noise and is computed directly from the data.  The simplest noise model is a constant, where this parameter is set to the genome-wide sample variance of the read depth around each vertex mean: $\sigma^2(v) = \hat{\sigma}^2 = \frac{1}{n-1} \sum_{v \in V_I} \sum_{j \in J(v)} (x_j - \rho(v))^2$.  In practice, we apply a vertex-specific variance estimate $\sigma^2(v)$ to account for heteroscedasticity in the read depth data (see "JaBbA model fitting" section below).

Given this model, the joint log-likelihood of the read depth data $x$ across the graph given copy number assignment $\kappa$ is
\begin{equation} \label{eq:jointloglikelihood}
    log P(x | \kappa) = - \sum_{v \in V_I} \mathcal{V}(v, \kappa, x, J) + Const(\kappa)
\end{equation}
We also refer to $V(G, \kappa, x, J) = \sum_{v\in V_I} V(v, \kappa(v), x, J)$ as the \textit{read depth residual} of the JBGG $(G, \kappa)$ relative to data $x$.

The satisfaction of junction balance and skew-symmetry constraints in Eq. \ref{eq:junction_balance_constraint}-\ref{eq:skew_symmetry} may place nonzero copy number at one or more loose end edges.   Each loose end in the input graph represents a slack variable that allows the junction balance constraint to be relaxed at specific internal vertices, allowing the data to be fit even when junctions are missing from the input (e.g. due to low mappability, sequencing depth, or purity).  Only loose ends that are given nonzero CN, are considered to be "used" in the final graph. To penalize solutions that require the use of many loose ends, we add an exponential prior with decay parameter $\lambda$ on the loose end CN in $(G, \kappa)$, which makes models with many missing junctions unlikely.  This prior has log likelihood

\begin{equation} \label{eq:looseendpenalty}
    log P(\kappa | G, \lambda) = -|V_I|log\lambda - \lambda \mathcal{R}(G, \kappa)
\end{equation}
where
\begin{equation} \label{eq:complexitypenalty}
    \mathcal{R}(G, \kappa) = \sum_{v \in V_I} \mathcal{R}(v, \kappa) = \sum_{v \in V_I} \kappa(E_L^-(v))  + \kappa(E_L^+(v))
\end{equation}

is a \textit{complexity penalty}. Adding the log likelihood in  Eq. \ref{eq:jointloglikelihood} to the prior in Eq. \ref{eq:looseendpenalty} yields a penalized log likelihood for the data with regularization parameter $\lambda$. Under this model, the maximum a posteriori probability (MAP) estimate of $\kappa$

\begin{equation} \label{eq:objectivefunction}
    f(G, \kappa, x, J, \lambda) = \mathcal{V}(G,\kappa, x, J) + \lambda \mathcal{R}(G, \kappa)
\end{equation}

which combines the quadratic read depth residual $\mathcal{V}$ and $\ell_1$-norm  complexity penalty $\mathcal{R}$ into a single quadratic objective.   In practice, we apply models that penalize the number of loose ends with nonzero copy number, i.e. applying an $\ell_0$-norm penalty  $\mathcal{R}(\kappa) = \sum_{v \in V_I} [\kappa(E_L^-(v))>0]  + [(\kappa(E_L^+(v))>0])$.  We use $f$ to define a MIQP, which we solve to infer a MAP estimate for $\kappa$ given data $x$ and genome graph $G$:

\begin{equation} \label{eq:MIQP}
    \begin{aligned}
         & \underset{\kappa: V_I \cup E \rightarrow \mathbb{N}}{\text{minimize}}
         &                                                                       & f(G, \kappa, x, J, \lambda)                                                                                                                               \\
         & \text{subject to}
         &                                                                       & \kappa(v) = \kappa(\bar{v}),\ \forall_{v \in V_I}                                                                                                         \\
         &                                                                       &                                                   & \kappa(e) = \kappa(\bar{e}),\ \forall_{e \in E}                                                       \\
         &                                                                       &                                                   & \kappa(v)= \sum_{e\in E^-(v)} \kappa(e)         & = \sum_{e\in E^+(v)} \kappa(e), \forall_{v \in V_I} \\
    \end{aligned}
\end{equation}



% \begin{figure} \label{fig:jabba_formulation}
%     \centering
%     \includegraphics*[width=0.95\textwidth]{figures/600ppi/jabba_formulation.png}
%     \caption{\textbf{JaBbA formulation.}}{A schematic of the mixed integer quadratic program optimization problem that JaBbA solves.}
% \end{figure}

The resulting MAP estimate $\hat{\kappa}$ defines the JBGG $(G,\hat{\kappa})$ which is outputted and returned to the user (Figure \ref*{fig:jabba_formulation}).


\subsection{JaBbA pipeline}
Overall, the pipeline can be split into 3 parts, 1) building a primitive genome graph from the primary segmentation of coverage, 2) fitting integer CN for vertices and edges by solving the MIQP problem, and 3) simplify the output graph and report in the form of gGraph of gGnome package (see section \ref{sec:implement_ggnome}).

Junction Balance Analysis (JaBbA, \url{https://github.com/mskilab/JaBbA}) is an R package freely available under the MIT license. The required inputs to JaBbA are binned (e.g. 200bp) and normalized read depth data $y$, purity $\alpha$, ploidy $\tau$, a set of junctions $\mathcal{A}$ (see mathematical formulation above), and a hyperparameter $\lambda$. The key output is a \texttt{gGraph} object representing a junction-balanced genome graph (i.e. solution to \textbf{Eq. \ref{eq:MIQP}}) which can be queried and analyzed using downstream algorithms, e.g. SV event classification algorithms in the \texttt{gGnome} package (\url{https://github.com/mskilab/gGnome}, see "Structural variant event classification" section below). The workflow of JaBbA is composed of four phases: read depth preprocessing, graph building, JaBbA model fitting, and postprocessing.

\subsubsection*{Read coverage preprocessing, primary segmentation, and purity-ploidy estimation}
Raw input read depth data are generated for tumor and matched normal (i.e. constitutional) samples by tallying the count of midpoints of well aligned (MAPQ $>$ 0) proper read pairs in a vector $z \in \mathbb{Z}^n_+$ (e.g. $n \approx$ 15 million for 200 bp bins across hg19).  These are further GC-corrected and mappability normalized through an iterative LOESS fitting procedure similar to \cite{Ha:2012kr} and implemented in the fragCounter R package (https://github.com/mskilab/fragCounter). Briefly, a local estimated scatterplot smoothing (LOESS) function $f_{GC}$ is fitted (R \ttt{stats} package \ttt{loess} function) to a random (100,000) subsample of bins to predict read depth as a function of GC content. Each bin's read depth $z_j,~j \in \{1, ... n\}$ is then divided by its fitted value $f_{GC}(z_j)$ to yield the normalized value $z^{GC}_j$.  A similar LOESS-based procedure is then applied to correct each GC-corrected read depth $z^{GC}_j$ with respect to its 100mer mappability score to yield the GC and mappability corrected read depth $z^{GC}_j$.  This procedure is applied separately for tumor and matched normal (if applicable).

The ratio of the GC and mappability corrected read depth profiles for the tumor $z^T$ and normal $z^N$ is then used to derive the normalized read depth profile $y \in \mathbb{R}^n_+$ where $y_j = \frac{z_j^Tm_j^N}{z_j^N},~j \in \{1, ... n\}$ and $m^N_j$ is the chromosomal-wide median in the normal read depth data for the chromosome containing bin $j$.  Correction of the tumor / normal ratio by $m^N_j$ maintains a linear relationship between read depth and locus abundance in the original sample. Without this final step, bins from the X and Y chromosomes in a diploid male tumor will show the same read depth as the autosomes.  Samples without a matched normal (e.g. cell lines) are divided by a "universal" (i.e. average) normal averaged across 943 GC and mappability corrected TCGA normal read depth profiles and then chromosome balanced.


We note that the read depth normalizations, described above, are formally performed upstream of the JaBbA pipeline (e.g. via fragCounter, R / Bioconductor). The goal of these normalizations is to reduce biases, with the goal of rendering the binned read depth $y_j$ proportional (Â± noise) to the abundance of locus $j$ in the tumor sample (which may include admixed normal cells).  The final (purity / ploidy) transformation, included formally as part of the JaBbA pipeline, will renders the normalized read depth approximately equal (Â± noise) to its clonal integer CN in the tumor (given an accurate purity / ploidy estimate). Given a tumor sample with purity $\alpha \in [0, 1]$, ploidy $\tau \in \mathbb{R}_+$, and normalized read depth data $y$, the read depth $y_j$, tumor CN $\kappa(j)$, and constitutional CN $\kappa_N(j)$ will obey:

\begin{equation} \label{eq:ppbins}
    \frac{y_j}{\sum_{\hat{j} = 1}^n y_{\hat{j}}} \approx \frac{\alpha \kappa(j) + (1-\alpha)\kappa_N(j)}{\sum_{\hat{j} = 1}^n \alpha \kappa(\hat{j}) + (1-\alpha)\kappa_N(\hat{j})}
\end{equation}

Given $\bar{y} = \frac{1}{n}\sum_{\hat{j} = 1}^{n} y_{\hat{j}}$, $\tau \approx \frac{1}{n}\sum_{\hat{j} = 1}^{n} \kappa(\hat{j})$, and a near diploid constitutional sample (i.e. in which $\frac{1}{n}\sum_{\hat{j} = 1}^{n} \kappa_N(\hat{j}) \approx 2$),

\begin{equation} \label{eq:ppbins2}
    y_j \approx \frac{\alpha\kappa(j) + (1-\alpha)\kappa_N(j)}{\alpha \tau + 2(1-\alpha)} \bar{y}
\end{equation}

Defining

\begin{equation} \label{eq:purityploidycorrection}
    \begin{split}
        \beta = \frac{\bar{y}\alpha}{\alpha\tau+2(1-\alpha)} \\
        \gamma = \frac{2\bar{y}(1-\alpha)}{\alpha\tau+2(1-\alpha)}
    \end{split}
\end{equation}

yields the following equation:

\begin{equation} \label{eq:ppbins3}
    \kappa(j) \approx x_j = \frac{2y_j - \kappa_N(j)\gamma}{2\beta}
\end{equation}

Given known constitutional copy number (in practice, $\kappa_N(j) = 2$ for all bins with the exception of X and Y chromosome in males, where $\kappa_N(j) = 1$), the right hand side of Eq. \ref{eq:ppbins3} is computed directly from the (GC and mappability corrected, tumor / normal ratio transformed) read depth data, purity, and ploidy and represents the binned read depth input $x$ to JaBbA.

\subsubsection*{Graph building}
To build the genome graph $G$, the binned read depth $x$ is segmented (e.g. using CBS) to yield the breakends $\mathcal{B}$.  These breakends are combined via \textbf{Algorithm \ref{algo:buildgraph}} with the junctions $\mathcal{A}$ to yield $G$.  Briefly, this procedure 1) divides the reference genome into internal vertices $v_I \in V_I$ each of which will be assumed to have a coherent CN, and 2) establishes edges $e \in E=E_{R} \cup E_{A} \cup E_{L}$ that each represents an adjacency consistent with the reference genome (REF edge, $E_R$), created by an rearrangement junction (ALT edge, $E_A$), or an unmatched breakend (loose end, $E_L$). The ends of the vertices are the union of junction breakends $getBreakPoints(\mathcal{A})$ and a primary segmentation of the genome $\mathcal{B}_{seg}$ using the Circular Binary Segmentation algorithm \cite{olshen2004} (CBS), or from user input.  The mapping $J(v)$ (see above) is constructed to associate internal vertices $v \in V_I(G)$ with partitions of bin indices.



\subsubsection*{JaBbA model fitting}
Given a genome graph $G$ and GC / mappability corrected, normalized, purity / ploidy transformed read depth data $x \in \mathbb{R}^n$, and bin set $J(v) \subseteq \{1, \ldots, n\},~v \in V_I(G)$ , we infer a junction balanced genome graph $(G, \kappa)$ using the procedure outlined above ("Inferring junction-balanced genome graphs" section) with the following modifications:  Instead of the arithmetic mean $\rho(v)$, we use the sample geometric mean $\rho(v) = exp(\frac{1}{|J(v)|}\sum_{j \in J(v)} ln(x_j))$, which is more robust to outliers.  To model heteroscedasticity in read depth, we compute a vertex specific variance function $\sigma^2(v)$. Specifically, we use LOESS to fit a smooth non-linear function $f$ that links copy number to variance by fitting the sample mean $\rho(v)$ and sample variance  $\hat{\sigma}^2(v) = \frac{1}{|J(v)|-1} \sum_{x \in J(v)} (x_j - \rho(v))^2$ across all $v \in V_I$.  This function is then used to determine the segment-specific variance term as $\sigma^2(v) = f(\rho(v))$.  These $\sigma^2(v)$ values are then used to populate the read depth residual sum of squares $\mathcal{V}(v, \kappa, x, J)$ portion of the objective function in \textbf{Eq. \ref{eq:nodeloglikelihood}}. Combining with the user-defined loose end penalty hyperparameter $\lambda$ (default 100, tuned for 200 bp binned tumor / normal pairs, see above), the MIQP problem (\textbf{Eq. \ref{eq:MIQP}}) is solved on the genome graph using an $\ell_0$ complexity penalty. The solver is \texttt{CPLEX} (v12.6.2, IBM).

To increase sensitivity for missed junctions, the JaBbA pipeline employs several iterations of MIQP inference when two tiers of junctions (high and low confidence) are provided.  Briefly, we fit a model using only high-confidence (e.g. FILTER=PASS in the unfiltered \texttt{SvABA} output VCF) junctions as input.  In subsequent iterations, we fit a modified model that includes additional low-confidence junctions (e.g. those with low read support) that occur near (<1 kbp) a loose end in the previous iteration.  This procedure is run for a maximum of four iterations, or until no additional low confidence junctions near loose ends from the previous iteration have been found.   Such a procedure particularly improves junction sensitivity in the setting of low purity and / or read depth.

% \subsubsection*{Graph partitioning}


\subsubsection*{Post processing and allelic CN fitting}
After the optimization with \texttt{CPLEX}, we further simplify the solution by merging neighboring vertices of the same total CN that are connected only through REF edges. If ALT and REF read counts at constitutional heterozygous SNP sites are provided as (optional) input to JaBbA, we also infer the tumor allelic CN $\kappa_{h}(v)$ and  $\kappa_{l}(v)$ corresponding to the "high" ($h$) and "low" ($l$) allele, where  $\kappa_{h}(v) + \kappa_{l}(v) = \kappa(v)$ and $\kappa_{l}(v) \le \kappa_{h}(v)$, for each internal vertex $v \in V_I$ that overlaps a heterozygous SNP.  (Only biallelic heterozygous SNPs with two constitutional states, i.e. ALT and REF are considered.)

Briefly, we model the "low" count at each heterozygous SNP ("low" can be REF or ALT, depending on which allele has fewer reads) overlapping $v$ as a Poisson distribution with mean $\beta\kappa_l(v) + \frac{1}{2}\kappa_{lN}(v)\gamma$.  Here, $\kappa_{lN}(v)$ is the constitutional allelic CN of the lower CN allele at each vertex $v$ (e.g. 0 for X and Y chromosome in males, 1 elsewhere in the genome), and $\beta$ and $\gamma$ are defined as for (non-allelic) read depth (\textbf{Eq. \ref{eq:purityploidycorrection}}) with the following adjustments: genome-wide average $\bar{y}$ is the mean allele-specific read count across all heterozygous SNPs, and $\tau$ is the mean non-allelic CN across heterozygous sites only. An analogous Poisson model is defined for the "high" allele (i.e. REF or ALT, depending which has more reads) at each heterozygous SNP associated with $v$.  A joint Poisson log likelihood for the allelic CN of a given vertex $v$, $\kappa_{h}(v)$ and $\kappa_{l}(v)$, is then computed by summing all log likelihoods across both high and low alleles of all heterozygous SNPs mapped to $v$. The maximum likelihood configuration for $\kappa_{h}(v)$ and $\kappa_{l}(v)$ is chosen and reported.  The process is repeated across all internal vertices $v \in V_I(G)$ with at least one heterozygous SNP.

The final output is then saved into a gGraph object that can be manipulated, visualized, and analyzed with the \texttt{gGnome} R package (\url{https://github.com/mskilab/gGnome}) and \texttt{gGnome.js} browser (\url{https://github.com/mskilab/gGnome.js}).

\subsection{JaBbA robustly produce accurate copy numbers for DNA segments and junctions} \label{sub:jabba_benchmarking}
Having described JaBbA algorithm and pipeline, we next set out to comprehensively evaluate its performance against other tools of similar purposes.

% a table summarizing all the graph inference methods
% In Table \ref{tab:jbgg_inference}, we compare the existing methods to infer JBGG from WGS data.

% \begin{table}[] \label{tab:jbgg_inference}
%     \begin{tabular}{@{}lllll@{}}
%     \toprule
%     Method & Scope & Phased & Subclone & Solver & Purity, ploidy & Junctions & Publication \\ 
%     \midrule
%     JaBbA & genome & no & no & MIQP & inferred or given & given & \cite{Hadi2020-um} \\
%     PREGO & genome & no & no & ILP & fixed? & given & \cite{Oesper2012-vw} \\
%     ReMixT & genome & yes & yes & Variational Inference & inferred & given & \cite{McPherson2017-ry} \\ 
%     Weaver & genome & yes & no & LBP & inferred & inferred & \cite{li2017} \\ 
%     RCK & genome & yes & yes & MILP & inferred & given & \cite{Aganezov2020-cu}\\ 
%     InfoGenomeR & genome & yes & no & ILP & inferred & given & \cite{Lee2021-rl} \\
%     AmpliconArchitect & amplicon & \\    
%     CouGaR \\
% % These following two are karyotype reconstruction given a graph
% %    AmpliconReconstructor \\
% %    CCR \\
%     \bottomrule
%     \end{tabular}
% \end{table}

\newpage

% \centering        
% \rotatebox[]{90}{


\begin{table}
    \centering
    \caption{\textbf{Comparison of JBGG reconstruction methods.}}
    {Except for AmpliconArchitect and CouGaR, which only reconstruct complex amplicons, most methods build genome-wide JBGGs. MIQP, mixed-integer quadratic programming; ILP, integer linear programming; LBP, loopy belief propagation; MILP, mixed-integer linear programming.}
    \label{tab:jbgg_inference}
    \begin{tabular}{@{}cccc@{}}
        \toprule
        Method            & Scope    & Solver                & Publication             \\
        \midrule
        JaBbA             & genome   & MIQP                  & \cite{Hadi2020-um}      \\
        PREGO             & genome   & ILP                   & \cite{Oesper2012-vw}    \\
        ReMixT            & genome   & Variational Inference & \cite{McPherson2017-ry} \\
        Weaver            & genome   & LBP                   & \cite{li2017}           \\
        RCK               & genome   & MILP                  & \cite{Aganezov2020-cu}  \\
        InfoGenomeR       & genome   & ILP                   & \cite{Lee2021-rl}       \\
        AmpliconArchitect & amplicon & ILP                   & \cite{Deshpande2019-gs} \\
        CouGaR            & amplicon & ILP                   & \cite{Dzamba2017-wo}    \\
        % These following two are karyotype reconstruction given a graph
        %    AmpliconReconstructor \\
        %    CCR \\
        \bottomrule
    \end{tabular}            
\end{table}
% }


\newpage

\subsubsection*{Comparison to previous genome graph reconstruction methods}
% TODO!
To my best knowledge, there have been 8 published methods to reconstruct JBGGs from WGS data (Table \ref{tab:jbgg_inference}). AmpliconArchitect and CouGaR work only within amplified regions of the genome, while the other methods all infer JBGG genome-wide. Most adopt a undirected variation of the interval graph structure, while JaBbA is built upon a skew-symmetric directed graph. Weaver and ReMixT solve the problem with probabilistic graphical models, and the others utilize different variations of integer programming.

% rewrite this!!!
JaBbA differs primarily from previous genome graph methods in its robust modeling of WGS read depth noise and missing (i.e. false negative) junctions within a MIQP framework. First, while previous approaches (PREGO \cite{Oesper2012-vw}, CouGaR \cite{Medvedev:2010bm}, ReMixT \cite{McPherson2017-ry}, and Weaver \cite{Li2016-qa}) directly model read counts as a Poisson (or gamma-Poisson) random variable, JaBbA gleans non-linear mean-variance relationships empirically from high-resolution fixed bin (200 bp) read depth data.  This allows JaBbA to utilize transformed and normalized read depth data (e.g. via GC and mappability correction, tumor / normal ratio, purity and ploidy transformation).  Such data violates Poisson assumptions (e.g. integer variable, linear or constant relationship between mean and variance) but more faithfully reflects CN in stromally admixed and aneuploid tumor genomes.

Second, JaBbA is robust to missing junctions by allowing (but penalizing) the use of \textit{loose ends} in its reconstructions.  Loose ends arise when a CN change is unaccompanied by a nearby junction. Such an event is a common occurrence in WGS data (e.g. due to low coverage, algorithmic filters, rearrangements in repetitive regions, etc.). Methods that do not explicitly model loose ends (e.g. PREGO, Weaver) suffer from either under- (PREGO) or over-segmentation (Weaver). Finally, by posing its inference as a MIP, JaBbA avoids analytic limitations inherent to PGM inference (ReMixT, Weaver), including the inability to identify global (expectation maximization, ReMixT) or local (loopy belief propagation, Weaver) optima or model very high-level CN states (ReMixT).  In contrast, JaBbA's MIP-based framework allows for the discovery of globally optimal model fits while allowing for an unrestricted range of CN states.

\subsubsection*{Benchmarking JaBbA}
\begin{figure}[!h]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/600ppi/jcn_accuracy_sim.png}
    \caption{\textbf{Junction copy number accuracy.}}{Benchmarking across 40 simulated genomes and HCC1954, with four technical replicates at ten different sample purity levels by admixing with matched normal genome. The numbers in parentheses after the names of the methods indicate the numbers of successful runs out of 40. F1 score is the harmonic mean of precision and recall. Same for Figure \ref*{fig:junc_inc_f1}, \ref*{fig:cn_bp_f1}, \ref*{fig:cn_mse}, \ref*{fig:cn_scatter}. This figure shows the accuracy of JCN estimation for simulated genomes only (fraction of junctions with correctly estimated JCN).}
    \label{fig:jcn_accuracy_sim}
\end{figure}

\begin{figure}[!h]
    \centering
    \includegraphics[height=0.8\textheight]{figures/600ppi/junc_inc_f1.png}
    \caption{\textbf{F1 score of junction incorporation.}}
    \label{fig:junc_inc_f1}
\end{figure}

\begin{figure}[!h]
    \centering
    \includegraphics[height=0.8\textheight]{figures/600ppi/cn_bp_f1.png}
    \caption{\textbf{F1 score of copy number change point placement.}}
    \label{fig:cn_bp_f1}
\end{figure}

\begin{figure}[!h]
    \centering
    \includegraphics[height=0.8\textheight]{figures/600ppi/cn_mse.png}
    \caption{\textbf{Root mean square error of estimated CN from gold standard.}}
    \label{fig:cn_mse}
\end{figure}

\begin{figure}[!h]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/cn_scatter.png}
    \caption{\textbf{Estimated versus gold standard CN in all successful runs.}}{Pearson correlation coefficients $\rho$ using all points are labeled.}
    \label{fig:cn_scatter}
\end{figure}

\begin{figure}[!h]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/example_bfb.png}
    \caption{\textbf{Reconstructed genome graphs around \textit{HER2} in HCC1954.}}{Example of a putative BFBC event in HCC1954 and the reconstruction by five different methods (top five) and the binned read coverage transformed in CN space (bottom).}
    \label{fig:example_bfb}
\end{figure}

In the simplest terms, reconstructing genome graphs consists of two tasks: estimating junction copy numbers and DNA segment copy numbers. A junction is said to be incorporated in the genome graph if it is assigned non-zero copy number. Thus, a genome graph reconstruction method's performance can be evaluated from these two main aspects, 1) incorporating correct junctions and estimating the correct JCNs, and 2) faithfully segmenting the genome and estimating their CNs. We compared JaBbA's performance in both aspects against three other genome graph reconstruction methods (PREGO \cite{Oesper2012-vw}, ReMixT \cite{McPherson2017-ry}, Weaver \cite{Li2016-qa}), and specifically in segmental CN estimation with two extra somatic CNA callers that do not infer genome graphs (BIC-seq \cite{Xi2011-oa}, CONSERTING \cite{Chen2015-sw}). Another genome graph reconstruction method, CouGaR, is limited to amplified regions, so we only applied it to one replicate of the HCC1954 to demonstrate its difference from the other methods at a putative BFB cycles locus (Figure \ref{fig:example_bfb}).
%% \texttt{FACETS}, \texttt{TITAN}, \texttt{FREEC}


The input junctions to all genome graph reconstruction methods were their respective default recommendations, namely SvABA unfiltered candidates for JaBbA; internal junction callers for Weaver, CouGaR; and the union of 3 junction calls (SvABA filtered \cite{wala2018}, Delly \cite{Rausch2012-ly}, Novobreak \cite{Chong2017-gl}) for PREGO, ReMixT, and CONSERTING as they make no default recommendations.  As a note, we saw very similar performance with JaBbA using standard settings (SvABA unfiltered junctions) and the union junction set (data not shown).

\subsubsection*{Junction incorporation and JCN estimation}
To evaluate junction incorporation and JCN estimation, the incorporated junctions were approximately matched to gold standard junctions, e.g. both breakends were within 1 kbp from the corresponding gold standard junctions with the same orientations. True positives were the number of matches, false positives were incorporated junctions without a match, and false negatives were gold standard junctions without a match. An F1 score of the incorporated junctions, the harmonic mean of precision and recall, was then computed for each benchmarking sample (Figure \ref{fig:junc_inc_f1}). Gold standard junctions for HCC1954 were defined as the consensus set inferred from the original high depth data (junctions identified by at least two callers out of SvABA, Delly, Novobreak).

Inferred JCNs of the graph reconstruction were compared to the matching gold standard junctions, if any, from the simulated dataset. The proportion of correctly fitted JCNs out of all incorporated junctions times recall of incorporation represented the accuracy or completeness of JCN estimation (Figure \ref{fig:jcn_accuracy_sim}).

\subsubsection*{Segmental CN estimation}
When evaluating CNA inference performance, two metrics were considered. One was the correct placement of CN change points. Analogous to matching junction breakends, we considered an inferred CN change-point to match a gold standard if their distance was within 1 kbp and they have the same direction of CN change, e.g. increasing or decreasing CN from the side with smaller coordinates to the larger. To prevent spurious matching between inferred and gold standard CN change points, occurring in cases with excessive hypersegmentation, gold standard CN change points were only matched to the nearest identically oriented inferred CN change point.  Based on the matching, F1 scores were computed and shown in Figure \ref{fig:cn_bp_f1}. For HCC1954, the gold standard CN change-points were defined as the consensus junction breakends.

% Too many (hypersegmentation), too few (undersegmentation), too distant from the gold standard (large error margin) were all adverse indications. To prevent extreme cases of hypersegmentation being over-optimistic, each one of the gold standard CN change-points was only matched to the nearest correctly oriented inferred change point.

The other set of metrics reflects the concordance between the inferred CN profile with the gold standard. For each segment in the gold standard, the overlapping inferred segments were identified. The inferred CN for that gold standard segment was then defined as the overlap width-weighted average of inferred CN. Subsequently, the root mean squared error (RMSE) $\sqrt{\frac{\sum{(\hat{x} - x_{goldstandard})^2}}{n}}$ was computed across all gold standard segments for each sample (Figure \ref{fig:cn_mse}). The gold standard for HCC1954 CN profile was derived from published microarray-based segmentation downloaded from the CCLE data portal. High-depth WGS coverage data was mapped onto these segments and transformed into copy number space using the known ploidy of 4.5 and purity of 0.99. To compactly show the relationship between inferred CN and gold standard from all runs, we drew scatter plots across all gold standard segments in all successful runs for each method, and the Pearson $\rho$ of each method was calculated from all data points pooled together (Figure \ref{fig:cn_scatter}).

%===========================%
\section{Implementation of genome graphs in gGnome package} \label{sec:implement_ggnome}
To facilitate the analyses of genome graphs, we built an R package \textit{gGnome} which allows fast and useful creation, exploration, visualization, and computation around genome graphs. The basic design principle is to keep the integrity of genome graphs, in particular the skew-symmetric property, on the back end with private fields and methods in R6 classes, while expose the most intuitive inputs and outputs like junctions and segments to the users.

% package structure
% construction
We constructed two main classes, \texttt{gGraph} and \texttt{gWalk}, corresponding to genome graph and walks. In the most basic form, a genome graph can be initiated with a reference genome alone, where each node is a chromosome and no edges exist (\texttt{gG()}). On top of that, we can segment the whole genome based on a set of breakpoint coordinates, by automatically connecting two consecutive nodes with a (pair of reverse complement) REF edge (\texttt{gG(breaks = breakpoints)}). When a ALT junction emerges, it joins two breakends that are not adjacent in reference genome to form new (pair of reverse complement) ALT edges (\texttt{gG(juncs = junctions)}). One can create any genome graph from the input data of segmentation (in a format equivalent to \texttt{GRanges}), and/or a set of junctions (in a format equivalent to \texttt{GRangesList}). More often in practical scenarios, we start from a copy number-annotated genome graph inferred from WGS. gGnome allows one to parse the results from most of the existing genome graph reconstruction methods (JaBbA, ReMixT, Weaver, PREGO, RCK, CouGaR, AmpliconArchitect).

% subgraph
With a genome graph (supposedly stored in variable \texttt{gg}), we can make a series of queries based on vertex and edge metadata, with the graph subsetting operator "[,]", where the expression before the comma filters the vertices and the latter filters the edges. For example, when looking up amplified subgraphs where the vertex CN is larger than twice the ploidy (supposedly stored in variable \texttt{pl}), one can execute \texttt{gg[cn>pl*2,]}. 

% viz by gTrack
To quickly explore any part of a genome graph, we integrated our static genome browser-style plotting package \texttt{gTrack}. With any genome graph, one can simply call \texttt{gg\$gtrack} to get the \texttt{gTrack} object to then plot using \texttt{plot} function. All the optional arguments of \texttt{gTrack} can be passed down natively from this interface. For example, if one wants to plot copy number stored in the metadata column "cn" on the Y-axis, it is as simple as \texttt{gg\$gtrack(y.field = "cn")}. As a more powerful alternative, we also provide interactive visualization through \textit{gGnome.js}, described in the next section. Here, with any \texttt{gGraph} object, one can simply generate the required JSON format with \texttt{gg\$json()}.

% distance
Given a genome graph, can also calculate a lower bound  shortest derivative (in the rearranged genome) genomic distance between any two genomic loci after rearrangements, by summing up the width of vertices the path traversed (encoded in the gGraph method \texttt{gg\$dist}). Taking advantage of the optimal implementation of Floyd-Warshall by the \textit{igraph} R interface, this can be immediately scaled up to find all shortest paths from one set of genomic loci another set, encoded in the function \texttt{proximity}.

% clusters
One of the more interesting topics in SV pattern discovery is the clustered junctions. We can easily cluster vertices and/or edges based on their connectivity, implemented in \texttt{gg\$ecluster} for edges and \texttt{gg\$cluster} for vertices. The output cluster membership (weakly connected or strongly connected), will be recorded in a metadata column and can be used to retrieve the corresponding subgraph, for instance, \texttt{gg\$ecluster(); gg[,ecluster==1]}.

% maxflow
Besides, with a non-negative numeric field associated with vertices and edges, for example, copy numbers, we can also regard that as \textit{capacity} in traditional flow problems. For example, the \textit{max flow} problem with CN means the "maximum number of copies of such DNA molecules that connects genomic locus A and B".

% event calling
Next, we provide functions to replicate our algorithms in \cite{Hadi2020-um} and this dissertation for finding the subgraphs of 13 different SV events. The \texttt{events(gg)} function will automatically search the given graph for these patterns with the default parameters and annotate any findings in the metadata columns with the name of the event, like "chromothripsis". Plus, after running the function, there will be a new data table of all the findings appended to the field of \texttt{gg\$meta\$events}.

%===========================%
\section{Interactive visualization of genome graphs in arbitrary genomic windows}
% from JaBbA 
To visualize the genome graphs we developed \texttt{gGnome.js} (source code at \url{https://github.com/mskilab/gGnome.js}), a Javascript-based genome browser that renders genome graphs and allows for interactive browsing across discontiguous genomic intervals accompanied by various genomic annotations. Here, we describe several important and unique features of gGnome.js.

\clearpage


% gGnome.js figure
\begin{figure}[ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{A snapshot of the gGnome.js interface of a BFB cycles event in an esophageal adenocarcinoma sample.}
        \label{fig:ggjs}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/ggjs.png}
    \end{adjustbox}
\end{figure}

\clearpage
% {From the bottom to the top: X axis is the reference genome. Y axis is mapped to JaBbA-inferred copy number. Colored bars are vertices and links edges. Viewing windows can be created by selecting regions on the whole genome bar above the plotting region. Sample information is displayed along with the currently viewing windows. Toolbar at the very top allows for quick search of samples, focus on an annotated region, locate a specific range by coordinates or gene name, as well as controlling visible tracks in the browser.}
% 
First and foremost, the flexibility of arbitrary window browsing is crucial for complex rearrangement events, which can vary in scale and involve many distal loci. Annotations of events (or any set of vertices/edges) were read by the browser and can be searched. For example, in Figure \ref{fig:ggjs} we can visualize a multi-chromosomal breakage-fusion-bridge cycle (BFB cycles) event in 6 discontiguous windows spanning 4 different chromosomes and locate this event instantly with the \texttt{annotation} selector on the top right corner. In Figure \ref{fig:ggjs}, from the bottom to the top: X axis is the reference genome. Y axis is mapped to JaBbA-inferred copy number. Colored bars are vertices and links edges. Viewing windows can be created by selecting regions on the whole genome bar above the plotting region. Sample information is displayed along with the currently viewing windows. Toolbar at the very top allows for quick search of samples, focus on an annotated region, locate a specific range by coordinates or gene name, as well as controlling visible tracks in the browser.


Annotations corresponding to segment-level and also junction-level information were included in the browser. Metadata for segments includes, but is not limited to, total copy number, allele-specific copy number (for samples and intervals in which heterozygous SNP information is available, i.e. cases with both tumor and normal pairs), and correspondence to event type if the segment was identified as part of an event. Edge metadata includes information output by the SV caller used in this study, SvABA, including number of reads supporting the SV and junction topology. These metadata can all be presented when the cursor hover over a specific element.

\clearpage
Second, the \texttt{gGnome.js} browser enables visualizing any genome graph, either with CN as inferred by other graph callers, or even without CN in the case of plain genome graphs. 

As a meaningful default, we plot integer copy numbers of the segments on the Y axis, that are inferred from cancer whole genome sequencing (WGS) with JaBbA (described in the next section), making it ideal for quick browsing copy number aberrations and immediately reveal the architecture of the complex events. 

%Annotations on subgraphs are searchable and useful to mark interesting events (like chromothripsis). A sample level search engine accelerates the browsing on large cohorts. We can also plot possible karyotypes as walks within the graph assisting on the identification of functional neo-connectivities or clarification of complex event topologies.

Third, when visualizing a larger cohort an integrated tag filter helps narrow down desired samples by any metadata string, exemplified by primary cancer types, data source projects, mutation status of genes.

Fourth, gGnome.js also supports visualizing additional genome tracks in the format of scatter plot, bar plot, and gene annotations, whose visibility can be controlled by the functionnalities in the top right corner.

% newly written
%To bring genome graphs better intuition for practical use, we developed \textit{gGnome.js} (https://github.com/mskilab/gGnome.js), a web application that draws genome rearrangement graphs onto its reference coordinates and allows for interactive browsing in any number of discontinuous windows accompanied by various types of genomic annotations. By allowing arbitrary windows and navigation by connectivities, the users can traverse the genome freely and simultaneously inspect various data at multiple relevant locations, including numeric (like read coverage) and qualitative (like gene annotations).

%===========================%
\section{Discussion}
In this chapter, I defined a skew-symmetric direct graph representation of a rearranged genome, termed genome graph. Genome graphs are general enough to encapsulate any junction that has corresponding coordinates within the reference genome, and thus can represent SV events of arbitrary complexity. 

% it's flexible even though it does not store sequences, just change your reference
Compared to variation graphs \cite{Hickey2020-cu} such formulation is almost equally general, except for foreign sequences and variants that cannot be associated with a coordinate (e.g. small non-templated insertions), yet exceedingly light-weight as the vertices are stored as coordinates instead of the actual sequences. In fact, for SV events involving foreign sequences, for instance, viral genome integration, simply concatenating new sequences to the reference genome will allow them to have a coordinate to be represented. Although here I only described the usage of genome graphs in representing each cancer genomes with SVs, it is straightforward to extend to germline SVs. \cite{Siren2020-tf,Eizenga2020-cf}. Following this thought it is apparent that as long as there is a reference genome, the SVs arising in that genome can be represented using gGnome, immediately applicable to other organisms. In fact, we have built genome graphs for the SVs detected in SARS-Cov-2 viruses sampled from COVID-19 patients sequenced at Weill Cornell Medicine with the original strain as a reference \cite{Butler2021-mp}.

% how to extend the graphs
More importantly, this formulation does not assume the represented genome must be one genome or that the vertices coordinates are disjoint. The user can append extra metadata to expand the graph in creative ways. There are at least two important applications of this flexibility.

First, we can use the framework to represent fully or partially phased genome graphs. In fully phased genomes (to either parental alleles, not somatic phase), we would have two pairs of vertices representing the same DNA segment for each parental phase assginment. Same goes with edges, so that the edges of one allele will never be crossing over to the vertices of the other allele, following the reasonable assumption that there is no somatic recombination. In fact, Weaver, ReMixT, RCK, and InfoGenomeR have attempted to infer partially phased genome graphs from bulk WGS. gGnome should be compatible to represent these results and such feature will be released in the near future. The clear advantage of inferring germline phase of junctions is that we can tease apart SV events that are happening on different alleles yet overlapping on the reference coordinates. For example, in pyrgo instances, the final derivative sequences can be very different depending on whether some DUP-like junctions are in \textit{cis} versus \textit{trans} \cite{Hadi2020-um}.

Second, gGnome allows the joint analysis of multiple genome graphs. Such situation can arise in multi-sample study designs such as diverging cell clones (shown in detail in Chapter \ref{chap:tel_crisis}), spatio-temporal sampling of tissue \cite{Paulson2021-nd}, primary and metastatic sampling, or even germline genome sampled from populations like 1000 Genome Project \cite{Khurana2013-qc}. Simply annotate the source identity in a metadata field, concatenate (extend) the each graph, and disjoin the graph to reduce to a pan-genome graph. This allows for coupling of junctions, loose ends, or vertices across multiple samples and jointly infer their structures, which is very powerful representing related but changing SV events.

Beyond genome graphs, we also implemented walks over a genome graph that directly map to linear DNA sequences that a graph can encapsulate. The conversion between reducing walks to graphs, and generating walks from graphs have very profound impact on the analysis of SVs. For the former, any array of genomic regions, including arbitrarily constructed \textit{de novo}, simulated from certain SV evolution scenario, sequencing read mapped to multiple loci, can be natively regarded as a walk, then trivially converted to a graph based on the junctions implied between consecutive elements in that array. Conversely, we can decompose a JBGG following the constraints in Eq. \ref{eq:walk_decomposition} to infer an optimal karyotype that constitute such a genome graph by minimizing some form of loss function like Eq. \ref{eq:walk_decomposition_parsimony}.

Using this template, genome graphs and walks are ideal data structures to bridge the fragmented yet high-throughput short-read sequencing, and the long-range, long-read sequencing or optical mapping techniques. Junction-balanced genome graphs (JBGGs) are stoichiometrically plausible compressions of all possible derivative sequences that can result from the junctions they contain, and can be used to generate a pool of walks which provide substrates for finding an optimal linear combination of alleles like that maximizes parsimony or likelihood of generating an observed long-range pattern \cite{Hadi2020-um}. In \cite{Hadi2020-um}, we gave a form of the algorithm \ref{eq:walk_decomposition} where the objective is to find the combination of walks that maximizes the amount of support from long-molecule profiling. In the paper we used 10X Genomics linked-read sequencing and Bionano Optical Mapping technologies, but we note that this approach can be extend to other platforms and even combine orthogonal evidence.

% TODO
Even without the exact inference of linear alleles, genome graphs can produce useful bounds for many biological applications. Because when the genome graph is complete, the true walk must be contained within, and then we can approximate and bound the various quantities between two reference coordinates in the face of SVs. For example, the shortest possible distances bewteen two reference regions in a graph is the sum of traversed nucleotides along their shortest paths. In PCAWG thyroid carcinoma \cite{Panebianco2017-xt}, I recaptured the 5 THCA samples with the THADA-IGF2BP3 enhancer hijacking event reported by PCAWG \cite{Rheinbay2020-no}, and further discovered a case where the juxtaposition of THADA near IGF2BP3 is achieved by two junctions from a more complex event, which also elevated the expression of IGF2BP3 like in the cases found in. In another example, two gene with distance zero in the genome graph is putatively fused.

% %===========================%
% \section{Neo-connectivities on a genome graph encode putative functional events}
% As explained in the previous sections, a walk (and its reverse complement walk) on a genome graph maps to a linear DNA sequence. If there exists a possible walk between two vertices it means the genomic elements within the two vertices may be connected on a contiguous DNA segment in the sample, if and only if all the ALT edges along the walk are phased in cis. A pair of ALT edges phased in cis means there are at least one copy of them found on the same DNA molecule; while none such copy is found they are deemed phased in trans. In reality, due to the large size of the genome and the limited read-length or effective range (e.g. for linked reads), complete phasing of all junctions is near impossible for most cases. However, the connectivities between genomic regions on the un-phased or partially phased genome graphs can serve as an approximation.

% As we established in earlier sections, a complete genome graph should enclose all the walks corresponding to the studied genome, so if any copy of two genomic loci are on the same part of a DNA molecule, that segment should have a corresponding walk on the complete genome graph. Based on this observation, the new genomic distance after rearrangements between any copy of two loci is summing over the width of vertices plus the within-vertex distances to the incident ends along the walk for the starting and terminating vertices. Thus, even when it is not possible to exactly decompress the graph to the walks consistent with the actual karyotypes, we know that the real shortest genomic distance between any copy of two loci should not be shorter than the shortest path distance on the complete genome graph.

% Admittedly, there may never be a complete genome graph constructed with the current technologies, and it is always possible that we failed to identify certain junction like centromeric \cite{Cameron2021-db} or involving repetitive or foreign sequences \cite{Behr2021-gf} and those may cut some genomic distances even shorter. Nevertheless, as shown in \cite{Behr2021-gf}, the state of the art short-read WGS should be effective at catching the majority of somatic rearrangements. Plus, even with the imperfect genome graphs, we can already expand the repertoire of functional genomic events by including the complex events. We use this approximation to show that potential functional events of arbitrary complexity can be represented as walks on genome graphs, exemplified by bridged fusion genes and enhancer hijacking.

% Some fusion genes have long been established as cancer drivers and even successful drug targets \cite{Gao2018-nm}; some have also been shown to encode chimeric proteins that can trigger immune responses as neoantigens \cite{Rathe2019-cr}. Expectedly, most fusion genes found to date are created by one somatic junction, as it requires fewer junctions to create and has simpler topology to detect. However, in theory, there should be no biological restraint against a more complex, multi-junction fusion gene. Indeed, recent large pan-cancer whole genome and transcriptome profiling efforts have identified numerous such examples termed \textit{bridged fusions} \cite{PCAWG_Transcriptome_Core_Group2020-zp,Yun2020-bi}. Naturally following our above definition, any fusion genes, no matter how many somatic junctions is involved, can be represented as a walk $w = (v_1, e_1, v_2, ..., e_{k-1}, v_k)$ on the corresponding genome graph, where the 5â partner is at the outgoing end of $v_1$ and the 3â partner is at the receiving end of $v_k$ (Fig x). Here I show an example walk of xxxâxxx fusion found by the PCAWG Transcriptome working group \cite{PCAWG_Transcriptome_Core_Group2020-zp} along with the supporting reads from the RNA-seq. By our graph-based SV event classification, this junction was part of a large scale xx (described in the next chapter).

% Another class of emerging SV driver events are enhancer hijacking, where an active yet distal enhancer is juxtaposed near a dormant proto-oncogene, activating its expression. In thyroid carcinoma, the insulin-like growth factor 2 mRNA-binding protein 3 (IGF2BP3, chromosome 7p15.3) has been found to be activated and contribute to cell proliferation by being juxtaposed next to thyroid adenocarcinoma-associated (THADA, chromosome 2p21) gene \cite{Panebianco2017-xt} which is constitutively expressed in normal thyroid tissue. This event is later confirmed by PCAWG consortium \cite{Rheinbay2020-no} to be recurrent in at least 5 different thyroid carcinomas it profiled with WGS. I reconstructed all these 5 paths that consist of only one junction each (FIGURE!!!!)

The last major obstacle before realizing the great potential to analyze complex SVs by genome graphs is the reliable inference of JBGGs from real data. Thanks to methods like JaBbA \cite{Hadi2020-um}, InfoGenomeR \cite{Lee2021-rl}, the problem of inferring a JBGG from a short-read whole genome sequencing is a generically tractable and robust to most sample purities. JaBbA differs primarily from previous genome graph methods in its robust modeling of WGS read depth noise and explicit accounting of false negative junctions, also called loose ends (see Section \ref{sec:jabba_formulation}). In our benchmarking experiments, JaBbA inferred JCN with consistently higher fidelity than published genome graph-based methods (ReMixT \cite{McPherson2017-ry}, Weaver \cite{Li2016-qa}, PREGO \cite{Oesper2012-vw}) across a wide range of tumor purities (Figure \ref{fig:jcn_accuracy_sim}).

Of particular interest, in the HCC1954 experiment, JaBbA is the only caller that has a wide enough dynamic range to accurately capture the extremely high CN of the \textit{ERBB2} complex amplicon (Figures \ref{fig:bfb_example}, \ref{fig:cn_scatter}) while correctly placing the junctions. Because of that, we successfully identified the underlying BFB event the led to the pathogenic amplification. This is one of the benefits of using a linear programming formulation over probabilistic graphical models.

In addition, JaBbA qualitatively outperformed genome graph-based methods and even classic (i.e. non-graph based) CNA callers (BIC-seq \cite{Xi2011-oa} and CONSERTING \cite{Chen2015-sw}) in estimating interval CN and change point locations across a wide range of tumor purities (Figure \ref{fig:cn_bp_f1}) as it utilizes the extra precision from the junction topologies. These results show that JaBbA is the first genome graph SV caller to accurately infer the topology of JCN while approaching (or exceeding) the fidelity of classic CNA callers.

We argue that one of the major reasons behind JaBbA's superior performance is the use of loose end. Loose ends allow for flexibility in fitting CN closer to the center of the coverage data even when the corresponding junction is not detected at the breakpoint. Loose end penalty \textlambda arise from an exponential prior and is a single hyperparameter that changes JaBbA's fundamental behavior. When it approaches zero, loose ends are free and the output will end up hypersegmented with wrong CN fitted due to coverage noise. When it approaches infinite, loose ends overrides the attraction of CN to the center of the coverage data and the output will end up in a rigid single segment. Properly set loose end penalty will only allow the algorithm place one when the coverage change is compelling and might reflect real CN change point without an identified junction. These \textit{bona fide} loose ends are abundant in our pan-cancer cohort, and they represent the detection limits of junctions with short-read data. Behr et al. \cite{Behr2021-gf} made a deep dive into the underlying reasons of these loose ends and found both the technical limitation of junction detection introduced by non-unique read mapping, and real biological events like viral integration and neo-telomeres.

As in its present form, JaBbA have three major limitations. One, the main product out of the pipeline is total CN, and the allelic CN is done conditioned upon inferred total CN and independently for each vertex based on maximum likelihood of the observed the allelic read counts at heterozygosity sites. In fact, assuming infinite site, at each breakend, only one of the two parental alleles can be incident to an ALT junction and it is a feasible computational problem to put two alleles in \textit{cis} or \textit{trans} when there is allelic imbalance, thus producing partially phased genome graphs even from short-read data. This is also attempted by Weaver, ReMixT, RCK, and InfoGenomeR, but from the former two methods we benchmarked, it often comes at the cost of inaccuracy in total CN estimation. This point is independently implicated in the InfoGenomeR publication \cite{Lee2021-rl}, where JaBbA is the second most accurate estimator behind InfoGenomeR in all tested cases. Plus, as shown in the next chapter, even with total CN alone, we can already rigorously discover novel classes of complex SV events from pan-cancer cohorts.

Two, over complicated graphs (unusually high number of segments and junctions) rapidly increase the difficulty to reach the exact global optima. We currently mitigate this by decoupling the whole genome graph into partitions, where the vertices from different partitions are disconnected or only connected through a set of vertices whose CN is fixed, due to very small variance in their maximum likelihood estimated CN (or plainly, long segments with enough data to have a nearly definite CN). However, there is not a particular reason why this quadratic loss (L2 norm) should not be replaced with absolute values (L1 norm) and be solved as a MILP with much higher efficiency. We leave that for the next phase of development.

Three, the graph inferred by JaBbA is a consensus representation over the whole sample, which does not directly model intra-tumoral heterogeneity. The inferred integer copy numbers can be interpreted as "average copies per cancer cell" and when there is heterogeneity, which is found to be prevalent \cite{Watkins2020-qp}, the true values of this quantity could deviate from integers. The inference of subclonal integer CNAs with clonal fractions directly from bulk short-read WGS remains a challenging problem \cite{McPherson2017-ry} while maintaining faithful CN estimation. However, as we will discuss again in the next chapter, most of the SV patterns we can observe from bulk JBGGs are ancestral or from major clone. Besides, with isogenic cloning designs \cite{Dewhurst2021-jk} or the emerging single-cell WGS technologies \cite{Zahn2017-re,laks2019,Salehi2021-xi}, JaBbA stays a reliable and promising tool to reconstruct high-fidelity JBGGs from clonal populations.

We also implemented the genome graph data structure in an open source R package gGnome, and to our knowledge, it is the first in class API to genome graphs that can 1) allow creation, combination, arithmetic of genome graphs, 2) support various algorithms including but not limited to decomposition to linear alleles, finding shortest paths, cluster vertices and edges, 3) parse the results of all the JBGG inference methods into a centralized object, 4) visualize genome graphs statically and interactively.

The main goal of gGnome is to provide the basic operations needed to build practical applications and analysis. It hides the skew-symmetry implementation from the user using private fields and methods, and present the more straightforward form similar to that of \cite{Greenman:2012eg}, so that the construction, dissection, combination of graph objects are more intuitive and easily adaptable to various common inputs in bioinformatics, including segmentation in SEG, BED formats and junctions in BEDPE, BND VCF (v4.2) formats. The users can immediately combine the low-level functions to achieve biological meaningful procedures. For example, in the next chapter, our graph-based SV event taxonomy typically start with filtering the nodes or edges by a field (like CN above certain threshold), followed by partitioning the whole genome graph into candidate subgraphs, and then extract features from the subgraphs to make the final call, which is extra annotation in the vertex/edge metadata fields.

gGnome.js is also the first genome browser that allows viewing arbitrary windows along the reference, which is a critical feature to browse large-scale complex SV events. It has been instrumental in discovering the complex event taxonomy in the next chapter, as we were able to go through thousands of genome graphs rapidly and notice recurring patterns thanks to the ultrafast modern web-based graphics libraries (WebGL). It also greatly benefits the whole genome analysis in oncology where we can pinpoint the potential SV drivers almost instantly, either by dynamically changing the zoom level to see the CNA-affected regions, or directly browser through annotated events, or even search the graph by gene of interest. We realize the versatility of not only representing one graph per tumor, and is building the next version to visualize pan-genome graphs and several related genome graphs together, named Pan Genome Viewer (PGV, http://github.com/mskilab/pgv).

We believe this suite of packages including gGnome/gGnome.js/JaBbA should greatly help the research community routinely incorporate graph-based analysis of SVs into the analysis of whole genome sequencing.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Chapter 3: discovery of novel SV patterns
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter{Distinct Classes of Complex Amplicons Uncovered across Thousands of Cancer Genome Graphs *} \label{chap:complex_events}
In this chapter I will describe the classification of three distinct types of complex amplification events from 2778 genome graphs built with JaBbA from a pan-cancer cohort of over 30 types of cancers. Furthermore, I will show that the patients in this cohort can be stratified based on a total of 13 simple to complex genome graph-based SV patterns to clusters of differential overall survival, tumor type enrichment, and association with specific germline or somatic alterations.

This work is part of our publication \cite{Hadi2020-um}*, led by Kevin Hadi, Julie Behr, Marcin Imielinski, and myself.

\textit{
    \textbf{Individual contributions:} Kevin Hadi and I collected and processed the WGS and associated biological or clinical data for the pan-cancer cohort. I executed and controlled the quality of the output genome graphs. I contributed to the classification of amplicons and carried out the analysis of fusion genes. I contributed to the algorithms identifying the 13 SV event types. I carried out the patient clustering analysis and with Kevin Hadi discovered the associations with the tumor types, genotypes, and overall survival.
}

% \chapter[Evolution and phase of clustered deletions]{Evolution and phase of clustered deletions*}\label{chrig}
\footnotetext[1]{Hadi, K., Yao, X., Behr, J.M., et al. Distinct classes of complex structural variation uncovered across thousands of cancer genome graphs. Cell, 183(1):197â210, 2020.}

\newpage
\begin{center}
    \begin{longtabu} to 0.95\linewidth {cccc}
        \label{tab:pancan_cohort} \\
        \caption{\textbf{Pan-cancer WGS tumor type composition.}} \\
        \hline
        \textbf{Abbreviation} & \textbf{Tumor Type} & \textbf{Number}  & \textbf{Cell lines} \\
        \hline
        \endfirsthead
    
        \hline
        \multicolumn{4}{c}{Table \ref{tab:pancan_cohort} (Continued)}\\
        \hline
        \textbf{Abbreviation} & \textbf{Tumor Type} & \textbf{Number}  & \textbf{Cell lines} \\
        \hline
        \endhead
    
        \hline
        \endfoot
    
        \hline
        \multicolumn{4}{c}{End of Table}\\
        \hline\hline
        \endlastfoot
    
        AML          & Acute myeloid leukemia                & 56   & 6             \\
        BE           & Barrett's esophagus                   & 340  & 0             \\
        BLCA         & Bladder urothelial carcinoma          & 36   & 7             \\
        BRCA         & Breast invasive carcinoma             & 257  & 35            \\
        CESC         & Cervical squamous cell carcinoma      & 18   & 0             \\
        COAD         & Colorectal adenocarcinoma             & 101  & 22            \\
        ESAD         & Esophageal adenocarcinoma             & 432  & 1             \\
        ESSC         & Esophageal squamous cell carcinoma    & 19   & 8             \\
        GBM          & Glioblastoma multiforme               & 76   & 7             \\
        HNSC         & Head and neck squamous cell carcinoma & 62   & 6             \\
        KICH         & Kidney chromophobe                    & 50   & 0             \\
        KIRC         & Kidney renal clear cell carcinoma     & 61   & 7             \\
        KIRP         & Kidney renal papillary cell carcinoma & 40   & 0             \\
        LGG          & Lower grade glioma                    & 57   & 5             \\
        LIHC         & Liver hepatocellular carcinoma        & 66   & 10            \\
        LUAD         & Lung adenocarcinoma                   & 144  & 29            \\
        LUSC         & Lung squamous cell carcinoma          & 68   & 15            \\
        MALY         & Malignant lymphoma                    & 112  & 4             \\
        MELA         & Melanoma                              & 256  & 34            \\
        MESO         & Mesothelioma                          & 4    & 2             \\
        MM           & Multiple myeloma                      & 4    & 4             \\
        OTHER        & Other                                 & 10   & 8             \\
        OV           & Ovarian serous cystadenocarcinoma     & 73   & 23            \\
        PACA         & Pancreatic carcinoma                  & 27   & 9             \\
        PBCA         & Pediatric brain carcinoma             & 4    & 4             \\
        PRAD         & Prostate adenocarcinoma               & 115  & 3             \\
        SARC         & Sarcoma                               & 58   & 8             \\
        SCLC         & Small cell lung carcinoma             & 46   & 46            \\
        STAD         & Stomach adenocarcinoma                & 64   & 21            \\
        THCA         & Thyroid carcinoma                     & 62   & 2             \\
        UCEC         & Uterine corpus endometrial carcinoma  & 60   & 11            \\
        \hline
        \textbf{Total}  & & 2778 & 337          \\
        \hline
    \end{longtabu}
        
\end{center}


\clearpage
\begin{table}[ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}\caption{
        \textbf{Pan-cancer WGS datasets and sources.}
        \label{tab:pancan_dataset}
        }}{\end{minipage}},rotate=90,center}
                \begin{tabular}{cccccc} 
            \hline
            \textbf{Abbreviation} & \textbf{Total Number} & \textbf{Attempted JaBbA} & \textbf{Analyzed} & \textbf{Reference} & \textbf{Repository} \\
            \hline
            BARRETTS     & 347           & 340             & 340      & \cite{Paulson2021-nd}    & TBD                      \\
            CA           & 190           & 116             & 115      & \cite{Hadi2020-um}       & TBD                      \\
            IPM          & 140           & 83              & 80       & \cite{Hadi2020-um}       & TBD                      \\
            CCLE         & 330           & 326             & 322      & \cite{Ghandi2019-wx}     & SRA: PRJNA523380         \\
            ESAD-UK      & 423           & 422             & 422      & \cite{frankell2019}      & EGA: EGAD00001004137     \\
            KLUAD        & 49            & 49              & 49       & \cite{Lee2019-nm}        & EGA: EGAD00001004793     \\
            MALY         & 101           & 100             & 100      & \cite{Richter2012-ur}    & EGA: EGAD00001002123     \\
            MELA         & 183           & 183             & 183      & \cite{hayward2017}       & EGA: EGAD00001003388     \\
            MI           & 29            & 0               & 0        & \cite{Imielinski2012-vv} & dbGaP: phs000488.v1.p1   \\
            NZ           & 131           & 122             & 122      & \cite{Nik-Zainal2016-bz} & EGA: EGAS00001001178     \\
            TCGA         & 1021          & 1017            & 990      & \cite{Ding2018-fa}       & dbGaP: phs000178.v10.p8  \\
            BACA         & 57            & 55              & 55       & \cite{baca2013}          & dbGaP: phs000447.v1.p1   \\
            \hline
            Total        & 3001          & 2813            & 2778     &                          &                          \\
            \hline
            \end{tabular}

    \end{adjustbox}
\end{table}

% \begin{table}[!h]
%     \begin{adjustbox}{addcode={\begin{minipage}{\width}\caption{
%         \textbf{Pan-cancer WGS datasets and sources.}
%         \label{tab:pancan_dataset}}\end{minipage}},rotate=90,center}
%         \begin{tabular}{cccccc} 
%             \hline
%             \textbf{Abbreviation} & \textbf{Total Number} & \textbf{Attempted JaBbA} & \textbf{Analyzed} & \textbf{Reference} & \textbf{Repository} \\
%             \hline
%             BARRETTS     & 347           & 340             & 340      & \cite{Paulson2021-nd}    & TBD                      \\
%             CA           & 190           & 116             & 115      & \cite{Hadi2020-um}       & TBD                      \\
%             IPM          & 140           & 83              & 80       & \cite{Hadi2020-um}       & TBD                      \\
%             CCLE         & 330           & 326             & 322      & \cite{Ghandi2019-wx}     & SRA: PRJNA523380         \\
%             ESAD-UK      & 423           & 422             & 422      & \cite{frankell2019}      & EGA: EGAD00001004137     \\
%             KLUAD        & 49            & 49              & 49       & \cite{Lee2019-nm}        & EGA: EGAD00001004793     \\
%             MALY         & 101           & 100             & 100      & \cite{Richter2012-ur}    & EGA: EGAD00001002123     \\
%             MELA         & 183           & 183             & 183      & \cite{hayward2017}       & EGA: EGAD00001003388     \\
%             MI           & 29            & 0               & 0        & \cite{Imielinski2012-vv} & dbGaP: phs000488.v1.p1   \\
%             NZ           & 131           & 122             & 122      & \cite{Nik-Zainal2016-bz} & EGA: EGAS00001001178     \\
%             TCGA         & 1021          & 1017            & 990      & \cite{Ding2018-fa}       & dbGaP: phs000178.v10.p8  \\
%             BACA         & 57            & 55              & 55       & \cite{baca2013}          & dbGaP: phs000447.v1.p1   \\
%             \hline
%             Total        & 3001          & 2813            & 2778     &                          &                          \\
%             \hline
%             \end{tabular}
%         % \includegraphics[scale=.6]{example-image}%
%         % \includegraphics[width=\textheight]{figures/600ppi/ggjs.png}
%     \end{adjustbox}
% \end{table}
\clearpage
\newpage


\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/pancancer.png}
    \caption{\textbf{Pan-cancer WGS tumor types.}}{See Table \ref{tab:pancan_cohort} for abbreviations. $*$ marks datasets with multiple samples per patient.}
    \label{fig:pancancer}
\end{figure*}

\clearpage
\section{Analysis of pan-cancer junction-balanced genome graphs}
To investigate the topology of JCN across cancers, we assembled a dataset comprising 2,813 short-read WGS tumor or cell line samples spanning 31 primary tumor types (Table \ref{tab:pancan_cohort}), including WGS for 539 previously unpublished cases  (Table \ref{tab:pancan_dataset}). In total, our analysis included 1,648 WGS samples not included in the Pan-Cancer Analysis of Whole Genomes (PCAWG) effort \cite{pcawg_marker2020-yi}. Application of harmonized pipelines followed by JaBbA (Figure \ref{fig:jabba_io}) yielded 2,778 high quality genome graphs (Figure \ref{fig:pancancer}, see Appendix \ref{app:b} for sample characteristics and exclusion criteria).

\begin{figure*}[!h]
    \centering
    \includegraphics[height=0.95\textheight]{figures/600ppi/variant_subgraph.png}
    \caption{\textbf{Examples of chromothripsis and chromoplexy.}}{}
    \label{fig:variant_subgraph}
\end{figure*}
\clearpage

Analyzing junction-balanced genome graph topology, we identified subgraphs associated with previously identified complex rearrangement patterns such as chromothripsis, hromoplexy, and TICs (Figure \ref{fig:variant_subgraph}) implementing criteria described in previous publications within our framework (see Appendix \ref{app:b}).

\clearpage
\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/jcn.png}
    \caption{\textbf{Illustration of elevated JCN.}}{Elevated junction copy number (JCN=2) arises from the duplication of an allele harboring a DEL-like junction, resulting in a characteristic read depth and junction pattern from which a junction-balanced genome graph can be reconstructed.}
    \label{fig:jcn}
\end{figure*}

\clearpage
\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/jcn_hist.png}
    \caption{\textbf{Pan-cancer junction copy number (JCN) distribution.}}{Colored bars show categories in the cohort. Low JCN, 1 to 3; elevated JCN, >3; high JCN, >7}
    \label{fig:jcn_hist}
\end{figure*}

While the vast majority of junctions demonstrated low-JCN (JCN $\le$ 3), we observed a long tail of junctions with elevatedâ (JCN > 3) and high-JCN (JCN > 7) (Figure \ref{fig:jcn_hist}).

% \begin{figure}
%     %\centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/amplicon.png}
%     \caption{\textbf{Candidate amplicon subgraphs and the feature space construction.}}{Framework to identify features of complex amplicons through the analysis of amplified subgraphs. Each subgraph is annotated for three features: number of elevated (JCN > 3) junctions, maximum JCN relative to the highest vertex CN in the subgraph, and total JCN of all fold-back junctions relative to maximum JCN.}
%     \label{fig:amplicon_features}
% \end{figure}




% TODO: fix the table insertion
% \begin{table}[!ht]
%     \captionsetup{justification=raggedright,singlelinecheck=off}
%     \caption{\textbf{Pan-cancer WGS cohort summary, related to Figure \ref{fig:Fig1}} Summary (table in pdf format) of the number of samples from each tumor type and the number of cell line samples of a tumor type.}
%     \label{tab:S1}
% \end{table}

% \begin{table}[!ht]
%     \captionsetup{justification=raggedright,singlelinecheck=off}
%     \caption{\textbf{Sources of pan-cancer WGS datasets used in \cite{Hadi2020-um}, related to Figure \ref{fig:Fig1}} Summary (table in pdf format) of the number of samples from each dataset, the completeness of the pipelines, and the citation if the dataset is published before.}
%     \label{tab:S2}
% \end{table}

%Consistent with our 10X Chromium WGS benchmarks (see above) (\textbf{Fig. \ref{fig:Fig1}C}), we observed wide variation in inferred JCN across our datasets which correlated with observed read depth changes at breakends (\textbf{Fig. \ref{fig:Fig1}F}). 

% \section{Low JCN clusters of deletion-like and duplication-like junctions form rigma and pyrgo}
% To distinguish between complex SV patterns associated with low-JCN vs. high-JCN junctions, we identified junction clusters based on their overlapping footprints on the reference and labeled each cluster as high- / low- JCN on the basis of its highest copy junction (high-JCN thresholded at $>$ 7). Considering clusters harboring three or more junctions, we found low-JCN clusters were significantly more likely to be dominated ($>$ 90\% representation of that type) by DEL-like ($P < \SI{2.2e-16}{}$, z-test, logistic regression) or DUP-like junctions ($P < \SI{2.2e-16}{}$) (\textbf{Fig. \ref{fig:S2}A}).

% To rigorously nominate clusters of low copy DUP-like and DEL-like junctions within each tumor sample, we identified genomic tiles (1 Mbp width, 500 kbp stride) enriched in low-JCN of a given type (e.g. DEL-like) relative to a calibrated gamma-Poisson background model (see \nameref{sec:methods}). In each analysis, non-outlier tiles harbored apparent simple deletions or duplications (\textbf{Fig. \ref{fig:Fig2}A-B}). In contrast, outlier bins in the DUP-like model resembled "towers" of low-JCN DUP-like junctions  (\textbf{Fig. \ref{fig:Fig2}A, top right}) that we named \textit{pyrgo} (\textgreek{p'urgos}, Greek meaning tower). Outlier loci in the DEL-like models comprised subgraphs of interval CN "chasms" flanked by low-JCN DEL-like junctions whose interval CN often reached 0 (\textbf{Fig. \ref{fig:Fig2}B, top left}).  We named these patterns \textit{rigma} (ÏÎ®Î³Î¼Î±, Greek meaning rift).

\section{Tyfonas is massively rearranged amplicon associated with elevated number of fusion genes}
We then sought to investigate the rearrangement patterns associated with high-JCN junctions (JCN $>$ 7) in our genome graphs.  \textit{A priori}, a junction at such an extreme of JCN may evolve through a double minute, BFB cycles, or an as yet undescribed mechanism for duplicating already rearranged DNA. To characterize classes of  amplification events associated with these high-JCN junctions, we first identified 12,588 subgraphs harboring an interval CN of at least twice ploidy among the 2,487 unique genome graphs (by patient) (Figure \ref{fig:amplicon_features}), then identified among these amplicons (amplified clusters within a genome) those that harbor at least one junction with JCN $>$ 7.  We annotated the resulting 1,703 high-JCN amplicons according to three features: 1) the maximum JCN normalized by the maximal interval CN, 2) the \textit{sum} of fold back inversion JCNs (INV-like junctions that terminate and begin at nearly the same location in the genome) relative to the maximal interval CN, and 3) the number of junctions with elevated JCN (JCN>3) (see Appendix \ref{app:a}).

\clearpage
\begin{figure}[!ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{Candidate amplicon subgraphs and the feature space construction.}
        \label{fig:amplicon_features}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/amplicon.png}
    \end{adjustbox}
\end{figure}


\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/amplicon_clustering.png}
    \label{fig:amplicon_definition}
    \caption{\textbf{Classification of amplicons with high-JCN junctions.}}
\end{figure*}

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/amplicon_clustering_jaccard.png}
    \label{fig:amplicon_jaccard}
    \caption{\textbf{Assessment of stability of amplicon clustering.}}{For a given setting of $k$, cluster stability was computed as the mean Jaccard similarity between each observed cluster and the most similar cluster across 100 75\% bootstraps of the data (see Appendix \ref{app:b}). The stability of each $k$ parameter setting was computed as the mean of the $k$ stability scores across the clusters generated at that setting.}
\end{figure*}

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/amplicon_pca.png}
    \label{fig:amplicon_pca}
    \caption{\textbf{Projection of the amplicons across the first two principal components of normalized amplicon features.}}{}
\end{figure*}
\clearpage

Clustering and classification of amplicons (Figure \ref{fig:amplicon_definition}) on the basis of these three features yielded three stable clusters (Figure \ref{fig:amplicon_definition},\ref{fig:amplicon_jaccard},\ref{fig:amplicon_pca}; see also Appendix \ref{app:b}).  Upon visual inspection, the first group, harboring low fold-back JCN but high maximal JCN, contained amplicons comprising a single high-JCN junction forming a high CN circular path in the graph (Figure \ref{fig:dm_example}) as well as more complex cyclic patterns spanning multiple discontiguous loci, consistent with a double minute (or extrachromosomal circular DNAs).  The second group, demonstrating high fold-back JCN ($>$ 0.5), a low burden of elevated-JCN junctions ($<$ 26), and a "stair step" pattern of copy gains, was consistent with a BFB cycles (Figure \ref{fig:bfb_example}, \cite{Zakov:2013cm,McClintock1939-oi}). The third group contained both high fold-back JCN ($\geq$0.50) and a significant burden of elevated JCN junctions ($\geq$26).  Upon visual inspection, these amplicons comprised dense webs of elevated  JCN junctions across subgraphs consisting of $>$ 100 Mbp of genomic material and often reaching CNs higher than 50 (Figure \ref{fig:tyf_example}).  We dubbed these extremely large amplicons, which did not fit in previously defined categories, \textit{tyfonas} (ÏÏÏÏÎ½Î±Ï, Greek meaning typhoon) (Figure \ref{fig:tyf_example}).

\clearpage
\begin{figure*}[!h]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/bfb_example.png}
    \label{fig:bfb_example}
    \caption{\textbf{An example of a BFB cycles pattern.}}
\end{figure*}

\begin{figure*}[!h]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/dm_example.png}
    \label{fig:dm_example}
    \caption{\textbf{An example of a double minute pattern.}}
\end{figure*}

\begin{figure}[!ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{An example of a tyfonas pattern.}
        % {Demonstrating example loci for 5 simple and 8 complex SV patterns.}
        \label{fig:tyf_example}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/tyf_example.png}
    \end{adjustbox}
\end{figure}
\clearpage

\subsection*{Tyfonas are distinct from BFB cycles and double minutes}
Comparing amplicon features, we found that tyfonas had significantly larger genomic mass (summed interval width weighted by CN, Figure \ref{fig:amplicon_mass}), maximum interval CN (Figure \ref{fig:amplicon_cn}), junction burden (Figure \ref{fig:amplicon_njunc}), and JCN entropy (based on the distribution of JCN in the amplicon, the higher the more diversity among the JCNs, Figure \ref{fig:amplicon_jcn_entropy}) than either double minutes or BFB cycles ($P$ < \SI{2.2e-16}, Wilcoxon rank-sum test).  All three amplicon classes frequently intersected CGC cancer genes residing within GISTIC pan-cancer amplification peaks (Figure \ref{fig:amplicon_driver}, recurrent amplification peaks from \cite{Zack:2013f1f}). \textit{EGFR} was the most frequent target of double minutes.  BFB cycles were most frequently implicated in \textit{ERBB2}, \textit{CCND1}, and \textit{CCNE1} amplification. Tyfonas were associated with \textit{MDM2} and \textit{CDK4}. While double minutes were enriched in glioblastoma and small cell lung cancer, BFB cycles were enriched in esophageal squamous cell cancer, lung squamous cell cancer, and head and neck squamous cell cancer (FDR $<$ 0.25, Fisher's exact test, Figure \ref{fig:amplicon_ttype}). In contrast, tyfonas events were enriched in sarcoma, breast cancer, and melanoma (FDR $<$ 0.25).

% \clearpage

% \clearpage

% The specificity of tyfonas for these tumor types was much higher than for liposarcoma-like or amplified chromothripsis, which are recently proposed PCAWG categories that also showed substantial overlap with BFB cycles and double minutes (\textbf{Fig. \ref{fig:S4}E-K}) \cite{pcawg_marker2020-yi}.

Analyzing tumor subtypes, we found tyfonas were common in dedifferentiated liposarcomas (80\%) and acral melanomas (40\%) but rarely seen ($<$ 2\%) in fibrosarcomas and cutaneous melanomas (Figure \ref{fig:tyf_ttype}).    80\% of dedifferentiated liposarcomas harbored a tyfonas on chromosome 12q, suggesting that tyfonas represent the genomic footprint of supernumerary ring chromosomes in this tumor type \cite{Reimann2008-tj}. Given the enrichment of tyfonas events in acral melanoma, an immunotherapy-responsive melanoma with a low SNV burden \cite{Shoushtari2016-de}, we hypothesized that tyfonas events may provide an alternate source of neoantigens through the generation of expressed protein-coding fusions.  Analyzing the subset of our samples with RNA-seq data within the TCGA dataset, we found that tyfonas were significantly enriched with expressed fusion transcripts relative to double minutes, BFB cycles, and even chromothripsis (CT) (Figure \ref{fig:tyf_fusion}, left), even after accounting for the larger genomic footprint of tyfonas (Figure \ref{fig:tyf_fusion}, right).

\clearpage
\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/amplicon_cn.png}
    \label{fig:amplicon_cn}
    \caption{\textbf{Tyfonas amplify parts of genome to higher CN states than BFB and DM.}}
\end{figure*}

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/amplicon_mass.png}
    \label{fig:amplicon_mass}
    \caption{\textbf{Tyfonas result in larger mass of amplified genomic DNA than BFB and DM.}}
\end{figure*}

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/amplicon_jcn_entropy.png}
    \label{fig:amplicon_jcn_entropy}
    \caption{\textbf{Tyfonas contain more heterogeneous junction copy numbers than BFB and DM.}}
\end{figure*}

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/amplicon_njunc.png}
    \label{fig:amplicon_njunc}
    \caption{\textbf{Tyfonas contain more junctions than BFB and DM.}}
\end{figure*}

\begin{figure}[!ht]
    \centering
    \includegraphics[height=0.95\textheight]{figures/600ppi/amplicon_driver.png}
    \caption{\textbf{Tyfonas, DM, and BFB recurrently amplify different oncogenes}}
    \label{fig:amplicon_driver}
\end{figure}

\begin{figure}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/amps_ttype.png}
    \caption{\textbf{Tyfonas, DM, and BFB enrich in different tumor types.}}
    \label{fig:amplicon_ttype}
\end{figure}


\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tyf_ttype.png}
    \label{fig:tyf_ttype}
    \caption{\textbf{Tyfonas is highly enriched in liposarcoma and acral melanoma.}}
\end{figure*}

\begin{figure*}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tyf_fusion.png}
    \label{fig:tyf_fusion}
    \caption{\textbf{Tyfonas create more fusion genes and highly expressed fusion genes than BFB, DM, and chromothripsis.}}{Enrichment of expressed protein-coding fusion transcripts by count (left) and density per Mbp of event territory (right) in tyfonas relative to other amplicon types and CT (Wilcoxon rank-sum test). Error bars on bar plots represent 95\% confidence intervals on the Bernoulli trial parameter.}
\end{figure*}

\clearpage


\section{Clusters of pan-cancer patients based on SV event burdens show differential prognosis and is associated with tumor types and genetic backgrounds}



% \begin{figure}
    % \centering
    % \includegraphics[width=0.95\linewidth]{figures/600ppi/sv_gallery.png}
    % \caption{\textbf{Dictionary of genome graph-derived event patterns}}{Demonstrating example loci for 5 simple and 8 complex SV patterns.}
    % \label{fig:sv_gallery}
% \end{figure}


Tallying normalized junction counts across 13 event categories and 2,487 patients, we found 14 stable clusters using standard model selection metrics (Figure \ref{fig:find_k}). Most clusters were dominated by 1-3 event types (e.g. CT = chromothripsis, BR = BFB cycles, rigma, DDT = deletion, duplication, TIC) with the exception of two: QUIET (few events) and SPRS (sparse, miscellaneous events).

% \clearpage
\begin{figure}[!ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{Dictionary of genome graph-derived event patterns}
        % {Demonstrating example loci for 5 simple and 8 complex SV patterns.}
        \label{fig:sv_gallery}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/sv_gallery.png}
    \end{adjustbox}
\end{figure}

\begin{figure}[!ht]
    \centering
    \includegraphics[width=\linewidth]{figures/600ppi/clustering_heat.png}
    \caption{\textbf{Genome graph-derived features define biologically distinct patient groups}}
    \label{fig:clustering_heat}
\end{figure}


\begin{figure}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/clustering_find_k.png}
    \caption{\textbf{Metrics for determining the optimal number of clusters.}}
    \label{fig:find_k}
\end{figure}
% \clearpage

% \begin{figure}[!ht]
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/clustering_heat.png}
%     \caption{\textbf{Genome graph-derived features define biologically distinct patient groups}}
%     \label{fig:clustering_heat}
% \end{figure}


% {Heatmap of normalized junction burden across 2,487 unique patients yields 14 clusters named after their most prevalent event types (CT, TYF [tyfonas], BFB [BFBC], BR [BFBC, rigma], SPRS [sparse], CP [chromoplexy], TIC, TRA [translocations], DM [double minute], PYR [pyrgo], DUP [various duplications], DDT [deletion, duplication, and TIC], and INVD [inverted duplications]).}



% \begin{figure}[!ht]
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/clustering_tt_all.png}
%     \caption{\textbf{All associations between clusters and tumor types.}}
%     \label{fig:clustering_tt_all}
% \end{figure}


Consistent with previous reports, the CT cluster was significantly enriched in prostate adenocarcinoma  (PRAD, $P = \SI{2.05e-5}{}$, $OR = 1.99$, single-sided z-test, Bayesian logistic regression, \cite{Kovtun2015-kq}) and glioblastoma multiforme (GBM, $P = \SI{5.00e-8}{}$, $OR = 2.61$, \cite{Furgason2015-rv}, Figure \ref{fig:clustering_tt}, \ref{fig:clustering_tt_all}). Similarly, the CP (chromoplexy) cluster was significantly enriched in PRAD ($P = \SI{2.32e-10}{}$, $OR = 3.18$, \cite{baca2013}). DDT tumors (defined by high burdens of deletions, duplications, and templated insertion chains) were enriched in triple-negative breast cancer (TNBC) ($P < \SI{2.2e-16}{}$, $OR = 8.80$),  ovarian cancers ($P = \SI{7.03e-16}{}$, $OR = 6.89$), and more broadly sex-hormone driven tumors ($P = \SI{3.18e-14}{}$, $OR = 19.0$).

\clearpage
\begin{figure}[!ht]
    \centering
    \includegraphics[width=\linewidth]{figures/600ppi/clustering_tt.png}
    \caption{\textbf{Selected associations between clusters and tumor types.}}
    \label{fig:clustering_tt}
\end{figure}


\begin{figure}[ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{All associations between clusters and tumor types.}
        \label{fig:clustering_tt_all}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/clustering_tt_all.png}
    \end{adjustbox}
\end{figure}
\clearpage

Inspection of the heatmap in Figure \ref{fig:clustering_heat} showed that the classes of complex SV introduced in this study (pyrgo, rigma, tyfonas) largely clustered independently from known complex SV types (double minute, BFB cycles, chromothripsis, chromoplexy). Among these, the BR (BFB cycles and Rigma dominated) cluster was primarily (60\%) composed of ESAD cases ($P < \SI{2.2e-16}{}$, $OR = 6.08$) and enriched in gastrointestinal tumors (e.g. esophageal, colorectal, and gastric adenocarcinoma) ($P < \SI{2.2e-16}{}$, $OR = 4.56$) (Figure \ref{fig:clustering_heat},\ref{fig:clustering_tt_all}). The TYF (tyfonas dominated) cluster was enriched in both luminal breast cancer ($P = \SI{4.87e-8}{}$, $OR = 3.25$),  HER2+ breast cancer ($P < \SI{2.64e-9}{}$, $OR = 4.96$), dedifferentiated liposarcoma ($P < \SI{2.2e-16}{}$, $OR = 24.5$), and acral melanoma ($P < \SI{1.84e-15}{}$, $OR = 7.40$). In contrast, cutaneous melanomas were enriched in the CT cluster.  Additional associations are shown in Figure \ref{fig:clustering_tt_all} and \cite{Hadi2020-um} Table S3.

\begin{figure}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/clustering_geno.png}
    \caption{\textbf{DDT cluster is associated with BRCA1 familial LoF alterations and BR cluster with \textit{TP53} LoF.}}
    \label{fig:clustering_geno}
\end{figure}

\begin{figure}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/clustering_geno_all.png}
    \caption{\textbf{All associations between clusters and LoF alterations of tumor suppressors or DNA damage response genes.}}
    \label{fig:clustering_geno_all}
\end{figure}

We associated somatic or constitutional genotypes in COSMIC Cancer Gene Census genes with cluster membership, after correcting for tumor subtype as a covariate. More than 20\% of cases in the DDT cluster harbored constitutional ($P = \SI{1.60e-4}{}$, $OR = 3.81$) loss of function lesions in \textit{BRCA1} (Figure \ref{fig:clustering_geno}). BR-cluster tumors were also significantly enriched in somatic \textit{TP53} mutations ($P < \SI{4.71e-7}{}$, $OR = 2.06$). Additional somatic genotype associations (Figure \ref{fig:clustering_geno_all}), including an enrichment of \textit{SMC4} ($P = \SI{2.33e-3}{}$, $OR = 3.11$) and \textit{RAD21} ($P = \SI{1.96e-3}{}$, $OR = 3.27$) mutations in the INVD (inverted duplication dominant) cluster, \textit{ARID1A} ($P = \SI{1.21e-3}{}$, $OR = 1.86$) mutations in the TIC (templated insertion chain dominant) cluster, and \textit{KMT2C} ($P = \SI{2.89e-3}{}$, $OR = 1.95$) mutations in the TRA (translocation-dominated) cluster.

Compared to QUIET cluster, Kaplan-Meier analysis revealed poorer survival among novel SV-class dominated clusters (BR, PYR, and TYF; FDR $<$ 0.1, log rank test) as well as several clusters dominated by previously-described SV classes (CP, CT, and INVD; Figure \ref{fig:clustering_km}). These effects persisted after correcting for clinical and molecular covariates in a Cox regression analysis, with BR ($P = \SI{1.17e-2}{}$, $HR = 1.72$, likelihood ratio test, Cox regression), PYR ($P = \SI{6.13e-3}{}$, $HR = 2.01$), TYF ($P = \SI{5.37e-3}{}$, $HR = 2.12$ , CP ($P = \SI{1.76e-3}{}; HR = 1.91$), CT ($P = \SI{6.69e-4}{}; HR = 1.83$), and INVD ($P = \SI{8.79e-4}{}; HR = 2.06$) clusters each demonstrating reduced survival relative to the QUIET cluster (Figure \ref{fig:clustering_os_cox}).


\clearpage
\begin{figure}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/clustering_km.png}
    \caption{\textbf{Overall survival worse in 6 clusters than the QUIET.}}
    \label{fig:clustering_km}
\end{figure}

\begin{figure}[!ht]
    \centering
    \includegraphics[height=0.9\textheight]{figures/600ppi/clustering_os_full.png}
    \caption{\textbf{Cox model correcting for covariates show the associations between TYF, INVD, PYR, CP, CT, BR and worse overall survival after correction of related covariates.}}
    \label{fig:clustering_os_cox}
\end{figure}
\clearpage

\section{Discussion}
Leveraging JaBbA reconstructed genome graphs from 2778 cancer WGS datasets, we define 13 types of simple and complex classes of structural variants including pyrgo, rigma \cite{Hadi2020-um}, and tyfonas (see Chapter \ref{chap:complex_events}) that have not been described rigorously classified before. These three new event classes all have distinct associations with regional, genotypic, and tumor types from previously known complex patterns like chromothripsis or BFB cycles. They also form clusters differently within the patient cohorts. All of these suggest that they result from novel mutational processes and/or somatic selection. In this chapter, I present two important parts of our discoveries, the taxonomy of complex amplicon and patient clustering based on SV event burdens.

Among amplified subgraphs with high-JCN junctions, tyfonas spontaneously emerge distinct from previously known complex amplification patterns double minutes and BFB cycles cycles by higher total fold-back inversion JCNs and higher elevated JCN counts. Tyfonas further differentiated from DM and BFB cycles in its massive scale, excessive junctions, enrichment in liposarcoma and acral melanoma, and recurrent amplification of \textit{MDM2}-\textit{CDK4} locus.

What is the mechanism that created tyfonas? We observed tyfonas are present in about 80\% of liposarcoma, which coincides with the high frequency of supernumerary rings in that tumor type, indicating they may be orthogonal readouts of the same structure. Furthermore, Garsed et al. \cite{garsed2014} proposed \textit{neo-chromosome} structures in five liposarcoma cell lines, which was suggested to arise from circular BFB with centromere eventually regaining telomeres to be stabilized linearly. Based on the similarity of amplification patterns presented in that study, extensive complex junctions, and the recurrent localization at \textit{MDM2}-\textit{CDK4} locus same as tyfonas, we conjecture that tyfonas is reflecting the same process. In fact, using fluorescent \textit{in situ} hybridization (FISH) with NCI-H526 small cell lung cancer cell line which possesses a tyfonas event covering both \textit{MYCN} and \textit{MYCL} genes, we found this particular tyfonas is integrated into two chromosomes near termini. This observation confirms the localization of the amplicon in linear form, that is likely neo-chromosomes.

However, our definition of tyfonas differ from the model proposed in \cite{garsed2014}, partially due to the abundance of fold-back inversions captured in tyfonas. Garsed et al. were propelled to construct the circular BFB theory due to the lack of "inverted duplication" signature of canonical linear form BFB. Yet, we observed many fold-back inversions, and it is very likely that these transient structures gain fold-back inversions in linear forms where the blunt ends can be fused to its newly synthesized 'sister chromatid'. Furthermore, our definition of tyfonas arise spontaneously from analyzing thousands of pan-cancer genome graphs, which indicates such events are not limited to liposarcoma alone, but widely observed in other tumor types like acral melanoma, breast carcinoma, small cell  and non-small cell lung cancers. 

In theory, there are at least two frequent mechanisms that allow genomic DNA to amplify exponentially, asymmetric inheritance and fold-back on sister chromatid. Double minute and canonical BFB cycles are the prototypical events, respectively. In the former, doubling is conferred when DNA unbounded by centromeres inequally distribute into two daughter cells. Theoretical models of double minute CN dynamics verifies a >1 selection coefficient can quickly result in high mean CN among the cell population \cite{Lange2021-if}. On the other hand, in a canonical BFB cycle, unprotected chromosome ends fuse and are broken asymmetrically into daughter cells. If new ends stay unprotected, the process can repeat, resulting in focal high-level amplification in the lineage that elongated the arm. We postulate that tyfonas might have taken a combination form of the two. 

Different from chromothripsis, initial shattering does not get repaired quickly enough to form scrambled new sequence, rather, the linear fragments are preserved in the nucleus through an extended period of time, for example, in structures like micronuclei. They exist in either linear or circular forms and experience continuous instability which can further 'hijack' distal loci in the genome that was not in the initially shatttered regions, resulting in numerous amplified loci.

Interestingly, tyfonas also have more heterogeneous JCN composition, supporting the asynchronous genesis of junctions during the ongoing amplification process. This is fundamentally different from an 'amplified chromothripsis', where the junctions are supposedly created almost at the same time and amplified together later, which would lead to rather homogeneous elevated JCN within the final amplicon. The events arising from PCAWG's analysis of clustered variants termed 'amplified chromothripsis' \cite{pcawg_marker2020-yi} actually map to multiple classes under our taxonomy (see \cite{Hadi2020-um}, Figure S4G-K). This improvement of specificity is also visible in the tumor type enrichments for tyfonas (see \cite{Hadi2020-um}, Figure S4F).

Despite the proposition of a new and likely impactful amplicon pattern, considerable amplicon candidates still remain unclassified. Some of them are excluded from our current analysis because they do not contain high-JCN junctions (JCN>7). While certain pyrgo instances can achieve this through only 'stacking up' low-JCN tandem duplications, more candidates are weaved by various types of junctions of low to moderate JCN (JCN $\le$ 7), both with and without fold-back inversions (see Chapter \ref{chap:tel_crisis}, Figure \ref{fig:tc_bft3b_complex}). We also notice certain double minutes with relatively lower CN than classic examples like \textit{EGFR} amplification in glioblastoma, whose maximum JCN barely passed the current threshold (see Chapter \ref{chap:gdan}, Figure \ref{fig:gdan_dm}), yet the homogeneous JCN composition indicates extrachromosomally amplified more than other hypotheses. It is not hard to conjecture then, that there could be other eccDNAs created by the same mechanism as high-level double minutes, yet for various reasons do not get amplified to such extraordinary levels \cite{Lange2021-if}.

Even within the high-JCN candidates in our amplicon clustering analysis, there is a fourth cluster with high maximum JCN like double minutes, low fold-back inversion JCN contrary to BFB cycles and tyfonas, yet extensively rearranged with high elevated JCN junction count. The cluster is small and not as stable hence named other and left out of the discussion of this paper \cite{Hadi2020-um}. Because of the simultaneous presence of high-JCN DUP-like junctions and abundant lower JCN junctions within the amplified regions, we hypothesize this subset may be reflecting unstable double minutes that keep on accumulating new junctions extrachromosomally. Recently, Shoshani et al. published a study showing cells with eccDNA carrying the  further rearranged their amplicon structures in response to continuous selective pressure. The structure and mechanism shown in the study resembles what we observed and is of particular interest to follow up, especially in post-treatment metastatic samples \cite{Priestley:20196a6}.

Looking at the big picture, the burden of our 13 simple to complex SV classes stratify pan-cancer patients into clear clusters, indicating once again the links to different mutagenesis pathways active during the formation of different tumor samples. It also shows the potential of a genome graph-based SV taxonomy to derive variant count matrix and extract SV signatures eventually. Poorer prognosis in several complex SV infested clusters compared to the one largely free of SVs, reaffirms the understanding that complex SVs can create more aggressive phenotypes. As more studies start to focus on chromosomal instability as a therapeutic target \cite{Bielski2021-ue}, we believe characterization of SV patterns, signatures, and mutagenesis will be key to the clinical application of WGS.

% Though tyfonas likely represent the footprint of supernumerary rings in dedifferentiated liposarcomas, their prevalence in tumor types (e.g. acral melanomas, breast cancers) not commonly associated with supernumerary rings suggest that the cytogenetic footprint of this event may be more diverse. If our Hi-C, optical mapping, and FISH-based model of tyfonas (\textbf{Fig. \ref{fig:Figure6}E}) generalizes, a key question is how these chromosomally integrated loci acquire such a wide breadth of JCN in \textit{cis}. We speculate that this allelic configuration may require a transient extrachromosomal phase, enabling the progressive self-assembly of many linear fragments ligating and duplicating in parallel across successive cell cycles.  These asymmetrically segregating shards may preferentially ligate with their sister shard in end-to-end orientation after each S phase, which may give rise to the tyfonas-associated fold back inversion pattern.

% The enrichment of breakend hypermutation across multiple non-APOBEC associated SNV contexts in tyfonas suggests that these junctions may arise in a specific mutagenic cellular environment.  Though we cannot rule out the role of APOBEC enzymes in generating these SNV patterns (e.g. within a non-canonical context), they may also arise through error prone Pol-$\alpha$ fill-in or alt-NHEJ repair, both of which are active at unprotected DNA ends and generate mutations across a wide spectrum of trinucleotide contexts \cite{sfeir2012,Mirman:2018jm}.  We anticipate that future investigations in model systems will resolve some of these exciting mechanistic questions regarding tyfonas genesis.

% Our comprehensive analysis of 2,778 genome graphs establishes JCN topology as a key signal for the classification of complex SVs in human cancer. The regional, genotypic, and tumor type distribution of SV classes nominated in this study (pyrgo, rigma, tyfonas) are distinct from previously identified complex rearrangement patterns (chromothripsis, chromoplexy, BFB cycles), suggesting that they represent novel mutational processes and/or foci of somatic selection.  We anticipate that these findings, combined with JaBbA's superior performance relative to previous CNA algorithms, will bring the genome graph paradigm to the mainstream of cancer genome analysis.

% Our study provides a template for integrating short-read WGS with long-range genome profiling to study patterns of somatic junction phase and allelic multiplicity in complex loci. Future studies that utilize long-range whole genome profiles (10X Chromium WGS, Pacific Biosciences, Oxford Nanopore Technologies, Hi-C, Bionano \mbox{\cite{Sedlazeck2018-ix}}) to phase highly rearranged somatic alleles may provide a more granular classification and mechanistic insight into complex SV mutational processes.  Larger short-read meta-analyses spanning >10,000 cancer genome graphs, which may be imminently possible through the combination of our approach and dataset with PCAWG (1,493 additional tumors) \cite{pcawg_marker2020-yi}, Hartwig Medical Foundation (4000+ cases, \cite{Priestley:20196a6}), and emerging WGS precision medicine efforts will help link novel features of genome-graph topology with clinical variables, including therapeutic response. This includes future investigations linking the presence of tyfonas to the efficacy of immune checkpoint inhibition in acral melanoma or other tumor types in which these events are found (e.g. small cell lung cancer).




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Chapter 4: SV evolution after telomere crisis
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter{Structural variant evolution after telomere crisis *} \label{chap:tel_crisis} 
In this chapter I set out to capture the SV events in human cell lines directly resulting from natural telomere crisis. This chapter is based on our published article \cite{Dewhurst2021-jk}*.

\textit{\textbf{Individual contributions:} Sally Dewhurst, Titia de Lange, Marcin Imielinski conceptualized and initiated the project. Sally Dewhurst collected the previous published cell lines spontaneously escaped telomere crisis. Sally Dewhurst designed and created the \textit{in vitro} telomere crisis model, executed all the experiments, and analyzed the data except for the WGS parts. I carried out all the analysis involving whole genome sequencing data. Marcin Imielinski and I designed multi-sample algorithms for inferring evolving karyotypes. Marcin Imielinski and I led the effort in reconstructing the evolutionary trajectory of the clones.}

\footnotetext[1]{Dewhurst, S. M., Yao, X., et al. Structural variant evolution after telomere crisis. Nat. Commun. 12, 2093 (2021)}

\section{Introduction} \label{chap:intro}
Telomeres are the protective sequences at human chromosome ends made up with hexanucleotide repeats 5'-(TTAGGG)-3' \cite{De_Lange1990-mz}. In healthy cells, telomeres are bound by shelterin complex, and prevent the natural chromosome ends from being recognized as double strand breaks by various DNA damage response (DDR) pathways \cite{De_Lange2005-cn}. Shelterin binds t-loop structures formed by single stranded 3' overhang at the terminus invading back into upstream double stranded telomere repeats. At every mitosis, the 3' overhang needs to be recreated, and at the same time DNA polymerase is unable to fully replicate the telomeres, hence telomeres shortens naturally by about 50bps per cell division \cite{De_Lange2005-cn}. 

In healthy cells, this replicative telomere attrition creates a barrier to their proliferation. When the telomeres are below a certain critical lengths that shelterin complex cannot no longer form a proper protective structure, the cells will undergo a period of time termed \textit{telomere crisis}, during which chromosome ends erroneously trigger DNA repair pathways, fusing telomere to telomere, create dicentric chromosomes \cite{Artandi2000-ab,Maciejowski2017-kx}, and eventually lead to senescence or apoptosis. Cancer cells on the other hand, need to overcome telomere crisis to become immortalized, and this is thought to occur at an early stage of cancer development \cite{Shay2005-fb} most commonly through reactivating telomerase, the reverse transcriptase responsible for synthesizing telomere sequences. Thus, the genomic sequences post-crisis cell reflect the alterations introduced during this period and it gives us a chance to deduce what has happened.

What exactly are the consequences of dividing cells when dicentric chromosomes are present? Many events are known with the earliest dated back decades, when McClintock \cite{McClintock1939-oi} defined breakage-fusion-bridge (BFB) cycles for the broken chromosome fused to sister chromatid during maize meiosis. Gisselsson et al. \cite{Gisselsson2001-hq} used cytogenetics and constructed breakpoint profiles for 102 pancreatic carcinoma an 140 osteosarcoma samples, to find some BFBs among extensively variable breakpoint patterns. In other cancer types, the iconic fold-back inversion junctions are found among complex chromothripsis-like patterns in acute lymphocytic leukemia (ALL) \cite{Li2014-mj}, triggering the hypothesis these could start with BFB by dicentric chromosomes and are followed by chromothripsis before finally stabilized. Modeling of telomere crisis in late generation telomerase-deficient mice lacking p53 showed that telomere dysfunction engenders cancers with non-reciprocal translocations, as well as focal amplifications and deletions in regions relevant to human cancers \cite{Artandi2000-ab,OHagan2002-rg}. Furthermore, mouse models of telomerase reactivation after a period of telomere dysfunction showed that acquisition of specific copy number aberrations and aneuploidy could drive malignant phenotypes \cite{Ding2012-at}.

Lately, high-throughput genome sequencing of cultured human cells started to paint a more detailed picture. Liddiard et al. 2016 \cite{Liddiard2016-ay} examined intra- and inter-chromosomal telomere-to-telomere fusions, and found that a single artificially deprotected telomere can fuse with multiple intra- and inter-chromosomal loci leading to complex fusion products. By expressing a dominant negative allele of \textit{TRF2}, a critical component of shelterin, multiple studies have shown that the resolution of the resulting dicentric chromosomes can lead to dramatic chromosome shattering known as chromothripsis \cite{maciejowski2015,Umbreit2020-kr,Maciejowski2020-bw}. However, all the studies mentioned above either used systems lacking crucial genome maintenance components (e.g. \textit{TRF2}, \textit{LIG3/4}), or are confounded by the selection process happened naturally (in cancer) or during culture.

The only direct observation of SV consequences in relatively healthy cells after a sustained period of telomere crisis did not capture major complex events implicated before \cite{Cleal2019-dk}. This is probably due to the fact that the HCT116 colon carcinoma cells used in this experiment have already expressed \textit{TERT} and are able to readily escape from telomere crisis. Again, in the same study, there are various complex SV events reported for engineered HCT116 cells deficient in non-homologous end joining (NHEJ), but the results cannot be simply attributed to telomere crisis.

Given the expanding repertoire of structural variant present in so many cancer types, and the potential contribution of telomere dysfunction to some of these aberrations, we set out to characterize the extent and type of structural variant that can be unleashed by telomere crisis and subsequent genome stabilization by telomerase expression. We tackle this problem from two aspects. First, we conducted WGS on nine isolated cell lines previously found to survive telomere crisis by spontaneous expression of telomerase, both in the post-crisis cells and in their pre-crisis ancestors as control. If we observe SV events specific to the post-crisis cell lines and not pre-crisis, those can be largely attributed to telomere crisis.

Second, considering the complications during spontaneous \textit{TERT} activation, we set out to design a system known to experience natural replicative telomere attrition, activates \textit{TERT} on cue, and allow us to harvest cells precisely after they escaped telomere crisis. We use low-pass (low read depth) WGS to screen for CNAs and further subject representative clones taking different CNA configurations to high-pass WGS to reconstruct the exact rearranged haplotypes after telomere crisis by genome graph analysis. This will enable us to infer phylogeny among diverging clones, deduce the sequence of events triggered by crisis, and gain much deeper insight into how they start, progress, and conclude. In short, through both approaches, we observe a wide spectrum of SV events, from totally quiet genomes to simple arm-level gain and loss to complex chromothripsis or BFB patterns. Hence, telomere crisis can instigate varying SV events, and using BFB and chromothripsis alone cannot accurately predict the presence of telomere crisis in the life history of a cancer.


\section{Genomic complexity after spontaneous telomerase activation}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_prev_circos.png}
    \caption{\textbf{Post-crisis SV landscape in 8 spontaneously derived cell lines.}}
    \label{fig:tc_prev_circos}
\end{figure}
% {Binned purity- and ploidy-transformed read depth is shown in the periphery, with colored links in the center representing variant (rearrangement) junctions. A series of red colors is used to show junctions and read-depth bins belonging to distinct clusters of complex gains in each cell line. Additional colors describe junctions and bins, including those belonging to simple losses and gains.}


% \begin{figure}
%     \centering 
%     \includegraphics[width=\linewidth]{figures/600ppi/tc_prev_all_circos.png}
%     \caption{\textbf{Post-crisis SV landscape with pre-crisis whole genome coverage in 9 spontaneously derived cell lines.}}
%     \label{fig:tc_prev_all_circos}
% \end{figure}
% {From innermost circle, variant junctions present only in post-crisis sample, colored by complex amplicon clusters or grey if not classified into any, JaBbA inferred copy number, purity-ploidy transformed binned coverage of the pre-crisis clone, of the post-crisis clone, and chromosome bands. SW13 is marked with an asterisk due to the presence of extensive pre-crisis structural rearrangements.}

\begin{figure}
    \centering
    \includegraphics[height=0.95\textheight]{figures/600ppi/tc_prev_tert.png}
    \caption{\textbf{SVs around \textit{TERT} locus in spontaneous post-crisis cell lines.}}
    \label{fig:tc_prev_tert}
\end{figure}
% {Each track shows binned purity- and ploidy-transformed read depth in units of CN, with variant (rearrangement) junctions and loose ends plotted as red arcs. The bottom track highlights the TERT gene among genes on chromosome 5p15. Violin plots to the left of tracks show the genome-wide distribution of read depth in units of CN, demonstrating that 6 of 7 clones have elevated CN at the TERT locus.}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_fig1b_ct_sw26.png}
    \caption{\textbf{Chromothripsis event in SW26 cell line.}}
    \label{fig:tc_ct_sw26}
\end{figure}

\begin{figure*}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_bft3k_bfb.png}
    \caption{\textbf{Primitive BFB event in BFT3K cell line.}}
    \label{fig:tc_bft3k_bfb}
\end{figure*}

\begin{figure*}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_bft3b_complex.png}
    \caption{\textbf{Complex amplification event in BFT3K cell line.}}
    \label{fig:tc_bft3b_complex}
\end{figure*}

In order to determine the SVs in post-telomere crisis genomes, we examined nine SV40 large T-transformed cell lines that had undergone spontaneous telomerase activation after passage into telomere crisis (Figure \ref{fig:tc_prev_circos}). In Figure \ref{fig:tc_prev_circos}, binned purity- and ploidy-transformed read depth is shown in the periphery, with colored links in the center representing variant (rearrangement) junctions. A series of red colors is used to show junctions and read-depth bins belonging to distinct clusters of complex gains in each cell line. Additional colors describe junctions and bins, including those belonging to simple losses and gains. 
% In Figure \ref{fig:tc_prev_all_circos}, purity-ploidy transformed binned coverage of the pre-crisis clone is also shown as an extra track.

\clearpage
\begin{table}[ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}\caption{
        \textbf{SV40T immortalized pre-crisis and post-crisis telomerase positive cell lines.}
        \label{tab:tc_sv40t}
        }}{\end{minipage}},rotate=90,center}
        \begin{tabular}{ccccc}
            \hline
            \textbf{Name}                  & \textbf{Cell Line/Tissue Source}              & \textbf{Telomerase Status}   & \textbf{Reference}                 & \textbf{Kind gift of}             \\ \hline
            HA1 p15               & Human Embryonic   Kidney cells       & -            & \cite{Counter1992-yg}  & Silvia   Bacchetti/AdVec \\
            SW13 PD73             & IMR90 lung   fibroblasts             & -            & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
            SW26 PD68.5           & IMR90 lung   fibroblasts             & -            & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
            SW39 PD72.5           & IMR90 lung   fibroblasts             & -            & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
            Bet3B p3              & NHBE-10 bronchial   epithelial cells & -            & \cite{Bryan1995-ik}     & Roger Reddel, CMRI       \\
            Bet3K p7              & NHBE-10 bronchial   epithelial cells & -            & \cite{Bryan1995-ik}  & Roger Reddel, CMRI       \\
            BFT3B p10             & BF-10 bronchial   fibroblasts        & -            & \cite{Bryan1995-ik}    & Roger Reddel, CMRI       \\
            BFT3G p7              & BF-10 bronchial   fibroblasts        & -            & \cite{Bryan1995-ik}  & Roger Reddel, CMRI       \\
            BFT3K p9              & BF-10 bronchial   fibroblasts        & -            & \cite{Bryan1995-ik}    & Roger Reddel, CMRI       \\
            HA1-IM PD216          & Human Embryonic   Kidney cells       & + & \cite{Counter1992-yg}  & Silvia   Bacchetti/AdVec \\
            SW13 PD184            & IMR90 lung   fibroblasts             & +  & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
            SW26 PD130+           & IMR90 lung   fibroblasts             & +  & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
            SW39 PD130+           & IMR90 lung   fibroblasts             & +  & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
            Bet3B p25 post-crisis & NHBE-10 bronchial   epithelial cells & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\
            Bet3K p25 post-crisis & NHBE-10 bronchial   epithelial cells & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\
            BFT3B p28 post-crisis & BF-10 bronchial   fibroblasts        & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\
            BFT3G p28 post-crisis & BF-10 bronchial   fibroblasts        & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\
            BFT3K p34 post-crisis & BF-10 bronchial   fibroblasts        & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\ \hline
        \end{tabular}
    \end{adjustbox}
\end{table}
\clearpage

These cell lines represent independent immortalization events in a variety of cell lineages \cite{Bryan1995-ik,Shay1989-ua,Counter1992-yg}. We carried out whole genome sequencing of these nine post-crisis cell lines and their pre-crisis counterparts to a median depth of 40X (range: 15-51) and generated junction-balanced genome graphs\cite{Hadi2020-um} via JaBbA from SvABA\cite{Wala2018-qa} and GRIDSS\cite{Cameron2021-db} junction calls (see Appendix \ref{app:tc_wgs_processing}). We then apply JaBbA algorithm (Chapter \ref{chap:gg}) to reconstruct an optimal genome graph for each cell line and identify the patterns of structural variants (Chpater \ref{chap:complex_events}).

% Using short-read WGS data, JaBbA optimally assigns a copy number to both vertices (intervals) and edges (junctions, adjacencies) of genome graphs by fitting a probabilistic model to binned genome-wide read depth. These graphs obey a basic stoichiometric constraint of DNA dosage, namely that every copy of every (interstitial) segment must have a left and a right neighbor. The topology of these genome graphs can be further analyzed to identify simple and complex SV events, including chromothripsis and BFB cycles.

Comparison of ancestral (pre-crisis) and derived (post-crisis) genome graphs showed that eight of nine post-crisis cell lines acquired virtually all (61.9\% - 100\%, median 96.6\%) of their observed Structural variant during or after crisis (Figure \ref{fig:tc_prev_circos} and Supplementary Figure 1b in \cite{Dewhurst2021-jk}). One cell line (SW13) had acquired significant aneuploidy and genome rearrangement prior to crisis and was therefore difficult to interpret (Supplementary Figure 1b in \cite{Dewhurst2021-jk}). The other eight post-crisis genome graphs demonstrated varying levels of aneuploidy (ploidy ranges: 1.9-3.4) with variable numbers of clonal junctions per genome (range: 5-115, median 25). Analysis of junction-balanced genome graphs \cite{Hadi2020-um} revealed complex multi-chromosomal gains in six samples, with the other two lines harboring only broad arm level losses or gains (Figure \ref{fig:tc_prev_circos}).

\clearpage
Strikingly, besides one instance of chromothripsis (Figure \ref{fig:tc_ct_sw26}), genome graph-based categorization of complex SVs \cite{Hadi2020-um} identified few classic footprints of chromothripsis or BFB cycles in these genomes. However, several amplified subgraphs were associated with step-wise copy number gains reminiscent of BFB cycles (Figure \ref{fig:tc_bft3k_bfb}). The majority of copy changes in these subgraphs could not be attributed to fold-back inversion junctions (a hallmark of BFB cycles) but were instead driven by a spectrum of duplication and translocation-like junctions and templated insertion chains. These patterns are exemplified in a 10 Mbp region of 20q of post-crisis cell line BFT3B that is amplified to 10-15 copies, incorporating Mbp scale fragments from 11 other chromosomes at lower copy number including chromosome 8 and 19 (Figure \ref{fig:tc_bft3b_complex}). Of note, five of eight cell lines showed modest increases in \textit{TERT} copy number, providing a possible genomic basis for escape from telomere crisis (Figure \ref{fig:tc_prev_tert}). In Figure \ref{fig:tc_prev_tert}, each track shows binned purity- and ploidy-transformed read depth in units of CN, with variant (rearrangement) junctions and loose ends plotted as red arcs. The bottom track highlights the TERT gene among genes on chromosome 5p15. Violin plots to the left of tracks show the genome-wide distribution of read depth in units of CN, demonstrating that 6 of 7 clones have elevated CN at the TERT locus.

In summary, across the eight post-crisis cell lines, spontaneous escape from crisis was associated with a highly variable spectrum of SV patterns, ranging from relatively unaltered genomes to complex non-canonical patterns of amplification as well as numerical gains and losses. Importantly, BFB-like patterns and chromothripsis were not a general feature of the post-crisis genomes.

\section{An in vitro system for telomerase-mediated escape from natural telomere crisis}
% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_system.png}
%     \caption{\textbf{\textit{In vitro} iCRISPRa system to activate endogeneous TERT expression}}
%     \label{fig:tc_system}
% \end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_tert_western.png}
%     \caption{\textbf{Detection of TERT protein after doxycycline induction.}}
%     \label{fig:tc_tert_western}
% \end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_tert_qpcr.png}
%     \caption{\textbf{Expression of TERT mRNA after doxycycline induction.}}
%     \label{fig:tc_tert_qpcr}
% \end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_growth_original.png}
%     \caption{\textbf{Population doubling in induced versus control group over time.}}
%     \label{fig:tc_growth_original}
% \end{figure}

% \begin{figure}
%     \centering
%     \includegraphics{figures/600ppi/tc_stela.png}
%     \caption{\textbf{STELA of XpYp telomeres in MRC5/Rbsh/p21sh/iCRISPRa-TERT cells with or without doxycycline treatment at 70 and 150 days of culture.}}{}
%     \label{fig:tc_stela}
% \end{figure}

Despite the conclusions that both simple and complex SVs can appear in genomes after spontaneous escapes from telomere crisis, it still remains unclear what SVs are the direct result of telomere crisis. These existing cell lines spent various time in crisis, represent limited lineages that emerged from crisis, probably had undergone selection \ref{fig:tc_prev_tert}. To that end, Sally Dewhurst designed a new \textit{in vitro} model of telomere crisis using MRC5 human lung fibroblasts, which is known to reach mitotic barrier after about 50 generations. The Rb and p21 pathways were silenced with shRNAs to bypass senescence (data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 2A). Meanwhile, an inducible CRISPR activation system (iCRISPRa) is introduced to activate \textit{TERT} transcription (data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 2B). This system couples nuclease-dead Cas9 to a tripartite transcriptional activator and is guided by four gRNAs binding to the \textit{TERT} promoter region. Upon addition of doxycycline (dox), these MRC5/Rbsh/p21sh/iCRISPRa-TERT cells acutely expressed \textit{TERT} mRNA within 96 hours, in contrast to untreated cells in which \textit{TERT} transcripts were undetectable (data not included, refer to \cite{Dewhurst2021-jk} Figure 2B). Sally further verified that the telomerase activity (TRAP assay, telomerase repeated amplification protocol; data not included, refer to \cite{Dewhurst2021-jk} Figure 2C) are detectable, but both the expression and activity are lower than the telomerase-positive control cell line HCT116. This is a desired feature of this system as it has been shown that common \textit{TERT} promoter mutations found in tumor cells only sustain a low telomerase level and are not able to restore bulk telomere lengths \cite{Chiba2017-gq}.

% To gain a clearer insight into the nature of SVs that arise during telomere crisis, we developed an in vitro system in which we could reproduce telomere crisis and generate a large number of post-crisis clones. MRC5 human lung fibroblasts were chosen to model telomere crisis since they lack telomerase activity and as a consequence have a well-defined in vitro replicative potential determined by telomere attrition. To bypass senescence, the Rb and p21 pathways were inactivated by infecting the population of MRC5 cells with a retrovirus bearing shRNAs targeting the respective transcripts (data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 2A). This population of MRC5/Rbsh/p21sh was then endowed with an inducible CRISPR activation system (iCRISPRa) to activate the TERT promooter and induce telomerase expression (Figure \ref{fig:tc_system}). The iCRISPRa system employed a doxycycline-inducible nuclease-dead Cas9 fused to a tripartite transcriptional activator (VP64-p65-Rta) \cite{Chavez2015-md} and four gRNAs targeting the TERT promoter (Figure \ref{fig:tc_tert_western}, \ref{fig:tc_system}). Addition of doxycycline (dox) to MRC5/Rbsh/p21sh/iCRISPRa-TERT cells resulted in induction of TERT mRNA within 96 hours, whereas without dox, TERT transcripts are undetectable in this cell line (p<0.001, Figure \ref{fig:tc_tert_qpcr}). A similar dox-induced increase in mRNA expression was noted upon introduction of sgRNAs to a control gene (data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 2C). Induction of telomerase activity was readily detectable in a TRAP (telomerase repeated amplification protocol) assay (data not included, refer to \cite{Dewhurst2021-jk} Figure 2C). However, the induced TERT mRNA levels and the TRAP activity were significantly lower than in telomerase-positive control cell lines. The relatively weak telomerase activity in this system harmonizes with recent work showing that cancer-associated TERT promoter mutations initially result in low levels of telomerase activity that is not sufficient to maintain bulk telomere length \cite{Chiba2017-gq}.

In about 120 days after the initial cultutre (55 days with dox treatment), the MRC5/Rbsh/p21sh/iCRISPRa-TERT cells emerged more proliferative than the untreated group (Figure \ref{fig:tc_growth}). Using Single Telomere Length Analysis (STELA, \cite{Baird2003-cj}) at 150 days, it is observed that the activated telomerase expression can sustain proliferation yet is unable to restore bulk (average) telomere length (data not included, refer to \cite{Dewhurst2021-jk} Figure 2E). However, fewer critically short telomeres can be seen in the telomerase-activated group (data not included, refer to \cite{Dewhurst2021-jk} Figure 2E-F). This phenomenon is consistent with the hypothesis that in the cell population without telomerase, cells with critically short telomeres shift towards decease, resulting in the generally longer telomeres in the remaining cells. On the contrary, with telomerase, cells gain the ability to salvage the critically short telomeres and survive longer with on average shorter telomeres.

% At approximately 120 days after the start of the experiment (55 days with dox) the MRC5/Rbsh/p21sh/iCRISPRa-TERT population was proliferating faster than their untreated counterparts (Figure \ref{fig:tc_growth_original}). Inspection of individual telomere lengths using Single Telomere Length Analysis (STELA, \cite{Baird2003-cj}) revealed that although telomerase expression was sufficient to allow the cells to proliferate, it was not sufficient to maintain bulk telomere length (Figure \ref{fig:tc_stela}). After 150 days of continuous culture, the majority (86\%) of XpYp telomeres in induced MRC5/Rbsh/p21sh/iCRISPRa-TERT cells were between 1-4 kb compared to 40\% in uninduced cells (data not included, refer to \cite{Dewhurst2021-jk} Figure 2F, Supplementary Figure 2D). Consistent with this, genomic blotting showed bulk telomere shortening in both induced and uninduced cells (data not included, refer to \cite{Dewhurst2021-jk} Figure 2G). These telomere dynamics are consistent with the expectation that in the culture without telomerase, cells with critically short telomeres will preferentially be lost, leading to a surviving population with relatively longer telomeres. In contrast, cells in the induced culture with (low) telomerase activity have the ability to elongate the shortest telomeres. As a result, the induced cells are expected to tolerate telomere attrition better and present with overall shorter telomeres at later time points.

\clearpage
\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_growth.png}
    \caption{\textbf{Clones are isolated at day 120 and day 150 for whole genome sequencing.}}{Arrows indicate when each construct was introduced. Days in culture represent total time in culture from parental MRC5 cells to late passage MRC5/Rbsh/p21sh/iCRISPRa-TERT cells. Time points for telomere analysis and the approximate onset of senescence in the parental MRC5 cells are indicated.}
    \label{fig:tc_growth}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_heatmap_tert_control.png}
    \caption{\textbf{Control group with exogenous \textit{TERT} show no CNA.}}
    \label{fig:tc_heatmap_tert_control}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_sum_coverage_low.png}
    \caption{\textbf{Aggregated coverage of the putative BFB clones.}}
    \label{fig:tc_sum_coverage_low}
\end{figure}

\begin{table}[h!] \label{tab:tc_sequenced_clones}
    \centering
    \caption{\textbf{Number of clones analyzed by high and low-pass WGS.}}
    \begin{tabular}{ccc}
    \hline
    \multirow{0}{0pt}{Name} & \multicolumn{2}{c}{\textbf{Number   Sequenced}} \\ \cline{2-3} 
                        & Low Pass         & High   Pass         \\ \hline
    Parental cell lines   & 3                & 1                   \\
    Day 120 (Y) clones    & 37               & 5                   \\
    Day 150 (Z) clones    & 83               & 8                   \\
    Control (CT) clones   & 8                & 0                   \\ \hline
    \end{tabular}
\end{table}



% \begin{table}[] % \label{tab:tc_sv40t}
%     \caption{\textbf{SV40T immortalized pre-crisis and post-crisis telomerase positive cell lines.}}
%     \rotatebox[]{90}{
%         \begin{tabular}{ccccc}
%             \hline
%             \textbf{Name}                  & \textbf{Cell Line/Tissue Source}              & \textbf{Telomerase Status}   & \textbf{Reference}                 & \textbf{Kind gift of}             \\ \hline
%             HA1 p15               & Human Embryonic   Kidney cells       & -            & \cite{Counter1992-yg}  & Silvia   Bacchetti/AdVec \\
%             SW13 PD73             & IMR90 lung   fibroblasts             & -            & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
%             SW26 PD68.5           & IMR90 lung   fibroblasts             & -            & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
%             SW39 PD72.5           & IMR90 lung   fibroblasts             & -            & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
%             Bet3B p3              & NHBE-10 bronchial   epithelial cells & -            & \cite{Bryan1995-ik}     & Roger Reddel, CMRI       \\
%             Bet3K p7              & NHBE-10 bronchial   epithelial cells & -            & \cite{Bryan1995-ik}  & Roger Reddel, CMRI       \\
%             BFT3B p10             & BF-10 bronchial   fibroblasts        & -            & \cite{Bryan1995-ik}    & Roger Reddel, CMRI       \\
%             BFT3G p7              & BF-10 bronchial   fibroblasts        & -            & \cite{Bryan1995-ik}  & Roger Reddel, CMRI       \\
%             BFT3K p9              & BF-10 bronchial   fibroblasts        & -            & \cite{Bryan1995-ik}    & Roger Reddel, CMRI       \\
%             HA1-IM PD216          & Human Embryonic   Kidney cells       & + & \cite{Counter1992-yg}  & Silvia   Bacchetti/AdVec \\
%             SW13 PD184            & IMR90 lung   fibroblasts             & +  & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
%             SW26 PD130+           & IMR90 lung   fibroblasts             & +  & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
%             SW39 PD130+           & IMR90 lung   fibroblasts             & +  & \cite{Shay1989-ua} & Jerry Shay, UTSW         \\
%             Bet3B p25 post-crisis & NHBE-10 bronchial   epithelial cells & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\
%             Bet3K p25 post-crisis & NHBE-10 bronchial   epithelial cells & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\
%             BFT3B p28 post-crisis & BF-10 bronchial   fibroblasts        & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\
%             BFT3G p28 post-crisis & BF-10 bronchial   fibroblasts        & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\
%             BFT3K p34 post-crisis & BF-10 bronchial   fibroblasts        & + & \cite{Bryan1995-ik}      & Roger Reddel, CMRI       \\ \hline
%         \end{tabular}
%     }
% \end{table}
\section{Genomic screening of post-crisis clones}
To assess the genome structure of proliferating post-crisis cells, single cell clones were isolated from induced MRC5/Rbsh/p21sh/iCRISPRa-TERT cells at day 120 (âY clonesâ) and day 150 (âZ clonesâ) (Figure \ref{fig:tc_growth}). The clonal yield at day 150 was greater than at day 120 in induced cells but no clones could be isolated from the uninduced population at either time point. The lower clonal yield at day 120 may be due to incomplete stabilization of the telomeres since clones from this time-point showed a higher burden of fused telomeres than those derived from day 150 (data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 4A). Post-crisis clones from both time points showed evidence of ultra-short telomeres and reduced telomere length (data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 4B-C). Telomerase activity in post-crisis clones was comparable to the parental induced population, indicating that clone viability was not due to selection for increased telomerase activity (data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 4D). To generate control clones which had not passed through a period of telomere crisis, early passage MRC5 cells were infected with a retrovirus expressing hTERT and single cell clones were isolated (data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 4E). Genome profiling with low pass (~5X) WGS was performed on eight hTERT-expressing control clones (CT clones), 36 Y clones from day 120, and 82 Z clones from day 150 (Table \ref{tab:tc_sequenced_clones}).

\clearpage
\begin{figure}[h!]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_circos_low.png}
    \caption{\textbf{Circular heatmap showing genome-wide binned purity- and ploidy-transformed read depth.}}{Units are CN across 118 low-pass WGS-profiled clones. Heatmap rows correspond to concentric rings in the heatmap. Clones are clustered with respect to genome-wide copy number profile similarity (see âMethodsâ).}
    \label{fig:tc_circos_low}
\end{figure}

\begin{figure}[ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{Post-crisis clones are clustered into six groups based on CN profile of chromosome 12 and 21.}
        \label{fig:tc_heatmap_low}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/tc_heatmap_low.png}
    \end{adjustbox}
\end{figure}
\clearpage

Analysis of genome-wide read depth across 118 clones from both day 120 (Y clones) and day 150 (Z clones) demonstrated predominantly diploid genomes with a striking enrichment of clones with DNA loss on most of chromosome 12p (63\%, 74/118, Figure \ref{fig:tc_circos_low}). Within the other 44 samples, we observed a subset of clones (5\%, 6/118, Figure \ref{fig:tc_circos_low}) with gains of chromosome 21q. As expected, control CT clones showed no evidence of SVs or copy number variants (Figure \ref{fig:tc_heatmap_tert_control}). Hierarchical clustering of all clones by their coverage on chromosomes 12p and 21q revealed six distinct clusters (Figure \ref{fig:tc_heatmap_low}). A minority of clones were diploid on chromosomes 12 and 21 and elsewhere in the genome and are therefore designated as âunrearrangedâ (32\% of clones, 38/118). Of note, the unrearranged group was enriched in day 120 (Y) samples compared to day 150 (Z) samples ($p = 1.79 \times 10^{-9}$, odds ratio 14.7, Fisherâs exact test; Figure \ref{fig:tc_heatmap_low}), suggesting that these clones may have largely avoided crisis prior to telomerase induction. The cluster of clones with 21q gain were diploid on 12p.

The remaining 74 clones (63\%) all showed a heterogeneous pattern of copy number alterations targeting 12p (Figure \ref{fig:tc_heatmap_low}). One out of the 118 clones (0.8\%) displayed the singular pattern of distinct interspersed losses that resembled chromothripsis. Complete loss of one copy of 12p ('arm loss') was found in a cluster of 6 clones (6/118, 5\%). A second cluster of 67 clones all shared a breakpoint near the distal end of 12p and a large deletion starting ~9 Mbp from the centromere. These clones were differentiated into two clusters by the presence or absence of an amplification around 8-9 Mbp from the 12p telomere. In the 47 clones that contained this amplification, aggregated consensus read depth profiles revealed step-wise gains at the distal end of 12p, a pattern reminiscent of BFB cycles (Supplementary Figure \ref{fig:tc_sum_coverage_low}). This cluster was therefore labeled âBFB-likeâ, a designation which is further supported by data presented below. The 20 clones (17\%) that lack the amplicon around 8-9 Mbp harbored varying boundaries of the shared larger deletion; based on the analysis described below we designate these as âearly BFB-likeâ. In summary, these low-pass WGS copy number profiles indicated a limited set of distinct lineages surviving telomere crisis, with at least two lineages independently converging on 12p.

% insert the formulation of multisample graph fitting -- walk decomposition
\section{Joint inference of junction balance in MRC5} \label{sec:joint_jabba_mrc5}
To chart structural variant evolution across sub-clades of MRC5 clones, a procedure was developed to jointly infer junction balanced genome graphs in a lineage (e.g. BFB lineage in Figure \ref{fig:tc_evo_tree}). This co-calling algorithm augmented the existing JaBbA model, described in detail in \cite{Hadi2020-um}, enabling application to a compendium of genome graphs by minimizing the total number of unique loose ends assigned a nonzero copy number across the graph compendium.  

Formally, we define a collection ${G^i = (V^i, E^i)}, i \in 1,\dots,n$ of identical genome graphs across $n$ clones, each a replica of a \textit{prototype} genome graph $G^0 = (V^0, E^0)$. The mapping $p:V \cup E \rightarrow V^0 \cup E^0$ maps each vertex $v \in V^i$ and edge $e \in E^i,i \in 1,\dots,n$ to its corresponding vertex $p(v)\in V^0$ and edge $p(e) \in E^0$ in the prototype graph. We then jointly infer unique copy number assignments $\kappa^i$ to the vertices and edges of each genome graph $G^i$ by solving the mixed integer program:

\begin{equation} \label{eq:multi_sample_jabba}
    \begin{aligned}
        \underset{\kappa^i:V_I(G^i) \cup E(G^i) \rightarrow \mathbb{N}, i \in 1,\dots,n}{\text{minimize}} &
        \lambda \mathcal{R}(\{G^i\}_{i \in 0, \dots, n}, \{\kappa^i\}_{i \in 1, \dots, n}, p) + \sum_{i \in 1,\dots,n}\mathcal{V}(G^i, \kappa^i, x^i, J^i) \\
        \text{subject to: } & \kappa^i(v) = \kappa^i(\bar{v}), \forall{v \in V_I^i} \\
        & \kappa^i(e) = \kappa^i(\bar{e}), \forall{e \in E^i} \\
        & \kappa^i(v) = \sum_{e \in E^+(v,G^i)}\kappa^i(e) = \sum_{e \in E^-(v, G^i)}\kappa^i(e) \\
        & \kappa^i(e) \le u^i(e)
    \end{aligned}
\end{equation}

where $x^i$ and $J^i$ represent the binned read depth data and bin-node mappings for clone $i$ and  $\mathcal{V}(G^i,Îº^i,x^i,J^i)$ is the read depth residual for genome graph i, analogous to \ref{eq:MIQP}.  A new term in this joint formulation is $u^i:E(G^i) \rightarrow \{0,\inf\}$, which is a constraint on the upper bound of each edge CN $\kappa(e) \in E(G^i)$, e.g. zero if there have been no read support for the junction in clone $i$. 

The coupling of multiple graphs happen through a joint complexity penalty $\mathcal{R}$, which only penalize a loose end once for all the clones. Formally,

\begin{equation} \label{eq:joint_loose_penalty}
    \mathcal{R}(\{G^i\}_{i \in 0, \dots, n}, \{\kappa^i\}_{i \in 1, \dots, n}, p) = \sum_{e^{0}_{L} \in E_{L}(G^0)}[\![ \sum_{\hat{e_L}|\hat{e_L} \in E^{i}_{L}, p(\hat{e_L})=e^{0}_{L}, i \in 1,\dots,n} \kappa^i( \hat{e_L} ) ]\!]
\end{equation}

As in \ref{eq:MIQP}, the hyperparameter $\lambda$ in Equation \ref{eq:multi_sample_jabba} controls the relative contribution of the read-depth residual and complexity penalty to the objective function. 

It is important to note that while each of the graphs $G^i$ have an identical structure, the constraints imposed by the upper bounds $u^i$ and bin profiles $x^i$ confine each graph to its junctions and read depth data, and hence lead to a unique fit $Îº^i$ on the basis of this data. The $L_0$ penalty (defined using the Iverson bracket $[\![]\!]$ operator) in Equation \ref{eq:joint_loose_penalty} couples the solutions $Îº^i$ by adding an exponential prior on the number of unique loose ends across the entire graph compendium, where uniqueness is defined by the mapping $p$ to the prototype graph $G^0$.

This joint mixed-integer programming model in Equation \ref{eq:multi_sample_jabba} is implemented in the âbalanceâ function of gGnome. The model was applied to a collection of genome graphs representing the structure of chromosome 12 across 13 clones. The prototype graph for this genome graph collection was built from the disjoint union of intervals of the 13 preliminary graphs (via the GenomicRanges âdisjoinâ function) and the union of junction calls fit across those graphs (via gGnome âmerge.Junctionâ function). Each graph was associated using the read depth data and bin-to-node mappings as per \cite{Hadi2020-um}. The mapping $u^i$ for each reference edge was set to $\inf$ while variant edges were assigned $\inf$ if bwa mem realignments of read pairs in each clone .bam file to the corresponding junction contig via rSeqLib (https://github.com/mskilab/rSeqLib, \cite{Wala2017-ud}), otherwise they were assigned 0.

Equation \ref{eq:multi_sample_jabba} was then solved using the IBM CPLEX (v12.6.2) MIQP optimizer within the gGnome package after setting the hyperparameter $\lambda$ to 100.  This value was chosen after a parameter sweep observing for the visual concordance of genome graphs, loose ends, and read depth profiles in the region.

\clearpage
\begin{figure}[!ht]
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_evo_tree.png}
    \caption{\textbf{Evolutionary trajectory of SVs in clones with high-pass WGS.}}
    \label{fig:tc_evo_tree}
\end{figure}
\clearpage
% {Each terminal node in the tree is associated with a unique 12p profile comprising a representative binned read-depth pattern (bottom track) from one or more clones mapping to an identical junction-balanced genome graph (second track from bottom). The top track in each profile represents a reconstruction of the rearranged allele. Each allele is a walk of genomic intervals and reference/variant junctions that, in combination with an unrearranged 12p allele (not shown), sum to the observed genome graph. Two distinct arrows linking Y11 and Y15 demonstrate that these clones are distinct lineages.}

% \begin{figure}
%     \centering
%     \includegraphics[angle=90,height=0.75\textheight]{figures/600ppi/tc_phylo.png}
%     \caption{\textbf{Phylogeny reconstructed based on SNV VAFs, CNA segmentation, and the presence of junctions.}}
%     \label{fig:tc_phylo}
% \end{figure}

% {From left to right: phylogenetic tree; heatmap of variant allele fractions (VAF) for SNVs detected among two or more clones, private SNVs are omitted for simplicity; copy number segmentation; junction presence.}

\section{High-resolution reconstruction and lineage of post-crisis genomes} \label{sec:tc_joint_decomposition}
To gain further insight into structural variant evolution along these lineages, we chose 15 representative clones spanning the 5 clusters with rearrangements involving 12p for high-depth WGS to a median read depth of 50X (range: 30-88). Phylogeny derived from genome-wide SNV patterns demonstrated a median branch length of 551 SNVs (range: 9-2,409), a low mutation density (<1 SNV/Mbp) that is consistent with previous WGS studies of clones in cell culture \cite{Petljak2019-cf}. This analysis revealed four major clades (Figure \ref{fig:tc_phylo}). These clades had good concordance with copy number alteration and rearrangement junction patterns in the same 12p region, suggesting these clones represent distinct post-crisis evolutionary lineages (Figure \ref{fig:tc_phylo}).

In order to further reconcile the shared and distinct rearrangement junctions present in the evolution of these clones, we carried out local assembly of rearrangement junctions and junction balance analysis (see \ref{sec:tc_joint_jabba}) which revealed 7 distinct junction-balanced genome graphs spanning 12p (Figure \ref{fig:tc_evo_tree}). With the exception of the chromothriptic lineage (see below), each of these distinct lineages was represented by more than one post-crisis clone.

To reconstruct a set of linear alleles that parsimoniously explain these different genome graph patterns3 (Figure \ref{fig:tc_evo_tree}), we applied gGnome to the data (see Section \ref{sec:tc_joint_decomposition}). We constrained our model to contain one intact allele of chromosome 12 for the following reasons: 1) karyotypes and chromosome painting showed a single copy of chromosome 12 was altered in the post-crisis clones (see Figure 6b-c of \cite{Dewhurst2021-jk}); and 2) rearrangement of one allele is more likely than rearrangement across two alleles. Application of this constraint to the full set of MRC5 clones in a joint inference revealed a parsimonious set of rearranged alleles that explained the observed collection of clonally-related junction-balanced graphs (Figure \ref{fig:tc_evo_tree}).

\section{Joint reconstruction of allelic evolution in MRC5} \label{sec:tc_joint_jabba}
Evolving 12p alleles were jointly reconstructed across 13 MRC5 clones through the decomposition of junction balanced genome graphs $(G^i,Îº^i)$ (see Section \ref{sec:joint_jabba_mrc5} section above). The procedure for joint allelic phasing described in \cite{Hadi2020-um} and Eq. \ref{eq:walk_decomposition} was extended to identify the most parsimonious (least unique walks needed) collection of paths and/or cycles $H$ and associated walk multiplicity $\kappa(H)$ that summed to the vertex and edge copy numbers in the compendium $(G^i,Îº^i)$.

Formally, the subgraph of vertices and edges with a nonzero copy number in each $(G^i,Îº^i)$ were exhaustively traversed to derive all minimal paths and cycles $H^i$, where for each walk $h \in H^i$ maps to incident sequences of $V(h) \subseteq V^i$ and $E(h) \subseteq E^i$ of vertices and edges in the graph $G^i$.  The nodes and vertices of these walks were then projected to via the mapping $p$ to define a unique set of walks $H^0$ in the prototype graph $G^0$. We extend our notation $p$ (see previous section) so that for a walk $h \in H^i$ the mapping $p(h) \in H^0$ denotes the walk formed by projecting the vertices and edges of $h$ via $p$ to $H^0$.  With these definitions, the single graph haplotype inference defined in \ref{eq:walk_decomposition} was extended to a joint inference by solving the following mixed integer linear program to assign a copy number $\kappa^i(h) \in \mathbb{N}$ to each walk $h \in H^i$:

\begin{equation} \label{eq:joint_walk_decomposition}
    \begin{aligned}
        \underset{\kappa^i(h):H^i \rightarrow \mathbb{N}, i \in 1, \dots, n}{\text{minimize}} & \sum_{h \in H^0} [\![ \sum_{\hat{h} \in H^i | p(\hat{h} = h)} \kappa^i(\hat{h}) ]\!] \\
        \text{s.t. } & \kappa^i(v) = \sum_{\hat{h} \in H^i} \kappa(\hat{h})\delta(\hat{h}, v), \forall v \in V^i, i \in 1, \dots, n \\
        & \kappa^i(e) = \sum_{\hat{h} \in H^i} \kappa(\hat{h})\delta(\hat{h}, e), \forall e \in E^i, i \in 1, \dots, n
    \end{aligned}
\end{equation}

where the function $\delta(v,h)$ and $\delta(e,h)$ is 1 if vertex $v$ and edge $e$ belong to walk $h$ and 0 otherwise, just like in \ref{eq:walk_decomposition}. The Iverson bracket ($[\![]\!]$) operator in the objective function Equation \ref{eq:joint_walk_decomposition} minimizes the total number of unique walks used across the compendium, hence identifying a jointly parsimonious assignment of copy number to walks across the compendium of graphs. Equation \ref{eq:joint_walk_decomposition} was solved using the IBM CPLEX (v12.6.2) MIQP optimizer within the gGnome package. Variant cycles and paths from the resulting solution were manually combined to yield a set of consistent linear paths, i.e. somatic haplotypes, to yield allelic reconstructions in Figure \ref{fig:tc_evo_tree}.

\clearpage
\begin{figure}[ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{Phylogeny reconstructed based on SNV VAFs, CNA segmentation, and the presence of junctions.}
        \label{fig:tc_phylo}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/tc_phylo.png}
    \end{adjustbox}
\end{figure}
\clearpage

\section{Evolutionary trajectory of a post-crisis chromosome 12}
Analyzing the clonal evolution of these rearranged 12p alleles, we identified 8 clones demonstrating progressive stages of a BFB cycle. This complex variant evolved after a long-range inversion junction (j1) joined a distal end of 12p to its peri-centromere. This junction was followed by subsequent fold-back inversion junctions (j2, j3, j4), clustered at the 8-9 Mbp focus on 12p, which are present in two different sets of post-crisis clones (Early BFB, BFB, Figure \ref{fig:tc_evo_tree}). The earliest of the fold-back inversion junctions (j2) in the BFB lineage was associated with a cluster of 3 G or C mutations within 2 kbp of each other, consistent with APOBEC-mediated mutagenesis \cite{Maciejowski2020-bw} (Figure \ref{fig:tc_kataegis}). The most complex locus in the BFB lineage (Z43, Late BFB, Figure \ref{fig:tc_evo_tree}), contained six variant junctions in \textit{cis}, including two late DUP-like junctions (j5, j6). Although j6, which connects the distal portion of 12p to the 12p centromere, was not directly observed in the short read WGS data, it was imputed (dashed line, j6, Figure \ref{fig:tc_evo_tree}) to resolve the duplication of j1 in clone Z43 (JCN=2), as well as two loose ends in the genome graph. Remarkably, the vast majority (97\%) of SNVs detected in this BFB lineage (Figure \ref{fig:tc_phylo}) were either shared by all clones or private to a single clone, indicating that these stages of BFB evolution occurred rapidly in the history of the experiment.

\clearpage
\begin{figure}
    \centering
    \includegraphics[height=0.85\textheight]{figures/600ppi/tc_kataegis.png}
    \caption{\textbf{Clustered GC mutations around junction breakends.}}
    \label{fig:tc_kataegis}
\end{figure}
% {g SNV patterns as a function of GC vs AT reference nucleotide context, with y-axis showing the logarithm of the inter SNV distance. Highlighted region represents a GC mutation cluster that is found across all four clones in the BFB cluster which harbour junction j2.}

\begin{figure}
    \centering
    \includegraphics[height=0.85\textheight]{figures/600ppi/tc_12q_del.png}
    \caption{\textbf{Simple deletion on chromosome 12q present in Y15, Y8, Z43, but not Y11.}}{Junction supporting read pairs and corresponding drop in coverage at the small deletion on chromosome 12q in Y15, Y8, Z43, and lack of such evidence in Y11.}
    \label{fig:tc_12q_del}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[height=0.85\textheight]{figures/600ppi/tc_loose_end.png}
    \caption{\textbf{Distinct loose ends among the diverging BFB clones.}}
    \label{fig:tc_loose_end}
\end{figure}
\clearpage
We confirmed a chromothripsis event in an independent lineage (Y8), which lacked j1 and all subsequent junctions of the BFB lineage, further supporting the idea that this is an independent lineage (Figure \ref{fig:tc_evo_tree}). Integration of copy number data with the SNV phylogeny showed clones from the unrearranged lineage (Y1 and Y4) and one of the 12p arm loss clones (Y11) to be mutationally distant (>2,000 SNVs) from the chromothripsis (Y8) and BFB lineages, which shared over 1,583 SNVs (Figure \ref{fig:tc_phylo}). Supporting this, a small (~21.5 kbp) simple deletion junction was shared across Y8, Y15, and all the BFB lineage samples, yet was absent in Y11 (Figure \ref{fig:tc_12q_del}).

This comparison established that the 12p loss in Y11 could not have occurred after j1 and indicates that a second independent arm loss must have given rise to Y15. Interestingly, the Y15 arm loss clone was clustered in the BFB/Y8 clade in the SNV phylogeny, sharing 30 SNVs with the BFB lineage which it did not share with Y8 (Figure \ref{fig:tc_phylo}). This indicates that the 12p arm loss in Y15 may have arisen either before or after j1. Although the breakpoints of the Y11 and Y15 arm losses could not be mapped due to their location in the 12 centromeric region, based on the SNV phylogeny, they likely represent distinct events. Taken together, these results support a model whereby at least three lineages independently rearranged a previously wild type 12p during telomere crisis (Figure \ref{fig:tc_evo_tree}). Our data appear to have captured sequential steps in the formation of an increasingly complex BFB-like event. Each of these stages must represent a stabilized allele since the post-crisis lines are clonal, and multiple clones share the same rearrangement junctions (Figure \ref{fig:tc_phylo}). This necessarily raises the question as to what caused the on-going instability, and how and where these complex alleles are terminated.

\section{Resolution of BFB cycles in telomere crisis}
% % \begin{figure}
% %     \centering
% %     \includegraphics[height=0.95\textheight]{figures/600ppi/tc_sky.png}
% %     \caption{\textbf{DAPI-banded karyotypes of post-crisis clones Z43 and Y8.}}
% %     \label{fig:tc_sky}
% % \end{figure}

% % \begin{figure}
% %     \centering
% %     \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_sky_all.png}
% %     \caption{\textbf{DAPI-banded karyotypes of ancestral MRC5 and post-crisis clones Y4, Z11, Z15, Z29, and Z41.}}
% %     \label{fig:tc_sky_all}
% % \end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[height=0.9\textheight]{figures/600ppi/tc_fish_bicolor.png}
%     \caption{\textbf{Representative images of metaphase spreads from post-crisis clones hybridized with both 12 (green) and 21 (red) chromosome paint in post-crisis clones Y8 and Z43.}}
%     \label{fig:tc_fish_bicolor}
% \end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_fish_bicolor_all.png}
%     \caption{\textbf{Representative images o fmetaphase spreads from post-crisis clones hybridized with both 12 (green) and 21 (red) chromosome paint in ancestral MRC5 and six post-crisis clones.}}
%     \label{fig:tc_fish_bicolor_all}
% \end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[height=0.7\textheight]{figures/600ppi/tc_fish_tree.png}
%     \caption{\textbf{Images of derivative chromosome 12:21 from representative clones from each branch of the evolution of chromo-some 12 post-crisis.}}{Metaphases were hybridized with whole chromosome paints for 12 (green) and 21 (red). DNA was stained with DAPI (gray).
%     }
%     \label{fig:tc_fish_tree}
% \end{figure}

Analysis of junction-balanced genome graphs allows for the nomination of âloose endsâ (or allelic ends), representing copy number changes that cannot be resolved through assembly or mapping of short reads. We identified three distinct loose ends across the 4 variant graphs spanning the 8 clones in the BFB lineage (Figure \ref{fig:tc_loose_end}). Each of these loose ends were placed at the terminus of their respective reconstructed allele, and we posit they represent the new âendsâ of the derivative alleles of the BFB lineage. Distinct ends for each of these rearranged lineages suggests the derivative 12p allele could have been stabilized independently. We did not observe telomere repeat-containing reads mated to these loose ends (Appendix \ref{app:tc_loose_end}), arguing against neo-telomere formation at these loci. Instead, loose reads represented highly repetitive unmappable sequences which may be a result of the junctions being in close proximity to centromeric regions (see below).

To resolve the genomic architecture at these loci we generated karyotypes from metaphase spreads for representative rearranged clones (refer to \cite{Dewhurst2021-jk} Figure 6a and Supplementary Figure 6a), which revealed that in the BFB and chromothripsis (Y8) lineages, the chromosome 12 derivative was likely linked to a copy of chromosome 21 with an intact long arm (q). These observations were confirmed with chromosome painting, demonstrating a derivative chromosome transitioning between 12 and 21 (refer to \cite{Dewhurst2021-jk} Figure 6b and Supplementary Figure 6b). Two possible events can explain these findings: the 12-21 fusion could have occurred as an early event during telomere crisis, preceding the divergence of Y8 (chromothriptic) and the BFB lineage; alternatively, independent 21 fusion events stabilized the derivative chromosome 12 following formation of the distinct junction lineages in Figure \ref{fig:tc_evo_tree}. We consider the first possibility unlikely since the creation of the long-range inversion (j1) and subsequent fold-back junctions in the BFB lineage would require the formation of interstitial 12p breaks on a 12p-21 derivative chromosome. Such breaks are predicted to result in the loss of 21, which would be distal to these junctions on the fusion allele. Furthermore, the acrocentric nature of chromosome 21 would make it more likely to stabilize the overall chromosome architecture, suggesting that an early 12-21 derivative chromosome would be unlikely to engage in the additional SV events observed in the BFB lineage. We therefore consider it likely that each of the BFB cycles and chromothripsis clones were independently resolved through subsequent fusion to 21 (refer to \cite{Dewhurst2021-jk} Figure 6c).

Unlike the BFB and chromothripsis clusters, one of the two 12p arm loss lineages (Y11) did not appear to be fused to chromosome 21. In this clone, the derivative chromosome 12 appears to contain a distinct fusion with a longer p-arm (refer to \cite{Dewhurst2021-jk}Supplementary Figure 6b). This is consistent with our analysis of the SNV phylogeny, showing Y11 to be mutationally distant from the BFB lineage (Figure \ref{fig:tc_phylo}, \ref{fig:tc_12q_del}).  We were unable to further resolve the nature of the stabilization event in this clone. It would be necessary to perform long molecule DNA sequencing across different lineages in order to confirm the distinct nature of the fusion junction in each of the post-crisis clones.

\clearpage
\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_phasing_define.png}
    \caption{\textbf{Defining L and R alleles of chromosome 12p based on its loss or retention in arm-loss clone Y11.}}{Genomic track plots of parental alleles phased into lost (L) and retained (R) haplotypes on chromosome 12p of clone Y11.}
    \label{fig:tc_phasing_define}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_phasing_y8.png}
    \caption{\textbf{Chromosome 12p haplotype in chromothripsis-like post-crisis clone Y8.}}
    \label{fig:tc_phasing_y8}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_phasing_z43.png}
    \caption{\textbf{Chromosome 12p haplotype in chromothripsis-like post-crisis clone Z43.}}
    \label{fig:tc_phasing_z43}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_allelic_cn_high.png}
    \caption{\textbf{Haplotype CN in high-pass WGS.}}{Scatter plot showing purity- and ploidy-transformed L and R haplotype-specific allelic read depth across 12p segments in high-pass WGS-profiled post-crisis clones.}
    \label{fig:tc_allelic_cn_high}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/tc_allelic_cn_low.png}
    \caption{\textbf{Haplotype CN in low-pass WGS.}}{Scatter plot showing purity- and ploidy-transformed L and R haplotype-specific allelic read depth across 12p segments in low-pass WGS-profiled post-crisis clones.}
    \label{fig:tc_allelic_cn_low}
\end{figure}
\clearpage

\section{A short telomere renders 12p vulnerable to telomere attrition}
The convergent evolution patterns observed in our system suggests either 12p vulnerability to rearrangements or selection for 12p loss during telomere crisis. We believe strong selection is unlikely, given the existence of day 150 clones with diploid 12p (15.8\%, 13/82, with or without 21q gain). The preferential rearrangement of the short arm of chromosome 12 in the post-crisis system could be explained if one of the two 12p telomeres is among the shortest telomeres in the MRC5 parental cells. Attrition of the shortest telomeres is predicted to generate the first telomere fusions and associated rearrangements in the culture.

We first asked whether the same parental allele was targeted across the chromosome 12- associated events in our cohort. Such allele specificity would argue against a selection for loss of 12p sequences since such selection should have occurred without allele preference. We phased heterozygous SNPs on 12p on the basis of whether they belonged to the lost (L) or retained (R) allele on the early 12p arm loss clone, Y11 (Figure \ref{fig:tc_phasing_define}). Analyzing phased SNP patterns across all the highâ and low-pass MRC5 clone WGS profiles in our dataset demonstrated that the L allele of 12p was the exclusive target of all chromosome 12 structural variants (Figure \ref{fig:tc_allelic_cn_high}, \ref{fig:tc_allelic_cn_low}). This included the clones from the chromothripsis (Y8, Figure \ref{fig:tc_phasing_y8}) and BFB (Z43, Figure \ref{fig:tc_phasing_z43}) lineages, which our phylogenetic clustering suggested to be likely independent events on a previously unrearranged chromosome 12 (Figure \ref{fig:tc_evo_tree}). On the basis of these results, we concluded that the short arm of the L allele of 12 was the most vulnerable to rearrangement in the MRC5 parental line.

%% shorten this part
We next tested whether the preferential 12p events could be due to the presence of a short telomere on one of the 12p alleles. To this end, we combined telomeric FISH with BAC probes specific for chromosome 12 and two other chromosome (6 and 8) that did not show evidence for structural variants in WGS (Figure \ref{fig:tc_heatmap_low}). Comparing the ratio of the telomeric signal of the shortest 12p telomeres to the signal of all other telomeres in individual metaphase spreads revealed that one of the 12p telomeres was significantly shorter (data not included, refer to \cite{Dewhurst2021-jk} Figure 7e). The shortest telomeres of 6 and 18 (data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 7b) were also shorter than the median but not to the same extent as 12p. The relative telomere length of the shortest 21p allele showed a heterogeneous distribution that overall was significantly longer than 12p in the parental cells (data not included, refer to \cite{Dewhurst2021-jk} Figure 7e). This does not exclude the possibility of 21 becoming critically short at later time points, and indeed the observation of a low percentage of clones in the 5X WGS screening with amplifications of 21q could indicate that this chromosome end did occasionally become deprotected in this population (Figure \ref{fig:tc_heatmap_low}). Such deprotection of a chromosome 21 telomere is consistent with chromosome 21 preferentially stabilizing the derivative chromosome 12 (data not included, refer to \cite{Dewhurst2021-jk} Figure 6b, Supplementary Figure 6a).

To look for evidence of chromosome 12 being involved in the initial fusion events in this system, we combined a chromosome 12 BAC probe with a centromere probe in MRC5/Rbsh/p21sh/iCRISPRa-TERT cells in crisis (at day 90). Strikingly, we observed a number of instances of chromosome 12 within chromosome fusion events (data not included, refer to \cite{Dewhurst2021-jk} Figure 7f). The fraction of chromosome fusions involving chromosome 12 is higher than expected (~50\% observed versus ~4\% expected, data not included, refer to \cite{Dewhurst2021-jk} Supplementary Figure 7c). Collectively, these data support the hypothesis that a short telomere on one allele of 12p increased the chance of 12p partaking in a fusion event that preceded subsequent rearrangement lineages.

\section{Discussion}
To precisely define the chromosomal consequences of telomere crisis, we have described the first whole genome profiles of cells emerging from natural telomere crisis, both in the setting of spontaneous and controlled telomerase activation. 

Previously published post-crisis cell lines show a wide variety of SV complexity. Some have little to only simple, arm-level CNAs. Some have extensively rearranged chromosomes ending up with complex amplification. However, there is only one instance that fit previously implicated chromothripsis patterns as defined by the gGnome event callers trained from pan-cancer data. Even though several amplicons with fold-back inversions are discovered, they are still relatively primitive compared to the BFB cycles defined in cancer genomes. Hence, canonical cancer complex SVs of chromothripsis and BFB cycles are not a necessary outcome of telomere crisis, even though they are possible. In other words, their presence alone cannot be used as a sufficient indicator that a cancer went through telomere crisis. These cells were not subject to any artificial controls and thus have spent varying time in crisis, potentially with different severity, and natural selection.

Hence we built a new \textit{in vitro} telomere crisis system. By adding an inducible system of \textit{TERT} expression in p16/Rb1 silenced MRC5 cells, we can precisely time the escape from telomere crisis. Plus, MRC5 cells do not have known defects in major DNA repair pathways, like the majority of cancers at early stages, so that the alterations post-crisis can be directly attributed to the telomere crisis. Indeed, based on our observations of relatively low \textit{TERT} expression, low dicentric chromosome count, we consider our model closer to what commonly happens in cancerous cels. Using this model, by a combination of WGS and cytogenetics, we characterized the start, progression, and termination of the SV evolution during telomere crisis. 
% (what else is good: natural attrition, not TRF2 del; low dicentric count; low TERT expr;)



In the low-depth WGS screening, we observed the only chromosome arms showing prevalent CNAs are chromosome 12p and 21q. The six clones of 21q amplification is mutually exclusive with the 12p altered groups, in which 21q appears to be diploid. The rest of the clones that are devoid of major CNAs throughout the genome, covering both early and late harvest time, and they represent the clones that escaped telomere crisis readily before any SVs could accumulate. The very existence of unaffected clones even on 150 days in, shows us that the SVs or lack thereof are not the result of positive selection.

To reconstruct the different yet related chromosome 12p structures, we further generalized our JaBbA formulation to penalize unique loose end locations across several genome graphs (see Eq. \ref{eq:joint_loose_penalty}), ergo minimizing complexity of a genome graph population. Without this extension, each genome graph may place loose ends based on each coverage profile, resulting in discrepancy between samples for the same true loose end, in turn, misleading the downstream phylogeny inference. This development demonstrated the adaptability of gGnome and bear crucial application in other multiple related samples study designs, for example, single cell DNA sequencing of heterogeneous cell populations.

% the start
The reconstructed graphs showed three main branches of outcomes to chromosome 12p: arm loss with breakpoint hidden within the centromere, chromothripsis, and several stages of BFB cycle. Taking advantage of the complete arm-loss clone Y11, we phased the germline variants on chromosome 12p and found a perfect accordance among all these clones in both low- and high-pass WGS: there is a single chromosome 12 allele that was altered in all these samples. This convinces us the initial trigger, or the first critically short telomere was from chromosome 12p. This is also indirectly shown as one intact chromosome 12 exists in each of these clones by FISH. This point is further verified by telomere binding FISH and found that chromosome 12p does have the shortest telomere when compared to several other chromosomes in MRC5 cell line. This finding is consistent with prior hypothesis that the shortest telomere becomes unstable and initiate the chromosomal alterations during crisis.

% the progression
To reconstruct a comprehensive evolutionary trajectory, we combined evidence from the exact structural haplotype and genome-wide SNVs. We exhaust all possible haplotypes that can be derived from the graphs and search for a linear combination of them that satisfies each graphs copy numbers. To achieve that through multiple related samples, we have to extend the Eq. \ref{eq:walk_decomposition_parsimony} to the form in Eq. \ref{eq:joint_walk_decomposition}. Our exhaustive nomination of elementary paths and cycles from a graph, even though guaranteed to be feasible, any cycle is singled out and on face value form itself a circular contig, even for a true tandem duplication. This is biologically improbable so we modified the pool of haplotypes by inserting cycles back to all possible locations among the walks, and fitted copy number to a more biologically meaning full haplotype repertoire. Thanks to the knowledge of a single chromosome 12 allele being affected, we further simplify the inference by requiring a unrearranged chromosome 12 haplotype to present at copy number 1 in each sample.

\clearpage
While we inferred this tree (Figure \ref{fig:tc_evo_tree}) in a customized deduction, there is great opportunity to build a general-purpose, automated evolutionary trajectory inference method from jointly inferred haplotypes. Interestingly, the two arm-loss configurations are actually distant on the SNV phylogenetic tree, indicating convergent evolution. Y15 is closer to the BFB clones than to the other arm-loss Y11 clone, which was further supported by a small simple deletion event on the q-arm of chromosome 12. Assuming a relatively constant rate of mutation, the diverging process of the BFB clones happened in a shorter time frame. 

% the stabilization
Since all of these data are sequenced from stable, supposedly isogenic cell populations, after telomere crisis, there should not be ongoing genome instability, and it requires each of them to be stabilized in some way. In other words, the damaged chromosome 12p must have regained functioning telomere. Judging from the karyotyping, most of these clones lack a copy of chromosome 21, even though in WGS chr21 seems intact in copy number. Supported by bi-color FISH of both chromosomes 12 and 21, we found the acrocentric chromosomes have been fused to the damaged chromosomes 12. Contrary to our first hypothesis that this fusion happened in ancestral state and early on, the topology of junctions gained through the process requires the breakage between chromosome 12p and 21q if it was the case. We deem this breakage and re-fusion highly improbable, so we have to land in the conclusion that these fusions happened independently, implying striking convergent mechanism to stabilize a structurally unstable chromosome. 

In conclusion, our data reveal that telomere crisis can instigate a wide spectrum of Structural variants in the viable descendants of this genomic trauma. First, our results indicate that cells can emerge from telomere crisis with minimally altered genomes. Second, BFB cycles and chromothripsis are not a universal hallmark of post-crisis cell lines. Third, our results indicate that natural telomere crisis can manifest as a focal and diverse cascade of SV events converging on a single chromosome arm. Since no single class of Structural variant appears to be a hallmark of past telomere crisis, other genomic insignia will have to be identified in order to determine whether a given cancer has experienced telomere dysfunction in its proliferation history.

% Analysis of a variety of post-crisis genomes from divergent lineages and independent immortalization events uncovered highly complex patterns of copy number amplification and rearrangement. However, many of the genomes showed minimal genomic alterations. The rearrangements we did observe were not typified by the expected predominance of fold-back inversions that are indicative of BFB cycles or low amplitude copy number oscillations associated with chromothripsis. These cell lines spent a varying amount of time in telomere crisis, potentially with very different numbers of chromosome fusions, which is hard to quantify with limited historical data available. Due to the limited similarities between these cell lines, we constructed an in vitro system that allowed us to sequence high numbers of post-crisis genomes.

% We consider our in vitro system to be a good representation of telomere crisis for a number of reasons. Telomeres in this system have been eroded through replicative attrition, rather than being subject to acute deprotection by the removal of TRF2. This is an important distinction since telomeres lacking TRF2 are repaired by c-NHEJ whereas other DNA repair pathways are active at naturally eroded telomeres \cite{Capper2007-oa,Smogorzewska2002-ki,Letsolo2010-bk}. Furthermore, the number of dicentric chromosomes in our system is low (generally 1-2 per metaphase spread) which is similar to the frequency observed in other natural telomere crisis systems \cite{Counter1992-yg}. Apart from the abrogation of the Rb/p21 pathways, which is considered likely to occur before telomere crisis in vivo \cite{Shay1991-rt,DAdda_di_Fagagna2003-oa,Brown1997-qr,Jacobs2004-zn}, these cells contain intact DNA repair pathways, and we make no assumptions as to what the predominant repair mechanisms will be in this context. Further, the relatively weak telomerase activity that can only sustain the shortest telomeres within the population is similar to what occurs in cancer, since many tumors maintain very short telomeres despite activation of telomerase \cite{Furugori2000-vn,De_Lange1990-mz,Mehle1994-jj,Barthel2017-va}.

% This system revealed striking convergent evolution of rearrangements on chromosome 12p, for which we consider the most plausible explanation to be a short telomere on one of the 12p alleles driving early chromosome fusions in telomere crisis. Although the rearrangement events on 12p are likely specific to this cell line, we can draw valid conclusions about the consequence of short telomeres across other systems. It seems likely that the first events during telomere crisis are driven by the shortest telomere(s) within a cell population. The data suggest a minimal set of events that can occur as a result of a single deprotected telomere. We document clean patterns of BFB-like events that represent progressive stages in the evolution of more complex genome architectures. This data has provided an important snapshot into the events that occur during a relatively short time period of telomere crisis. The comparatively flat genomes in the majority of post-crisis clones suggest that the consequences of telomere crisis do not have to be spectacular. It may be that in this system there is selection against complex events involving multiple chromosomes. The more complex events that can be observed in the immediate aftermath of dicentric chromosome resolution \cite{Umbreit2020-kr} may not lead to viable post-crisis clones. Our data also point to a surprising role for acrocentric chromosomes in stabilizing fusion events, which has also been suggested and observed in other studies \cite{Umbreit2020-kr,Stimpson2010-uu}. Since it was technically challenging to resolve these stabilizing events directly through assembly or mapping of short reads, it is possible that these types of events have been overlooked in large-scale WGS analyses and could be an important hallmark of post-crisis genomes.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Chapter 5: Whole genome analysis of RPA- lung adenocarcinoma
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter{Whole-genome characterization of lung adenocarcinomas lacking alterations in the RTK/RAS/RAF pathway *} \label{chap:gdan}

In this final chapter, I describe the collaborative effort within the Cancer Genome Atlas (TCGA) consortium to characterize the whole genomes of lung adenocarcinoma lacking canonical pathogenic alterations in RTK/RAS/RAF pathway as determined by whole exome sequencing (WES) and transcriptome sequencing (RNA-seq).

\textit{\textbf{Individual contributions:}} The Cancer Genome Atlas consortium conceptualized, initiated the project, and collected data. Jian Carrot-Zhang, Nicolas Robine, and Marcin Imielinski, led the parts of the study shown in here. Matthew Meyerson, Ramaswamy Govindan, and Marcin Imielinski oversaw the project and the final manuscript. I executed the parts of the whole genome analysis presented here and prepared the results within the manuscript \cite{Carrot-Zhang2020-vl}.

\footnotetext[1]{Carrot-Zhang, J. et al. Whole-genome characterization of lung adenocarcinomas lacking alterations in the RTK/RAS/RAF pathway. Cell Rep. 34, 108784 (2021)}

\section{Introduction}
As one of the most common and deadliest cancer \cite{Siegel2019-gb}, lung adenocarcinoma has been known to be driven by genetic alterations (e.g. mutations, fusions, amplifications, deletions) in receptor tyrosine kinases (RTK) and its downstream RAS/RAF/MEK cascade proteins, that eventually activate MAPK signaling pathway \cite{Desai2014-qe}. Previously, a series of TCGA studies estimated 70-80\% of LUAD cases harbor drivers in the RTK/RAS/RAF pathway using WES and transcriptome sequencing (RNA-seq) \cite{Campbell2016-xv,Cancer_Genome_Atlas_Research_Network2014-ju,Imielinski2012-vv}. Based on these knowledge, many small molecule inhibitors have been developed to target various proteins along this pathway including \textit{EGFR}, \textit{BRAF}, \textit{ALK}, \textit{RET}, \textit{ROS1}, \textit{KRAS}, and provide unprecedented clinical benefit for LUAD patients \cite{Zhong2021-st}. Thus, it is crucial to design therapeutic regime based on accurate information about if and what RTK/RAS/RAF pathway alterations (RPA) are present and likely functional within a tumor.


Apparently, the remaining 20-30\% of cases, which we refer to as RTK/RAS/RAF pathway alteration-negative or RPA(-) LUADs, pose a major clinical challenge in precision thoracic oncology \cite{Campbell2016-xv}. One immediate question is whether RPA(-) LUAD driven by some distinct RTK/RAS/RAF-independent pathways, or whether the profiling technology (WES and RNA-seq) missed relevant events due to technical factors like uneven coverage or low sample purities. 

Whole genome sequencing offers a chance to gain a better picture of the variants within these tumors. We anticipate the addition of WGS data to at least 1) either verify the RPA(-) status or recover previously missed known RPAs, 2) nominate potential SV and non-coding drivers hard to capture with WES, 3) determine the extent and type of genomic instability to compare to RPA(+) LUADs. Our genome graph paradigm offers a unique advantage to not only enhance the accuracy of identifying candidate alterations, but also reveal their underlying mutational mechanism. To that end, we performed WGS on LUAD samples from The Cancer Genome Atlas (TCGA) cohort that had appeared in previous analyses to lack an RTK/RAS/RAF pathway-activating alteration \cite{Campbell2016-xv} and comprehensively characterized their variant landscape in Chapter \ref{chap:gdan}.

% Nonetheless, the primary and ultimate goal of cancer genomics is to improve the clinical managements of the disease. This means identifying the actionable features of cancers from sequencing data, including driver mutations in or amplification of (proto)oncogenes, loss-of-function mutations or deletion of tumor suppressor genes, fusion genes, so that patients can be paired with therapies specifically targeting these vulnerabilities of their tumor to achieve better effectiveness or lower toxicity. In short, this is the core promise of precision oncology.
% Naturally, the very first step in this process is to know upon which features (e.g. mutation, SCNA, fusion) the tumors depend. As theorized in Lawrence et al.
% with identified recurrent somatic variants, transcriptomic or proteomic subtypes, aberrant signaling pathways. 
% After we approach the full capacity of WES on defining driver events in cancer cohorts, there are usually 
% and Concluding Remarks \ref{chap:conclude}, we shall discuss about how our genome graph paradigm achieves a more comprehensive characterization of whole genomes of clinically relevant cancer samples.


\clearpage
\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_pie.png}
    \caption{\textbf{Driver alterations composition in 501 lung adenocarcinoma of TCGA.}}{About 23\% of TCGA LUAD cases have not been associated with an known alteration in the RTK/RAS/RAF pathway.}    
    \label{fig:gdan_pie}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_cohort.png}
    \caption{\textbf{Classification of RPA(+) and RPA(-) LUADs.}}{From the entire LUAD cohort, 383 RPA(+)\textsubscript{E} and 118 RPA(-)\textsubscript{E} samples are identified based on WES and RNA-seq data, and 85/118 RPA(-)\textsubscript{E} samples are sent for WGS. Our WGS analyses re-classify 28/85 RPA(-)\textsubscript{E} cases into RPA(+)\textsubscript{G}. Among the 383 RPA(+)\textsubscript{E} cases, 40 cases have WGS data from a previously published TCGA study (Imielinski et al. 2017). Together, we use 57 RPA(-)\textsubscript{G} cases, 68 RPA(+)\textsubscript{G} cases and a total of 411 RPA(+) cases for downstream analyses.}
    \label{fig:gdan_cohort}
\end{figure}

\clearpage
\begin{figure}[ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{RPA identified in WGS of the 85 RPA(-)\textsubscript{E} samples.}
        \label{fig:gdan_rpa_oncoprint}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/gdan_rpa_oncoprint.png}
    \end{adjustbox}
\end{figure}
\clearpage

\section{Identification of RPA(-) LUADs}


% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_rpa_oncoprint.png}
%     \caption{\textbf{RPA identified in WGS of the 85 RPA(-)\textsubscript{E} samples.}}{25/85 show at least one RPA while the remaining 57 are classified as RPA(-)\textsubscript{G}.}
%     \label{fig:gdan_rpa_oncoprint}
% \end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_wes_missed_kras.png}
%     \caption{\textbf{Read alignment at \textit{KRAS} p.G12C mutation found in WGS missed by WES.}}{}
%     \label{fig:gdan_wes_missed_kras}
% \end{figure}
A previously published TCGA study of lung adenocarcinoma \cite{Campbell2016-xv} identified 383 of 501 (76\%) of LUAD cases as RTK/RAS/RAF-alteration-positive, or RPA(+), using WES and RNA-seq (data not included, refer to \cite{Carrot-Zhang2020-vl} Table S1). Samples with an activating mutations in \textit{KRAS}, \textit{EGFR}, \textit{BRAF}, \textit{ERBB2}, \textit{MET}, \textit{RIT1}, \textit{NRAS}, \textit{RAF1}, \textit{HRAS}, \textit{ARAF}, \textit{MAP2K1} or \textit{SOS1}; loss of function mutations in \textit{NF1} or \textit{RASA1}; fusions in \textit{ALK}, \textit{ROS1}, \textit{RET}, \textit{MET} or \textit{NTRK2}; and amplification of \textit{EGFR}, \textit{ERBB2}, \textit{KRAS}, \textit{MET}, \textit{FGFR1} or \textit{MAPK1} were classified as RPA(+) by WES and RNA-seq, which we henceforth designate as RPA(+)\textsubscript{E}. Among the remaining 118 LUADs, we performed WGS for 85 tumor-normal pairs (Figure \ref{fig:gdan_pie}, \ref{fig:gdan_rpa_oncoprint}, \ref{fig:gdan_cohort}) at average 73.5-fold (Â± 7.9 s.d.) and 37.0-fold (Â± 5.8 s.d.) coverage for tumor and matched normal samples, respectively, followed by somatic single-nucleotide variant (SNV), indel, copy number, and structural variant (SV) analysis (Appendix \ref{app:d}).

Our multi-step analysis schema is shown in Figure \ref{fig:gdan_cohort}. In the first step, we re-analyzed the 85 RPA(-)\textsubscript{E} cases to determine if there was truly no evidence of coding mutations in the RTK/RAS/RAF pathway. Surprisingly, we found that 20 of the 85 cases harbored \textit{KRAS} hotspot mutations, with 8 samples showing the recently targetable p.G12C mutation \cite{Canon2019-aj,Ostrem2013-lt} (Figure \ref{fig:gdan_rpa_oncoprint}). Re-examination of the WES data for those samples confirmed the mutation calls for 16 of the 20 samples, but the read support was insufficient to enable high-confidence variant calling without invoking the WGS information (refer to Figure 1C of \cite{Carrot-Zhang2020-vl}). The poor WES coverage for those \textit{KRAS} mutations was likely due to low capture efficiency \cite{Clark2011-iq} for the first coding exon of \textit{KRAS}, which contains codons 12 and 13. 
% In addition, the samples with KRAS mutations missed by WES showed lower tumor purity than did KRAS-mutated samples identified by WES (Fig. S1C, Table S1), confirming that high sequencing depths were necessary for detecting the mutations reliably.

\clearpage
\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_egfr_amp.png}
    \caption{\textbf{An example of \textit{EGFR} amplification and over-expression by BFB.}}{Top panel, \textit{EGFR} RSEM among RNA-seq of LUAD samples. Bottom panel, purity-adjusted copy number and SV junctions (red lines) support a BFB cycles event underlying the amplification. Lower panels indicate WGS read depth and gene location in the region. CN, copy number.}
    \label{fig:gdan_egfr_amp}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_rasa1_del.png}
    \caption{\textbf{An example of \textit{RASA1} deletion and under-expression.}}{Top panel, \textit{RASA1} RSEM among RNA-seq of LUAD samples. Bottom panel, deletion spanning from exon 21 to the end of the gene (12 kbp).}
    \label{fig:gdan_rasa1_del}
\end{figure}
\clearpage

Among the 85 samples, we also identified eight cases with somatic SV and copy number alterations (SCNAs) in known RTK/RAS/RAF pathway members. Among those alterations, we found complex SVs driving high-level amplification and over-expression of oncogenes including \textit{EGFR} (n=1) and \textit{MAPK1} (n=3) (Figure \ref{fig:gdan_rpa_oncoprint}). Further classification of complex SVs showed the \textit{EGFR} amplification to be driven by a breakage-fusion-bridge cycle (BFB cycles) (Figure \ref{fig:gdan_egfr_amp}). We also found deletions coupled with loss of heterozygosity (LOH) resulting in decreased expression of \textit{RASA1} (n=1) or \textit{NF1} (n=1) (Figure \ref{fig:gdan_rasa1_del}, also refer to \cite{Carrot-Zhang2020-vl} Table S2), both of which negatively regulate RTK/RAS/RAF signaling. The focal deletion (379bp length) in \textit{NF1} affected only a single exon that is not well-represented in WES or RNA-seq, highlighting the advantage of WGS for identification of focal SV events (refer to \cite{Carrot-Zhang2020-vl} Table S2). We also identified one case with \textit{ARAF} amplification and one case with \textit{NRG1} fusion, which are alterations previously shown to activate RTK/RAS/RAF signaling in LUADs \cite{Fernandez-Cuesta2014-wr,Imielinski2014-ve}. Interestingly, we found amplification and over-expression of \textit{SOS1} in one case (TCGA-62-8399, Figure \ref{fig:gdan_rpa_oncoprint}). Although \textit{SOS1} mutations have been shown to activate RTK/RAS/RAF signaling \cite{Cai2019-tw}, the role played in RTK/RAS/RAF activation by the amplification described here is unclear.

Overall, we identified 28 (33\%) additional RPA(+)\textsubscript{G} cases among the 85 RPA(-)\textsubscript{E} that had undergone WGS (Figure \ref{fig:gdan_cohort}). We labeled the remaining 57 cases as RPA(-)\textsubscript{G} LUADs because they lacked an RTK/RAS/RAF pathway alteration identified by WGS (as well as WES and RNA-seq). The remainder of the results reported here focus on that subset.

% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_rpa_neg_oncoprint.png}
%     \caption{\textbf{Alterations outside the conventional RTK/RAS/RAF pathway in RPA(-) samples.}}{}
%     \label{fig:gdan_rpa_neg_oncoprint}
% \end{figure}


\clearpage
\begin{figure}[!ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{Alterations outside the conventional RTK/RAS/RAF pathway in RPA(-) samples.}
        \label{fig:gdan_rpa_neg_oncoprint}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/gdan_rpa_neg_oncoprint.png}
    \end{adjustbox}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_keap1_del.png}
    \caption{\textbf{An example of \textit{KEAP1} deletion and under-expression.}}{Top panel, \textit{KEAP1} RSEM among RNA-seq of LUAD samples. Bottom panel, 3kbp deletion covering the transcription start site.}
    \label{fig:gdan_keap1_del}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_stk11_del.png}
    \caption{\textbf{An example of \textit{STK11} deletion and under-expression.}}{Top panel, \textit{STK11} RSEM among RNA-seq of LUAD samples. Bottom panel, 8kbp deletion covering the transcription start site.}
    \label{fig:gdan_stk11_del}
\end{figure}
\clearpage

\section{Recurrent coding alterations in RPA(-)\textsubscript{G} LUADs}
Having identified the 57 RPA(-)\textsubscript{G} LUAD cases, we sought to define their protein-coding driver alteration landscape. Among the coding sequences of 14,987 known protein-coding genes with a median expression RSEM $\ge$ 1 in all of the TCGA LUAD tumors, we analyzed protein-altering indels and SNVs across 57 RPA(-)\textsubscript{G} LUADs to identify recurrently mutated genes. The algorithm used employs a gamma-Poisson regression background model \cite{Imielinski2017-nt} correcting for known covariates of LUAD mutation density (e.g. chromatin state, replication timing, GC content, see Appendix \ref{app:d}). We found four tumor suppressor genes, \textit{TP53}, \textit{STK11}, \textit{KEAP1} and \textit{SMARCA4} significantly mutated in the RPA(-)\textsubscript{G} samples (FDR<0.1, Figure \ref{fig:gdan_rpa_neg_oncoprint}). GISTIC analysis \cite{Mermel2011-gt} of WGS-derived single copy number alterations (SCNAs) identified significantly deleted regions harboring \textit{SETD2} and significantly amplified regions harboring \textit{NKX2-1}, \textit{KAT6A}, \textit{CCNE1}, \textit{MDM2}, \textit{MYC}, \textit{MCL1}, and \textit{MYCL} (FDR<0.1, Figure \ref{fig:gdan_rpa_neg_oncoprint}). Of note, we did not identify novel genes that were significantly mutated or amplified/deleted in the RTK/RAS/RAF pathway; in other words, we were unable to identify new putative driver alterations in an obvious candidate member gene of the RTK/RAS/RAF pathway, possibly owing to the limited sample size.


% Identification of focal deletions within the gene bodies of STK11, SMARCA4, and TP53 was reinforced by concordant exon-skipping junctions in the corresponding RNA-seq data, suggesting that those transcriptomic variants were driven by DNA alterations and not by alternative splicing (Fig. S2A-C). 

Our SV analysis (see Appendix \ref{app:d}) identified simple, focal deletions (<100kbp) targeting \textit{STK11} (Figure \ref{fig:gdan_stk11_del}) and \textit{KEAP1} (Figure \ref{fig:gdan_keap1_del}), coupled with LOH, leading to the decreased expression of these genes (refer to \cite{Carrot-Zhang2020-vl} Table S2). In two cases (TCGA-86-7711 and TCGA-50-6592), the focal deletions targeted the promoter/transcription start site of \textit{KEAP1} without altering the coding regions but resulted in reduced expression of \textit{KEAP1} (Figure \ref{fig:gdan_keap1_del}, also refer to \cite{Carrot-Zhang2020-vl} Table S2). Alterations in \textit{STK11} in KRAS-driven LUADs have been shown to be associated with immune exclusion and a poor response to immunotherapies \cite{Skoulidis2018-be}. We found that loss-of-function events in \textit{STK11} to be anti-correlated with the computationally estimated fraction of leukocytes in the tumor (refer to \cite{Carrot-Zhang2020-vl} Figure S2D). % We then combined the 28 additional RPA(+)\textsubscript{G} cases identified from our cohort, with 40 RPA(+)\textsubscript{G} cases that have WGS data from a previously published TCGA study (Imielinski et al., 2017) (Fig. S1A). We found STK11 focal deletions in 4/68 RPA(+)\textsubscript{G} samples, but found no significant difference in STK11 deletion frequency between RPA(+)\textsubscript{G} and RPA(-)\textsubscript{G} (P=0.22, Fisherâs exact test), suggesting that this event may be equally prevalent in both LUAD types.

As opposed to 411 RPA(+) cases from the full TCGA LUAD cohort (including the 28 cases rescued from the RPA(-)\textsubscript{E} category by our WGS analysis), we found a significant enrichment of \textit{TP53} mutations ($P=5.5 \times 10^{-4}$, OR=2.97, Fisher's exact test), \textit{KEAP1} mutations ($P=4.3 \times 10^{-4}$, OR=3.06) and \textit{SMARCA4} ($P=3.4 \times 10^{-4}$, OR=4.16) in the RPA(-)\textsubscript{G} cases (Figure \ref{fig:gdan_diff_alteration}) for genes listed in Figure \ref{fig:gdan_rpa_neg_oncoprint}. When expanding the analysis to 239 COSMIC cancer genes, we found additional enrichment of mutations in \textit{NRG1} (Figure \ref{fig:gdan_nrg1_muts}, $P=1.2 \times 10^{-5}$, OR=12.0), \textit{ESR1} ($P=4.5 \times 10^{-4}$, OR=11.4), \textit{BLM} ($P=1.1 \times 10^{-3}$, OR=12.4) and \textit{FOXO3} ($P=2.3 \times 10^{-3}$, OR=9.29, Figure \ref{fig:gdan_diff_alteration}, ). The RPA(-)\textsubscript{G} samples also showed significantly higher tumor mutation burden (TMB) in a linear regression model controlled for tumor purity (Figure \ref{fig:gdan_tmb}). We found that recent smokers (last smoking year less than 15) were enriched in the RPA(-)\textsubscript{G} group compared to the RPA(+) group, although mutagen activity associated with tobacco smoking activity (COSMIC SNV signature 4) was not different between the RPA(-)\textsubscript{G} group and RPA(+)\textsubscript{G} group, controlling for tumor purity (Figure \ref{fig:gdan_recent_smoke}). We did not find additional differences between RPA(+)\textsubscript{G} and RPA(-)\textsubscript{G} LUADs in their molecular/clinical features (leukocyte fraction, genome doubling, degree of aneuploidy, age of diagnosis and genetic ancestry \cite{Carrot-Zhang2020-vl}.

\clearpage
\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_diff_alteration.png}
    \caption{\textbf{Differentially altered driver genes in RPA(-) versus RPA(+) samples.}}{P values are obtained from Fisherâs exact test. NS: not significant.}
    \label{fig:gdan_diff_alteration}
\end{figure}

% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_nrg1_muts.png}
%     \caption{\textbf{\textit{NRG1} mutations in RPA(-) versus RPA(+).}}{'-' indicates the sample has only WES data available.}
%     \label{fig:gdan_nrg1_muts}
% \end{figure}

\begin{figure}
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{\textit{NRG1} mutations in RPA(-) versus RPA(+).}
        \label{fig:gdan_nrg1_muts}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/gdan_nrg1_muts.png}
    \end{adjustbox}
\end{figure}
% {'-' indicates the sample has only WES data available.}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_tmb.png}
    \caption{\textbf{Higher TMB in RPA(-) than RPA(+) samples.}}{ P value is calculated from a Wilcoxon rank test. The violin plot reflects kernel density estimations.}
    \label{fig:gdan_tmb}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_recent_smoke.png}
    \caption{\textbf{RPA(-) enriched in patients who smoked within 15 years from diagnosis.}}{P value is obtained from a two-proportions z-test. Error bars represent the 95\% confidence intervals from the two-proportions z-test.}
    \label{fig:gdan_recent_smoke}
\end{figure}
\clearpage



\section{Recurrent non-coding alterations in RPA(-)\textsubscript{G} LUADs }
% TODO!!! add Khurana 2016
We next asked whether the RPA(-)\textsubscript{G} cases harbored recurrent novel SNVs or indels outside of the coding genome. We focused the search on regions nominated by a recent TCGA ATAC-seq (Assay for Transposase-Accessible Chromatin using sequencing) study that identified regions of open chromatin, and required that the region be identified in at least 2 of the 44 LUAD samples subject to ATAC-seq \cite{Corces2018-nl}. Open chromatin regions are associated with active promoters, enhancers, and transcription factor binding sites, and thus may be targets of positive somatic selection \cite{Fu2014-so,Khurana2016-vj}. When we analyzed 60,572 total mutations in 139,841 LUAD-specific open chromatin regions (2.7\% of the genome) using gamma-Poisson regression with known covariates of LUAD passenger mutation density (Appendix \ref{app:d}), we found three loci that were nominally enriched (FDR<0.25) in SNVs or indels, located near \textit{ILF2}, \textit{CUL2}, and \textit{TSN} (Figure \ref{fig:gdan_noncoding_all}). Applying the intuition that expressed and dosage sensitive genes might be targets of non-coding alteration, we examined genes that were consistently expressed across TCGA LUAD (median RSEM>10) and recurrently amplified in RPA(-)\textsubscript{G} samples (see Appendix \ref{app:d}, \cite{Zack:2013f1f}). This analysis yielded \textit{ILF2} as the sole significant peak (FDR<0.1, Figure \ref{fig:gdan_noncoding_amp}).

The promoter region of \textit{ILF2} was mutated in six out of 57 cases (P=$2.7 \times 10^(-6)$, coefficient=2, Gamma Poisson Regression; Figure \ref{fig:gdan_nondocing_ilf2_lolipop}). We used the FunSeq2 method to annotate the sequence motifs bound by transcription factors \cite{Fu2014-so}. All six mutations lay in the âsensitiveâ and âultrasensitiveâ (i.e., highly conserved) regions of the genome \cite{Khurana2013-qc}. One mutation (chr1:153643633, G->A) was predicted to disrupt a \textit{HOXB6} motif and another mutation (chr1:153643690, G->T) was predicted to disrupt an \textit{NR3C1} motif. We did not observe any mutational signature enriched in the six mutations. Moreover, RPA(-)\textsubscript{G} cases harboring \textit{ILF2} promoter mutations showed increased expression of \textit{ILF2}, compared to ILF2-wildtype cases (Figure \ref{fig:gdan_nondocing_ilf2_expr}), or cases harboring other mutations within the +/-10kbp window of the promoter region (Figure \ref{fig:gdan_ilf2_lolipop_wide}, \ref{fig:gdan_ilf2_expr_full}), after controlling for the local copy number of \textit{ILF2} and tumor purity. On the other hand, non-coding mutations near \textit{CUL2} did not affect \textit{CUL2} expression, and intronic mutations in \textit{TSN} showed a non-significant trend towards an increase in \textit{TSN} expression (P=0.066). \textit{ILF2} is located in chr1q21.3, which is frequently amplified in LUADs. In myeloma, increases in \textit{ILF2} expression through amplification have been shown to promote tolerance of genomic instability and drive resistance to DNA damaging therapies, through dysregulation of RNA splicing and DNA damage response pathways \cite{Marchesini2017-ly}. Consistent with that role of \textit{ILF2}, we found \textit{ILF2} expression to be associated with increased SV burden (P=0.01, coefficient=0.9, negative binomial regression).

\clearpage
\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_noncoding_all.png}
    \caption{\textbf{Recurrent non-coding mutations within LUAD-specific ATAC-seq peaks in RPA(-)}}{Three genes with non-coding mutations nominated through recurrence analysis across LUAD-related ATAC peaks. Red dots indicate loci with FDR < 0.25.}
    \label{fig:gdan_noncoding_all}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_noncoding_amp.png}
    \caption{\textbf{Recurrent non-coding mutations within the intersect of LUAD-specific ATAC-seq peaks and LUAD-specific recurrent SCNA peaks in RPA(-)}}{Same as Figure \ref{fig:gdan_noncoding_all}, but restricted to ATAC-seq peaks in genes with RSEM $\ge 10$ across TCGA LUAD and recurrently amplified in RPA(-)\textsubscript{G} samples. Red dots indicate FDR < 0.1.}
    \label{fig:gdan_noncoding_amp}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_noncoding_ilf2_lolipop.png}
    \caption{\textbf{Recurrent promoter mutations near ILF2 gene}}{Among 57 RPA(-)\textsubscript{G} samples, 6 SNVs are observed in the promoter region of \textit{ILF2}; all are located within ATAC-seq peaks.}
    \label{fig:gdan_nondocing_ilf2_lolipop}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_noncoding_ilf2_expr.png}
    \caption{\textbf{Promoter mutations and amplifications both increase \textit{\textit{ILF2}} expression compared to wildtype.}}{P values are calculated from linear regression analysis correlating expression, adjusting for the local copy number of \textit{ILF2} and purity. Boxplot shows median, interquartile range, and 1.5 times the interquartile range.}
    \label{fig:gdan_nondocing_ilf2_expr}
\end{figure}

\begin{figure} 
    \centering
    \includegraphics[width=0.9\linewidth]{figures/600ppi/gdan_ilf2_lolipop_wide.png}
    \caption{\textbf{Mutations around 10kb window from the \textit{ILF2} promoter peak.}}{Nine other SNVs among 57 RPA(-)\textsubscript{G} LUAD samples are observed in the 10kbp+/- window outside of the promoter region of \textit{ILF2}. Promoter mutations are indicated in blue circles. Other mutations are indicated in black circles.}
    \label{fig:gdan_ilf2_lolipop_wide}
\end{figure}

\begin{figure} 
    \centering
    \includegraphics[width=0.9\linewidth]{figures/600ppi/gdan_ilf2_expr_full.png}
    \caption{\textbf{Expression of \textit{ILF2} with respect to alterations.}}
    \label{fig:gdan_ilf2_expr_full}
\end{figure}

\begin{figure} 
    \centering
    \includegraphics[width=0.9\linewidth]{figures/600ppi/gdan_tsn_expr.png}
    \caption{\textbf{Expression of \textit{TSN} with respect to alterations.}}
    \label{fig:gdan_tsn_expr}
\end{figure}

\clearpage
\section{Complex SV patterns in RPA(-)\textsubscript{G} LUADs}
% \begin{figure}
%     \centering
%     \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_sv_heatmap.png}
%     \caption{\textbf{Burden of simple and complex SV events in RPA(-) samples.}}{Lower panel: SVs resulting in copy-number loss (chromothripsis, rigma, chromoplexy, templated insertion chain, simple deletion). Key indicates the range of event count of SV types observed in each sample.}
%     \label{fig:gdan_sv_heatmap}
% \end{figure}
By integrating read depth changes with rearrangement breakpoint locations to generate junction-balanced genome graphs (JBGG, see Subsection \ref{sub:JBGG}), we analyzed genome graphs \cite{Hadi2020-um} of the 57 RPA(-)\textsubscript{G} cases to identify complex patterns of Structural variant. Analysis of subgraphs (https://github.com/mskilab/gGnome) identified multiple instances of complex amplicons (18 double minute, 5 BFB cycles, 5 tyfonas, 52 pyrgo), as well as simple duplications (mean=16.3 per sample, Figure \ref{fig:gdan_sv_heatmap}). Tyfonas are recently identified complex amplicon comprising hundreds of high junction copy number (JCN) and fold-back inversion junctions \cite{Hadi2020-um} that are enriched in cancers such as dedifferentiated liposarcomas and acral melanomas. As an example, we show an amplification of \textit{NKX2-1} driven by tyfonas (Figure \ref{fig:gdan_nkx21_tyf}). Pyrgo, which comprise "towers" of low copy duplication junctions \cite{Hadi2020-um}, were also found to drive the amplification of LUAD loci including \textit{NKX2-1} (Figure \ref{fig:gdan_nkx21_pyrgo}).

Genes located inside double minute, BFB cycles, and tyfonas events were markedly enriched in expression outlier genes ($P < 1 \times 10^{-16}$, Mann-Whitney U test) relative to genes involved in pyrgo and simple duplication events (Figure \ref{fig:gdan_amp_expr}), suggesting that the former events were retained in the cancer cell due to the growth-promoting effects of altered gene expression. Although none of these complex SV types were correlated with \textit{TP53} mutations, which are thought to generate genomic instability, there was a significantly higher incidence of simple deletions observed in the \textit{TP53}-mutant cases ($P=1 \times 10^{-4}$, Mann-Whitney U test, Figure \ref{fig:gdan_del_vs_p53}). That association held true when including the 68 RPA(+)\textsubscript{G} samples and controlling for purity and RPA status ($P=2 \times 10^{-4}$, coefficient=12, linear regression).

Double minutes were the most common complex SV type seen in the RPA(-)\textsubscript{G} cases (12 of 57 samples, Figure \ref{fig:gdan_sv_heatmap}). Like extrachromosomal circular DNA segments, double minutes do not segregate symmetrically; thus, their dosage per cell is exquisitely responsive to selection pressure \cite{Verhaak2019,Wu2019-ap}. As a result, at least one of the genes in any given double minute likely contributes to tumor development. To leverage that intuition, we focused on a relatively small double minute identified in case TCGA-55-5899 (Figure \ref{fig:gdan_dm}). We found that this double minute fused and amplified multiple focal regions on chromosome 13 spanning 1.0 Mbp, and resulted in the high-level gain ($>$10 copies) of three intact genes (\textit{UBL3}, \textit{SOX21}, and \textit{LIG4}). Two of these, \textit{UBL3} and \textit{LIG4}, were over-expressed in TCGA-55-5899 relative to the full LUAD cohort with RNA-seq data (Figure \ref{fig:gdan_dm_expr}). Because we did not observe any genes to be amplified and over-expressed in more than one RPA(-)\textsubscript{G} case (refer to \cite{Carrot-Zhang2020-vl} Table S3), larger numbers of cases would have to be analyzed to gain an understanding of the possible role genes amplified by double minutes play in driving RPA(-) LUADs.

\clearpage
\begin{figure}[!ht]
    \begin{adjustbox}{addcode={\begin{minipage}{\width}}{\caption{
        \textbf{Burden of simple and complex SV events in RPA(-) samples.}
        \label{fig:gdan_sv_heatmap}
        }\end{minipage}},rotate=90,center}
        % \includegraphics[scale=.6]{example-image}%
        \includegraphics[width=\textheight]{figures/600ppi/gdan_sv_heatmap.png}
    \end{adjustbox}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[height=0.75\textheight]{figures/600ppi/gdan_del_vs_p53.png}
    \caption{\textbf{Excess deletion burden associated with \textit{TP53} loss of function alterations.}}{P value is obtained from Mann-Whitney U test. Violin plots reflect kernel density estimations.}
    \label{fig:gdan_del_vs_p53}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_nkx21_tyf.png}
    \caption{\textbf{\textit{NKX2-1} amplification by tyfonas.}}{}
    \label{fig:gdan_nkx21_tyf}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_nkx21_pyrgo.png}
    \caption{\textbf{\textit{NKX2-1} amplification by pyrgo.}}{}
    \label{fig:gdan_nkx21_pyrgo}
\end{figure}
\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_dm.png}
    \caption{\textbf{An example of double minute (DM) amplifying multiple distal loci.}}{}
    \label{fig:gdan_dm}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[height=0.95\textheight]{figures/600ppi/gdan_dm_expr.png}
    \caption{\textbf{Over-expression of genes amplified in the DM.}}
    \label{fig:gdan_dm_expr}
\end{figure}
% {2 of 5 (\textit{UBL3} and \textit{LIG4}) showed marked over-expression relative to RNA-seq data for the full TCGA LUAD cohort.}

\begin{figure}
    \centering
    \includegraphics[width=0.95\linewidth]{figures/600ppi/gdan_amp_expr.png}
    \caption{\textbf{Over-expression of genes amplified by any of the three complex amplification events.}}{}
    \label{fig:gdan_amp_expr}
\end{figure}
\clearpage


\section{Discussion}
Although large-scale genomic studies have shown LUADs to be molecularly heterogeneous, the majority of LUAD cases share the common feature of RTK/RAS/RAF signaling activation through a genetic driver \cite{Desai2014-qe,Swanton2016-lg}. A key question, however, is whether RPA(-) LUADs â which, by definition, lack RTK/RAS/RAF drivers â represent a biologically distinct entity. Our results suggest that they are heterogeneous but that they do share common biological features, including a high frequency of \textit{TP53} mutations and high mutation burden. Those features tend to distinguish RPA(-) LUADs from their RPA(+) counterparts.

A key confounding factor when we try to define RPA(-) LUADs as a distinct entity is the technical limitation of reliably detecting RTK/RAS/RAF pathway genomic lesions in impure tumor samples. Strikingly, 28 of 85 cases in the present study that were previously found negative for any RTK/RAS/RAF lesion by WES and RNA-seq pipelines were subsequently shown by our WGS analysis to harbor a somatic RTK/RAS/RAF driver. The high prevalence of overlooked \textit{KRAS} mutations is explained in part by low tumor purity and/or decreased probe affinity for the GC-rich exons in \textit{KRAS} during WES library preparation \cite{Clark2011-iq}. 

The recent discovery of small molecules with activity against \textit{KRAS} p.G12C LUAD \cite{Canon2019-aj,Ostrem2013-lt} highlights the importance of precise identification of mutations at that locus. However, many of the challenges could potentially be overcome by higher read depth (e.g. $>$40X minimal on-target coverage), which is now routinely achieved by some clinical-grade target capture assays \cite{Goodman2017-wb,Zehir2017-ue}. Nevertheless, the high rate of missed RTK/RAS/RAF lesions in our WGS cohort tells a cautionary tale: false-negative calls should always be considered even when working with high-quality datasets.

Eight of the 28 rescued RPA(+)\textsubscript{G} cases in our WGS cohort harbored cryptic SV lesions (protein-coding fusions, SCNAs), which represent alternative mutational mechanisms for (in)activating RTK/RAS/RAF genes (e.g. \textit{EGFR} BFB cycles, focal deletions of \textit{NF1} and \textit{RASA1}). WGS is naturally adapted to detect such complex or subtle structural alterations \cite{Hadi2020-um}. We leveraged that capability to identify a spectrum of SV patterns among the 57 RPA(-)\textsubscript{G} cases. Notably, 9/85 (11\%) samples in the cohort harbored focal deletions in \textit{STK11} that were undetected by WES. Alterations in genes such as \textit{STK11} and \textit{KEAP1} have come into focus as possible prognostic and/or predictive biomarkers in patients with lung cancer \cite{Arbour2018-qg,Skoulidis2018-be}, inclusion of full genomic capture probe sets for those genes may become necessary in the near future to identify samples with alterations accurately. Double minutes, the most prevalent SV type among complex SVs in the RPA(-)\textsubscript{G} cases have recently been implicated in genomic plasticity, oncogene selection, and chromatin evolution \cite{Verhaak2019}. Further studies analyzing larger cohorts will be valuable for dissecting the role they may play in driving RPA(-) LUAD biology in greater detail.

If the 57 RPA(-)\textsubscript{G} cases identified in this study indeed represent a distinct biological entity of RTK/RAS/RAF-independent LUADs, then what pathway drives them to proliferate? Do RPA(-) LUADs show distinct genetic or therapeutic vulnerabilities? Although our analyses have nominated candidate drivers in LUAD (e.g. \textit{ILF2}), it is unclear how tumors harboring an alteration in such genes would phenocopy the proliferative effect of RTK/RAS/RAF alteration. Perhaps the frequent SV-driven loss of tumor suppressors (e.g. \textit{STK11}) or amplification of genes operating downstream of RTK/RAS/RAF signaling (e.g. \textit{MYC}), which we observed in our RPA(-)\textsubscript{G} LUADs, can cooperate to fill the missing role \cite{Sears2000-ua}. It is also possible that a small subset of RPA(-) tumors are still RTK/RAS/RAF-driven but activate the pathway through epigenetic dysregulation, or genetic alterations in genes that are less frequently altered in LUAD, and therefore not detected by the statistical methods employed in this study. Alternatively, the biology of RPA(-) LUADs may resemble that of other cancer types in which RTK/RAS/RAF alterations are rarely seen and are marked by early \textit{TP53} loss and high TMB; those tumor subtypes may take a different evolutionary path that is less dependent on the sustained proliferative signaling that RTK/RAS/RAF activation provides \cite{Chen2019-bi,Drosten2014-el,Salgueiro2020-xg}. Such a path may accumulate key genetic alterations in a different order than RPA(+) tumors \cite{Lee2019-nm}, begin in an alternate cell of origin, or undergo lineage switching in the course of evolution.

Taken together, our findings suggest that RPA(-) LUAD is likely to represent a heterogeneous entity and that WGS of much larger cohorts of RPA(-) and RPA(+) LUADs would be necessary to fully address the nature of their underlying biology.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Concluding remarks
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter{Concluding remarks} \label{chap:conclude}
% Marcin's comments
% Marcin Imielinski: we've demonstrated that the genome graph paradigm can uncover new classes of SV events, provide insight into mutational processes like telomere crisis, and characterize cancers with no known targetable alterations
% future directions where this paradigm can be applied and/or even extended include
% Start with the big pictures
Merely over a decade young, genomic sequencing has already revolutionized our approach to study cancer and will surely continue to grow, fueled by the ever decreasing sequencing cost and the pressing need to optimize clinical benefits \cite{Stratton2009-ty}. The study of somatic variant patterns has delivered ample successes in elucidating genome maintenance mechanisms and translating to clinical utility. The extraordinary complexity of SVs in cancer genomes pose the next major analytical challenge. In this thesis, I have shown the definition, implementation, and inference of genome graphs for analyzing complex SVs. I have used it to categorize complex amplicon, stratify pan-cancer patients, and reconstruct exact SV evolution in cells surviving telomere crisis. I have also demonstrated its utility in whole genome characterization of cancers without known driver alterations. As discussed in each chapter, all these parts have room for improvement, but the current results still suffice to show the potential of such a general, flexible, and powerful analysis paradigm. Looking beyond these results, in this concluding chapter, I draw up 5 future areas in which this genome graph paradigm will make great impact: 1) systematic discovery of SV signatures, 2) resolving complex haplotypes by integrating emerging long-molecule profiling technologies, 3) dissecting the intra-tumoral heterogeneity of SVs, 4) reconstructing the evolutionary trajectory of SV mutagenesis, and 5) eventually translating to applications in clinical sequencing.

% systematic discovery of new SV classes and signatures --> here describe possible directions in 
% -- de novo motif discovery, and also 
% -- applying NMF / LDA approach to count matrices
% -- Integrating and SV and SNV events
% -- most fundamentally the question is are there more classes of events to be found? 
% -- where could these exist, for example pediatric cancers seem to have complex SV

\section{Systematic discovery of SV classes and signatures}
Mutation signatures have been instrumental in today's cancer genomics research as they provide an interpretable summary of the footprints of all mutagenesis pathways active in the life history of the tumor sample. As we have discussed in the introduction (\ref{sec:philo}), the extraction of mutation signatures can be seen as finding latent combinations of distinct variant classes from an observed count matrix. 

While there are exciting new algorithms that can better extract signatures given such a count matrix, especially the ones that balance the abundant SNVs and rarer SVs \cite{Funnell2019-ti} and ones that integrate genomic covariates \cite{Vohringer2021-bv}, the embedding of SVs into a system classification is still the utmost limiting step, due to their intrinsic complexity. Such an ideal system would cover the complete spectrum of SVs and have clear distinction between variant classes. To achieve this, we must ask the fundamental question: are there more distinct classes of SVs? It spawns at least two major inquiries: 1) developing methods to nominate SV classes from existing WGS data, and 2) profile new samples that would most likely reveal new patterns.

% In Chapter \ref{chap:complex_events} and our publication \cite{Hadi2020-um}, using our genome graph paradigm, we attempted to classify some clusters of junctions within cancer genome graphs and defined 13 types of simple to complex SV events. However, a considerable fraction of junctions are not definitively classified. 

Thus far there have been two systematic methods to codify complex SV patterns. The \textit{top-down}, like I have demonstrated in Chapter \ref{chap:complex_events} and \cite{Hadi2020-um}, takes whole genome graphs, breaks down to clustered subgraphs, extracts various relevant summary statistics from candidate subgraphs, and let the groups of similar recurrent patterns emerge unsupervisedly. On the other hand, the \textit{bottom-up} approach like described in \cite{Li2020-ds} enumerates all possible configurations of junctions and copy number resulting from a given number of junctions, then search against the observations in pan-cancer whole genomes to quantify their abundances. They both have advantages and limitations of their own.

The top-down approach is hollistic and allow similar yet not exactly isomorphic subgraph candidates to group together, but rely on careful choice of summary statistics based on human knowledge and are labor intensive. The bottom-up approach is precise and exhaustive, but the possible motifs grows super-exponentially with the number of junctions, which is why \cite{Li2020-ds} only derive up to 5 junction patterns. However, more complex patterns are prevalent and not all can be broken down to the current enmeration. Plus, as we have explored in \cite{Behr2021-gf}, there are blindspots of short-read sequencing where a junction breakend cannot be mapped, and these loose ends require the method to consider incomplete matching of patterns.

Generalizing the two, if we see each rearranged genome as a graph, the task of identifying 

% , their successes in SV signature discovery rely on answers to two more fundamental questions, namely: 1) to expand and refine the vocabulary of variant classes, and 2) to find new patterns in new samples.

In terms of new samples, it may seem granted the number of WGS-profiled tumors keep growing in the near future. The completion of PCAWG marked an important milestone of whole genome profiling of primary cancers \cite{pcawg_marker2020-yi}, that for the first time, all variants including SVs from various cancer types are harmonized and available for a systematic analysis. More recently, there has been a study of substitution mutation signatures in a cohort of more than 12000 prospectively collected new pan-cancer WGS samples \cite{Degasperi2022-am}, the largest to date. Yet, there is plenty to consider on what type of samples to sequence next, with what type of technology (discussed in the next sections), to optimize the gain of insights and clinical benefit.

First of all, metastatic cancers are the leading cause of deaths in patients, and can rise out of therapies and surgeries. The Hartwig Medical Foundation profiled the whole genomes of 2520 solid metastatic tumors and did not find striking gene that is specifically associated with metastasis not primary cancers. find much higher whole genome doubling rate than reference primary tumors plus wide-spread CNAs \cite{Priestley2019-bw}. Moreover, previous WGS studies and experimental systems have pointed to complex amplifications as a prevalent mechanism to sustain mitotic growth or gain therapeutic resistance \cite{Watkins2020-qp,Viswanathan:2018hi,Van_Dessel2019-gi,Shoshani2020-zj}. For instance, in one study \cite{Van_Dessel2019-gi}, more than half of metastatic castration-resistant prostate carcinomas harbor high CN amplification of androgen receptor gene (\textit{AR}). In a experimental system, Shoshani et al. \cite{Shoshani2020-zj} found HeLa cells respond to increasingly aggressive methotrexate treatment by shattering and forming extrachromosomal DNA at the previously mildly amplified homogeneous staining regions covering the resistence gene \textit{DHFR}. These findings

Secondly, pediatric tumors have shown very distinct driver landscape from adult tumors \cite{Grobner2018-qp} and numerous reports have pointed to complex SV events producing putative drivers like \textit{MYCN} amplification and \textit{PRDM6} enhancer hijacking \cite{Northcott2017-xm}. 

Last but not least, beyond tumor tissues from biopsy, germline, other pathological tissues, and even healthy tissue are also informative to the study of mutational processes of SVs. In the germline genome of autism spectrum disorder and other developmental abnormalities various \textit{de novo} complex SV patterns have been categorized including chromothripsis \cite{Collins2017-uc}. Meanwhile, somatic mosiacism in various normal tissues have been characterized and surprinsingly cancer driver alterations are common even in healthy tissues \cite{Martincorena:2018fk,Sekar2020-cv}. Such studies are still very challenging owning to the low variant allele frequencies in normal tissue, low sequencing depth, and the relatively high error rate of single cell DNA sequencing. Hence most pioneering studies are focused on SNVs and INDELs, yet detecting traces of junctions and CNAs is apparently the next step.
%In , there have also been data indicating the presence of CNAs, even chromothripsis \cite{McDermott2015-xc}.

% the distillation of signatures of mutagenesis pathways out of a count matrix of events in a sample cohort requires a non-overlapping and complete (or at least approximately) taxonomy of classes of variant patterns. Due to the inadequate representation of complexity, current efforts to extract SV signtures are limited to features of junction breake+nds and not 


% more patterns, more mechanisms
% (Do I keep this or no?)The algorithmic characterization of SV patterns is far from a defined question. There are still very heterogeneous entities attached to the same terminology, while similar processes described from different angles. This current genome graph-based SV taxonomy is not complete. Instead, it is exposing the conceptual difficulties in improving the systematic definition of SV patterns and laying a common language for the community to further communicate and compare results. Nevertheless, in our paper \cite{Hadi2020-um}, we show genome graph-defined tyfonas reveal more striking specificity in liposarcoma and acral melanoma than the broader definition of \textit{amplified chromothripsis} by the PCAWG consortium \cite{pcawg_marker2020-yi}, indicating the genome graph paradigm enabled a more textured classification. Regardless of approaches, a unified framework where different definitions can be compared is sorely needed, and I believe our theories and implementations is a viable option for this purpose.

% Our current definitions involve a considerable amount of human judgment based on the understanding of existing literature, reasoning of proposed mechanisms, and manual verification in real data. 
% Thus far, most if not all discovered complex SV patterns involve a considerable amount of manual deduction and rather customized and simplified simulation \cite{stephens2011,baca2013}. A more unbiased, perhaps data-driven, method would be one able to recognize recurrent themes from genome graphs and let the classes of patterns emerge unsupervised. For instance, similar tasks have been a hot topic in 



% haplotype structure from long molecules- directly building genome graph from long read / long molecule data and modeling haplotypes instead of junctions
% from pattern to mechanism and back - algorithms to simulate patterns a la greenman or bafna or others
% (edited)
% extrachromosomal vs chromosomal question - validating this on a large scale ie whether tyfonas are always integrated and in general addressing this question with short reads, long reads, or other tools
\section{Resolving complex haplotypes with long-range sequencing and mapping}

% The future of genome graph reconstruction, 1) allelic phased, 2) multiple related samples and automated phylogenetic inference, 3) subclonality of SV events, 4) unify germline SVs and interface pangenome graphs.
Recently, long-molecule profiling has become the frontier in characterizing SVs, for obvious superiority over short-read data which often fall short of uniquely mapping breakends in repetitive regions \cite{Behr2021-gf}. With longer reads, most of the community expect better precision and even better recall in identifying the junctions in a sample \cite{Zhao2021-bi}. While this is true, we argue that the main strength of long-molecule profiling is to phase and resolve the order of multiple junctions that constitute the derivative sequence by a complex SV event \cite{Behr2021-gf,Hadi2020-um,Luebeck2020-ow,Aganezov2020-cu}. The recent achievement of a complete human reference genome sequence by assembling multi-platform long-molecules \cite{Nurk2021-ro} has shown the possibility of obtaining the exact sequences in highly repetitive regions like centromeres. Furthermore, by combining long-read sequencing with single-cell template strand sequencing (Strand-seq), Ebert et al. have reported a haplotype-resolved human germline structural variants and small variants in 35 individuals \cite{Ebert2021-hl}. In the meantime, new algorithms for assembling haplotype-resolved genomes like Hifiasm \cite{Cheng2021-nh} and for subsequently compiling them into pangenome graphs like Minigraph \cite{Li2020-ai}, show that we are approaching a new era to directly assemble the derivative sequences altered by SVs.

Nonetheless, the current long-read technologies are still progressing towards affordability, longer-ranges, and higher throughput. The cost of any long-molecule profiling is still not on par with the already widely used short-read sequencing, making it harder to scale up and achieve the higher number of samples that SV analyses require. In terms of the length of the profiled molecule, both mainstream long-read sequencing technologies, by Pacific Biosciences and Oxford Nanopore, are still in the median range of tens of kilobases, so that fully resolving long-range complex structures drive the cost even higher. Optical mapping such as Bionano can achieve about 200 kilobases, more suitable for resolving the complex events, yet it is not sequencing and cannot provide base-level resolution. Due to lower number of reads profiled, the coverage data from long-molecule technologies is not as robust for copy number analysis as short-read counterparts. There have been creative sample preparation and algorithmic enhancements \cite{Baslan2020-hz,Prabakar2019-dw} for copy number inference on long-read sequencing platforms, but they have to be adapted to sequence shorter fragments, practically abandoning the particular strength of long-reads. In sum, at least in the near term, using long-read to resolve the structure of complex SVs still requires integration with short-read WGS.

To that end, our genome graph paradigm has built a native way to incorporate the unphased yet more accurate JBGGs from short-read sequencing, with the long-molecule data to reconstruct optimal haplotypes, exemplified by Figure 3B and 6E in \cite{Hadi2020-um}. AmpliconReconstructor attempts at the same task but only for amplified regions \cite{Luebeck2020-ow}. RCK is another tools that aims to achieve this \cite{Aganezov2020-cu}. The comparison of performance is left for future work, but our tools are immediately applicable to such benchmarks.


% 11:52 AMMarcin Imielinski
% subclonal SV evolution in heterogenous samples - can describe possible algorithmic improvements to relax integer or do multi-sample analysis, and then single cell genome graphs
\section{Uncover SV heterogeneity with multi-sample or single cell WGS}
Genetic heterogeneity is a prerequisite for evolution and subclonal SVs have been extensively documented within various tumors \cite{Watkins2020-qp,Paulson2021-nd}. For instance, Watkins et al. \cite{Watkins2020-qp} characterized intra-tumor copy number heterogeneity from multi-sample, longitudinal biopsies of 394 tumors across 22 tumor types, indicating the continuous selection of CNAs throughout evolution.

In terms of studied materials, multiple spatio-temporal sampling study design is gaining popularity as it taps into the change of SV patterns in related lineages. For example, in \cite{Paulson2021-nd} we sampled Barrett's esophagus tissues from a prospective patient cohort at two or three time points during the course of disease, from upper and lower esophagus separately. We found abundant rigma events in both the patients that eventually developed esophageal carcinoma and those who did not, at both time points, indicating that rigma is a ubiquitously active mutagenesis process in esophagus regardless of cancer outcome \cite{Paulson2021-nd}. Yet, the presence of BFB cycles is significantly associated with cancer outcome, implying that the activation of a mutational process that generates BFB cycles triggers the malignant transformation. In another study, 



Promising even greater resolution, single-cell WGS is providing a lens into the intra-tumoral heterogeneity of SVs \cite{Zaccaria2020-bn,laks2019,Salehi2021-xi}. Zaccaria and Raphael proposed an innovative method CHISEL to infer allelic-specific copy numbers by aggregating weak signals of SNPs . In general, the analysis of single cell profiling will assign each cell a cluster based on the lower-resolution CN segmentation, and then aggregate the reads in each cluster to form \textit{pseudo-bulk} that represent a homogeneous clone. Our approach applied to the isogenic clones in the post-crisis MRC5 cells are immediately compatible with these data. We expect the combination of the two to yield unprecedented clarity to the dynamic of SV evolution.

\section{From patterns to mechanisms and back: the trajectory of SV evolution}

The other side of the problem to SV pattern identification in real genomes is the bottom-up theory of the genesis of SVs cares for the abstraction of. Such theories originates with the goal to infer ancestral genome organization from multi-species synteny data \cite{Medvedev:2010bm}. Greenman et al. set out on a series of papers to rigorously define the features of the outcomes given some known operations on the genomic sequence, for example BFB cycles \cite{greenman2016-ho}.  These are fundamental conceptual constructions that will be extremely useful in generalizing to real patterns which are often way more complex. Our genome graph paradigm provides a genericb framework to simulate \textit{de novo} sequences of accumulating SVs. We have used its results in our benchmarking of JaBbA Section \ref{sub:jabba_benchmarking}. Admittedly, there are many patterns in the result that look biologically improbable, e.g. large portions of loss among many chromosomes, but it proves the feasibility and value doing simulations by incrementally adding more biologically meaningful processes and constraints.


% clinical WGS - can these approaches be used to enhance clinical interpretation and how
\section{SVs as biomarkers in clinical sequencing}

% Limited by the sample size, we do not have the statistical power to detect new significantly recurrent alterations that is lacking in the RPA(+) samples, but we observe biologically important genes being amplified in likely eccDNA forms that result in their over-expression, which is usually considered a result of positive selection \cite{Lange2021-if}. Due to the exponentially larger search space for a pair of loci recurrently juxtaposed by SVs, it will be challenging to keep the conventional statistical models developed for SNVs and INDELs \cite{Rheinbay2020-no}. By integrating the SV pattern recognition and modeling of the dynamics of mutational processes \cite{Lange2021-if}, we can discover a whole new domain of cancer drivers resulted from rearrangements of the non-coding genome \cite{Weischenfeldt2017-jr,Helmsauer2020-do}.
% At this crossroad of exhausting recurrence signal to find cancer drivers, it is increasingly important to analyze genome structures, mutational processes, and getting creative ways to nominate new hypotheses.
WGS is becoming recognized as a powerful alternative to the more popular and thus far more cost-effective capture-based platforms \cite{Meggendorfer2021-wo}. Not only it uncovers all classes of known drivers at a pristine accuracy, the variant patterns derived from vastly more abundant passenger variants can indicate the vulnerability of the tumor. They can serve as a new kind of biomarkers for DNA repair defects or immunogenecity.



% Marcin Imielinski: we've demonstrated that the genome graph paradigm can uncover new classes of SV events, provide insight into mutational processes like telomere crisis, and characterize cancers with no known targetable alterations
% future directions where this paradigm can be applied and/or even extended include
% (edited)
% systematic discovery of new SV classes and signatures --> here describe possible directions in de novo motif discovery, and also applying NMF / LDA approach to count matrices, Integrating and SV and SNV events, most fundamentally the question is are there more classes of events to be found? where could these exist, for example pediatric cancers seem to have complex SV
% 11:52 AMMarcin Imielinski
% subclonal SV evolution in heterogenous samples - can describe possible algorithmic improvements to relax integer or do multi-sample analysis, and then single cell genome graphs
% (edited)
% haplotype structure from long molecules- directly building genome graph from long read / long molecule data and modeling haplotypes instead of junctions
% from pattern to mechanism and back - algorithms to simulate patterns a la greenman or bafna or others
% (edited)
% extrachromosomal vs chromosomal question - validating this on a large scale ie whether tyfonas are always integrated and in general addressing this question with short reads, long reads, or other tools
% clinical WGS - can these approaches be used to enhance clinical interpretation and how














% Like we demonstrated with MRC5 telomere crisis model, a well-controlled \textit{in vitro} system can be instrumental to precisely define the structural outcomes of a mutational process, but it relies on suitable computational frameworks to describe the nuanced evolutionary procedure of the processes.Applying the genome graph paradigm to these data 

% Finally, our genome graph paradigm provides practical tools to smoothly integrate into whole genome analysis of cancers. We show our characterization of lung adenocarcinoma lacking alterations in RTK/RAS/RAF pathway as an example. Some deletions of the TSS of tumor suppressors we captured are very focal and subtle, that is hard to identify by WES platform, yet they can greatly reduce the expression of affected genes. We were able to see them thanks to the increased sensitivity of SVs by combining the signals from both junctions and coverage, demonstrating the technical advantage of JaBbA over decouple approaches.




\appendix

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Appendix A
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter*{Appendix A: Chapter 2 supplementary materials and methods} \label{app:a}
\subsection*{Genome simulations}
Simulated sequencing of rearranged tumor samples for benchmarking were first constructed as haplotype-specific genomic sequences in \ttt{.fasta} format, then read sampling, alignment, and finally mixed with normal reads to a series of purity levels. In the first step we generated a set of \textit{de novo}, forward simulated, rearranged cancer genomic sequences from an initial set of input junctions (\texttt{SimBLE}, \url{https://github.com/mskilab/sim.ble}). SimBLE iterates through simulated cell cycles to gradually incorporate the input junctions into the derived genome from previous steps until exhausting the input junction set, while keeping track of the actual rearranged haplotypes. As a result, it generates a coherent sequence of the rearranged genome guided by the haplotypes encoded in the reference genomic ranges. We then simulated sequencing reads from this FASTA file with ART read simulator \cite{Huang2012-zn} to an average depth of 40X and aligned them to the reference genome hg19 to obtain the simulated BAMs. Trivially, the reference genome itself is also subjected to the same \textit{in silico} sequencing to provide as normal controls. We did 40 distinct simulations with different random subsets of somatic junctions (from 5 to 333 total) identified in the HCC1143 breast cancer cell line. 

In addition to these \textit{de novo} simulated BAMs, we also obtained WGS for HCC1954 breast cancer cell line and HCC1954BL the corresponding normal fibroblast cell line.  To simulate stromal admixture, we combined tumor and normal simulated (or HCC1954) BAM files, then downsampled, and mixed tumor and normal reads across ten tumor DNA proportions from 0.1 to 1.0. We created four technical replicates for each of these ten purity levels, which yielded 40 pairs of tumor and normal BAM files for the 40 distinct simulated genomes and HCC1954, respectively.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Appendix B
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter*{Appendix B: Chapter 3 supplementary materials and methods} \label{app:b}

\subsection*{Structural variant event classification}
To identify simple and complex structural variant events in the genome graph output of JaBbA, we implemented a set of classifiers.  These are implemented in the $\ttt{events}$ function in the \texttt{gGnome} package (freely available from: \url{https://github.com/mskilab/gGnome}), which calls the classifier functions below. Procedures followed to discover each event type are described below.

\subsubsection*{Rigma and simple deletions}
Rigma candidates were nominated as clusters of at least 2 overlapping DEL-like junctions with JCN less than ploidy and sizes between 10 kbp and 10 Mbp. \texttt{fishHook} was used as a Poisson model on a per sample basis to statistically nominate regions of the genome enriched with DEL-like junctions using sliding windows of 1 Mbp (500 kbp stride) while correcting for the occurrence of other junction types. From the model, regions with a significant enrichment of $FDR < 0.5$ were chosen. Any significant regions that were adjacent were collapsed into contiguous regions. Candidate deletion clusters that fall within these statistically enriched collapsed regions were then marked as rigma events. The remaining isolated low-JCN deletions were called simple deletions.  This caller is implemented as the \ttt{del} function of \ttt{gGnome}.

\subsubsection*{Pyrgo and simple duplications}
The procedure to identify pyrgo and simple duplications was analogous to the rigma identification above, but with the use of DUP-like junctions instead of DEL-like junctions. The same filters were applied as above. For candidate duplication clusters that pass these filters, fishHook was again used to statistically nominate genomic regions enriched for duplications while correcting for the occurrence of non-duplication junctions. Candidate duplication clusters that fall within the statistically enriched windows were then marked as a pyrgo. Collapsing was performed as for rigma. Remaining isolated low-JCN duplications were called simple duplications. This caller is implemented as the \ttt{dup} function of \ttt{gGnome}.

\subsubsection*{Double minute, BFB cycles, and Tyfonas}

To nominate amplicons (amplified clusters), weakly connected components were identified among vertices with copy number above $2\times$ploidy in JaBbA graphs. The resulting 12,588 subgraphs were further subsetted to only those whose maximum JCN $>7$, leaving 1,703 such high level amplicons with at least one high copy junction. Four curated features were then used to annotate each of these amplicons, 1) maximum segmental/interval copy number 2) the sum of all fold back inversion JCN divided by maximal interval copy number, 3) the ratio between maximum JCN and maximum interval copy number, and 4) the number of junctions with elevated JCN (thresholded on JCN $> 3$). To separate these amplicons, hierarchical clustering with Euclidean distance and complete linkage was performed on the basis of these features.

% , cutting the tree to arrive at $k = 4$, with each group corresponding to patterns of double minutes (DMs), breakage fusion bridge cycles (BFB cycless), tyfonas, and a small subset labeled "Other".

To assess the most optimal number of amplicon clusters, $k$ clusters from 2 to 15 were tested via a bootstrapping procedure.   Briefly, for each $k$, the amplicons were clustered across 100 bootstraps in which 75\% of the data was sampled and assessed using function "clusterboot", from the \texttt{clusterboot} R package. For each bootstrap and setting of $k$, $k$ clusters were computed and a Jaccard similarity index was computed between every observed cluster and its most similar cluster in the sample.  The stability of each observed cluster associated with a given $k$ was computed as the average Jaccard similarity index across all bootstraps for that observed cluster.  The cluster stability of the solution with parameter $k$ was then computed as the average stability of all observed clusters associated with $k$.

Using the elbow method, $k=4$ was identified as the parameter setting with the maximal cluster stability.  Furthermore, within each original cluster for a given $k$, a mean Jaccard score was computed to assess the level of evidence for a cluster. Clusters with mean score > 0.75 were considered stable, meaning there was sufficient evidence in the data to support the cluster as a true pattern, while those with mean score < 0.6 were likely artefactual.

Clustering was performed by building an unsupervised classifier that combines hierarchical clustering with a decision tree.  Specifically, a decision tree was trained using recursive partitioning (function \texttt{rpart} in the \texttt{eponymous} R package) from the hierarchical clustering result based on the above features. The decision tree arrived at a fold back JCN $\ge$ 0.5 to distinguish tyfonas/BFB cycles amplicons from the double minute/Other amplicon categories. Out of the amplicons with fold-back JCN $\geq$ 0.5, amplicons with total number of junctions with high JCN (\# $\geq$ 26) were called tyfonas and the rest were called BFB cycles. Within amplicons containing fold-back JCN < 0.5, those amplicons with $\geq$ 31 high copy junctions distinguished otherwise unspecified amplicons ("Other") from regular double minutes, respectively. This algorithm was then used to nominate the tyfonas, double minutes, BFB cycles, and the unspecified amplicons ("Other") among high level amplicons. These decision tree-based calls were then used for the subsequent analyses on these amplification events (see below). Although 4 was the optimal number of clusters, only three of the clusters (tyfonas-, double minute- and BFB cycles-labeled amplicon clusters) showed a mean Jaccard score of > 0.75, with a fourth cluster scoring 0.67. Due to the inconclusiveness of this fourth cluster within the amplicon data, this pattern was excluded from further analyses. This caller is implemented as the \ttt{amp} function of \ttt{gGnome}.



\subsubsection*{Chromothripsis}

Subgraphs that best fit the features of chromothripsis patterning described in \cite{korbel2013} were identified. Candidate clusters (i.e. weakly connected components) of genomic intervals overlapping the footprints of at least three ALT junctions were nominated.  These were further filtered to those clusters whose segmental copy numbers occupied a narrow range of states. Specifically, each interval within every footprint of a candidate cluster was subject to three criteria: 1) occupancy of at most 3 different copy states (for a diploid genome, adjusted up proportionally to the ploidy of the genome), 2) composition of at least 8 segments, and 3) having an interval copy number at the width-weighted 99\textsuperscript{th} percentile that did not exceed 4 (for a diploid genome, or adjusted to the ploidy of the genome). Clusters that survived this three-step filter were required to contain junctions with a near-uniform mixture of basic configurations ($\chi^2$ test, p-value $> 0.001$). Out of the final remaining clusters, several further criteria were required to yield the final chromothripsis calls as follows: each cluster must have 1) at least 7 internal junctions, 2) no fewer than two sub-100kbp footprints and no more than four $\geq$ 100 kbp segments within the cluster, and 3) on average at least 3 junctions with inter-breakend spans that overlap one another. This caller is implemented as the \ttt{chromothripsis} function of \ttt{gGnome}.

\subsubsection*{Chromoplexy}

Chromoplexy as first described in \cite{baca2013} can be identified by a series of reciprocal (or nearly-reciprocal, with small deletion bridges between adjacent breakends) long-range junctions (span more than 10 Mbp on the reference). Accordingly, chromoplexy was identified from a pool of low-JCN ($\leq 3$) edge clusters in which junction breakends are no further than 10 kbp away from the next junction breakend. Edge clusters that contained at least 3 long-range junctions, and whose footprints occupied at least 3 discontiguous genomic territories separated by >10 Mbp on the reference were called as chomoplexy events. This caller is implemented as the \ttt{chromoplexy} function of \ttt{gGnome}.

\subsubsection*{Templated insertion chains (TICs)}

We nominated templated insertion chain (TIC) events that result from short insertions involving more than 1 junction, usually with a gain of copy of all loci within the event, or linking disparate loci through shorter segmental "hops" within the genome \cite{Li2020-ds}. To capture this pattern, the following procedure was used: Breakends identified across the whole genome were ordered along reference coordinates, and pairs of breakends that fall within an interval of $\le$ 500 kbp and $\ge$ 50 kbp of each other were kept as candidates. These candidate breakend pairs comprise a "$+$"breakend followed by a "$-$" breakend in the direction of increasing reference coordinates, with each breakend originating from a different junction. A new graph was constructed with vertices comprising each junction and edges comprising adjacencies linked by candidate breakend pairs. All walks through the graph that traverse through at least 2 vertices/junctions were obtained. ALT junctions within each  walk traversed on this graph were then labeled as a unique TIC event.  This caller is implemented as the \ttt{tic} function of \ttt{gGnome}.



% \subsubsection*{Simple translocations, inverted duplications, and inversions}
\subsubsection*{Other simple events}

Inversions, inverted duplications, and translocations were also called as part of the compendium of SV event types in this study. Inversions and inverted duplications were both defined as pairs of overlapping, oppositely oriented, INV-like junctions of the same JCN as well as equal left and right vertex CN, with no third junction or a loose end interfering. We defined translocations as single or reciprocal pairs of TRA-like junctions connecting two different reference chromosomes that are not connected by any other junctions.  This caller is implemented as the \ttt{simple} function of \ttt{gGnome}.

\subsection*{Cancer genome analysis} Harmonized pipelines for junction detection, high-density read depth estimation, purity / ploidy inference, somatic SNV / indel calling, and loss of function mutation calling were applied to the full dataset of 2,813 WGS cases as described below. Several standard analyses for mutational interpretation, signature inference, and APOBEC analyses were performed.

\subsubsection*{Junction detection}

High- and low-confidence somatic junction calls for tumor / normal pairs were obtained using \texttt{SvABA} \cite{wala2018}, based on the optimal settings for input into JaBbA (see above). For samples (e.g. CCLE cell lines) that lacked a paired normal sample, we used HCC1143BL WGS as the normal / constitutional reference sample.

To further eliminate constitutional junctions, we used \texttt{SvABA} to obtain normal / constitutional junction calls for 1,017 TCGA tumor/normal pairs and construct a panel of normals (PON).  All somatic junction candidates within 1 kbp of a PON junction (determined by overlap at both breakends) were flagged as constitutional and removed.  The remaining \texttt{SvABA} calls were treated as "high-confidence" calls if they passed \texttt{SvABA} internal filters.  The remaining calls were treated as "low-confidence" (see "JaBbA model fitting" section above).

\subsubsection*{Read depth} \label{meth:readDepth}
Coverage profiles were derived using fragCounter, which counts the midpoints of proper pairs in 200 bp bins tiling hg19 using \texttt{samtools} (v1.9) and corrects the binned read counts by LOESS (function \texttt{loess} within R base package) (see "read depth preprocessing" section for mathematical details). We then segmented tumor / normal ratios using Circular Binary Segmentation \cite{olshen2004} (segment function in DNAcopy R package). For samples lacking a matched normal (i.e. CCLE cell lines), a composite of the 1,017 TCGA normal coverage profiles was used, comprising the average of the 200 bp bins across all autosomal chromosome coordinates.

\subsubsection*{Quality control}
From the 2,813 WGS sequenced samples, we excluded 34 samples for which JaBbA optimization did not converge and one sample which exceeded RAM requirements during SV classification. Among 69 graphs which did not converge on first pass ($<$ 16G RAM, $<$ 24 hours compute time), 35 successfully completed with increased memory and run time ($<$ 500G RAM, > 24 hours of compute time).  One of these 35 failed SV classification due to the number of incorporated junctions exceeding RAM requirements.  Upon inspection, graphs that failed to converge at the JaBbA were associated with noisy read depth data and many segments ($>$ 130,000 internal vertices).

\subsubsection*{Purity and ploidy estimation}

For all cases with both a tumor and normal WGS profile, read counts supporting germline heterozygous SNP alleles were obtained by intersecting SNV sites present in both tumor and normal with SNPs from HapMap 3.3. Purity and ploidy estimates were obtained for all samples through \texttt{Sequenza} \cite{Favero2015-zd}, \texttt{TITAN} \cite{ha2014}, \texttt{Ppurple}  (\url{https://github.com/mskilab/Ppurple}), or a custom least squares grid search \texttt{ppgrid} (available through JaBbA). Consensus purity and ploidy was determined by curation from the panel of the three calls across all tumor normal pairs. \texttt{ppgrid}, which does not require germline SNP sites, was used for CCLE cell lines to obtain purity and ploidy estimates since germline SNP sites were unavailable.

%% With the germline filtered SV, CBS segmentation and coverage profiles, and initial purity/ploidy estimates, we used the novel algorithm, Junction Balance Analysis (JaBbA) to obtained a coherent genome-wide estimation of integer copy number changes simultaneously explained by concomitant SVs. We obtained JaBbA graph representations of the underlying rearranged genomes for the 2847 genomes in our purview, and parsed them for downstream analysis using \texttt{gGnome}.

\subsubsection*{Somatic SNVs and indels}

To obtain somatic SNV/indel calls \texttt{Strelka2} \cite{kim2018} was run under paired (i.e. tumor / normal) mode with default parameters using hg19-based references. Somatic SNVs and indels were obtained only for those cases where tumor and normal BAMs were available (2481 out of 2813 cases). In addition to the recommended filters, a universal mask (https://github.com/lh3/sgdp-fermi/releases/download/v1/sgdp-263-hs37d5.tgz) was used to remove common artifacts in low-mappability regions, described in \cite{mallick2016}.  After these initial filters, only sites determined to pass \texttt{Strelka2}'s quality filter (i.e. sites where the "FILTER" field was marked as "PASS") were considered, yielding a high quality set of somatic SNV calls. Variant annotations were obtained for SNV/indel using SnpEff with the GRCh37.75 database.

\subsubsection*{Germline SNVs and indels}

For constitutional SNV/indel calls \texttt{Strelka2} \cite{kim2018} was run as above, except in normal-only mode. Germline SNVs and indels were obtained only for non-cell line samples within the study. The universal mask was also used for germline SNV/indel calls. Additional filters restricted the germline variants used to those which met the following criteria: i) sites that do not overlap with common variants i.e. those variants that matched coordinates and ALT alleles with sites from the normal ExAC population that have a minor allele frequency of >1\% (\url{ftp://ftp.broadinstitute.org/pub/ExAC_release/release0.3.1/subsets/)}, ii) high quality sites as determined by \texttt{Strelka2}'s quality thresholds (i.e. sites in which the "FILTER" field was marked as "PASS"), and iii) sites that overlapped and matched ALT alleles with known pathogenic variants from ClinVar annotations (\url{ftp.ncbi.nlm.nih.gov:/pub/clinvar/vcf_GRCh37}). Variant annotations were obtained for this final, high quality set of germline SNVs/indels using SnpEff as above.  For constitutional heterozygous SNP calls (see "Postprocessing" section above), we used HapMap 3.3 defined on hg19 (https://www.sanger.ac.uk/resources/downloads/human/hapmap3.html).  We assessed tumor and normal SNP counts using Rsamtools (R / Bioconductor) and nominated constitutional heterozygote sites as any site with $>0.2$ and $<0.8$ ALT variant allele fraction in the normal sample.

\clearpage
\subsubsection*{Mutation interpretation}
Alteration status of genes was obtained across the cohort. Homozygous deletions, heterozygous deletions, and amplifications were detected from absolute copy number calls available from the JaBbA graphs. Filtered SNV/indel annotations falling within protein coding regions, were considered only if they constituted somatic missense or truncating events, germline pathogenic variants, or germline truncating variants. To obtain loss-of-heterozygosity calls allele-specific copy number was called in cases with both tumor and normal samples. With genome-wide germline heterozygosity coverage for these samples, allele specific copy number was measured using JaBbA using the \ttt{jabba.alleles()} function within the \texttt{JaBbA R} package. For CCLE, our calls only included the publicly available mutational drivers.

\subsection*{Driver amplification analysis}

For the scope of amplification drivers, we selected Cancer Gene Census genes that intersect with the pan-cancer GISTIC amplification peaks in \cite{Zack:2013f1f}. Any overlap between the genomic footprint of a double minute, BFB cycles, or tyfonas event with an amplified driver gene (thresholded as having a segmental CN > twice ploidy) was counted to calculate the frequency of each event upon one of these genes.

\subsection*{Protein-coding gene fusions across complex SV types}
We obtained expressed protein-coding gene fusion calls from a published TCGA RNA-seq analysis \cite{Dehghannasiri2019-hk}.  Coordinate overlaps between expressed ($>10$ supporting RNA-seq reads) protein-coding gene fusion calls and SV event footprints from JaBbA graphs were tallied across 870 intersecting tumors. The number and genomic density of unique, expressed fusion products attributed to each event type were plotted in Figure \ref{fig:tyf_fusion}.  Enrichment between tyfonas and other SV event categories (double minutes, BFB cycles, chromothripsis, other) was assessed using Wilcoxon rank-sum test.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Appendix C
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter*{Appendix C: Chapter 4 supplementary materials and methods} \label{app:c}
\subsection*{WGS basic data processing} \label{app:tc_wgs_processing}
Reads were aligned to GRCh37/hg19 using the Burroughs-Wheeler aligner (bwa mem v0.7.8, http://bio-bwa.sourceforge.net/). Best practices for post-alignment data processing were followed through use of Picard (https://broadinstitute.github.io/picard/) tools to mark duplicates, the GATK (v.2.7.4) (https://software.broadinstitute.org/gatk/) IndelRealigner module, and GATK base quality recalibration.

Variant rearrangement junctions were identified using SvABA \cite{wala2018} and GRIDSS \cite{Cameron2017-pz} with standard settings. For MRC5 samples, the somatic variant setting of each tool was used, with the ancestral MRC5 line as the matched normal.  SvABA was applied using a panel of normals (PON) that was constructed by running SvABA to obtain constitutional junction calls for 1,017 TCGA tumor/normal pairs (TCGA dbGaP: phs000178.v11.p8).  For GRIDSS, a PON was obtained from the Hartwig Medical Foundation (https://nextcloud.hartwigmedicalfoundation.nl). 1 kbp binned GC and mappability corrected read depth was computed using fragCounter (https://github.com/mskilab/fragcounter). Systematic read depth bias was subsequently removed using dryclean (https://github.com/mskilab/dryclean).

\subsection*{Low-pass WGS clustering}
Genome-wide binned read depth was aggregated across 118 low pass WGS clones across 10 kbp bins by taking the median of 1 kbp binned normalized read depth from dryclean (see above). To minimize read depth noise in unmappable regions, recurrent (>10\% of the cohort) low-quality coverage regions (defined in \cite{Hadi2020-um}) are combined with regions bearing consistently high variance in our high-pass sequencing dataset (standard deviation >0.3 for bin value over the mean in 100 kbp windows). Hierarchical clustering was then applied on the genome-wide Euclidean distance of bins, with âmethod = ward.D2â option. Six clusters were identified following dendrogram inspection.  

\subsection*{Junction balance analysis}
Preliminary junction balanced genome graphs were generated for MRC5 and SV40T cell lines from binned read depth and junction calls (see above) using JaBbA (https://github.com/mskilab/JaBbA)\cite{Hadi2020-um}. Briefly, 1 kbp binned read depth output from dryclean was collapsed to 5 kbp and JaBbA was run with slack penalty 500. gGnome (https://github.com/mskilab/gGnome) was used to identify complex structural variant patterns. Genome graphs and corresponding genomic data (e.g. binned coverage, allelic bin counts) were visualized using gTrack (https://github.com/mskilab/gTrack). 

\subsection*{Loose end classification} \label{app:tc_loose_end}
Each loose end in each MRC5 genome graph was analyzed to identify a clone specific (i.e. absent in the ancestral MRC5 line) origin for the mates of high mapping quality (MAPQ=60) reads mapping to the location and strand of the loose end. These mates were assessed for neo-telomeric sequences by counting instances of 11 permutations of a 12-bp telomere repeat motif (TTAGGGTTAGGG) (using the R / Bioconductor Biostrings package) in the mates. The mates were also assembled into contigs using fermi \cite{Li2012-ac} aligned using bwa mem \cite{Li2014-mj} via the RSeqLib R package \cite{Wala2017-ud} to hg38 (which contains a more  highly resolved centromere build) to characterize novel repeat (e.g. centromere) fusions (GATK Human reference genome, hg38, data bundle including Homo\_sapiens\_assembly38.fasta, gs://gcp-public-data--broad-reference). The loose end loci were also assessed through overlap with the hg19 repeatMasker database (human\_g1k\_v37\_decoy.repeatmasker) for the presence of reference annotated repeats that might explain the absence of a mappable junction explaining the copy number change.  

\subsection*{SNV phylogeny}
To compute an SNV phylogeny across MRC clones, we first identified SNV that were acquired in MRC5 clones relative to the ancestral MRC5 line using Strelka2 \cite{kim2018} (https://github.com/Illumina/strelka) under paired (i.e. tumor / normal) mode with the clone as the âtumorâ and the MRC ancestral line as the ânormalâ sample and default parameters and GATK hg19 resource bundle (Genome Analysis Toolkit GATK Resource Bundle for hg19; gs://gatk-legacy-bundles). Acquired SNVs were first filtered according to the Strelka2 PASS filter as well as additional filters (MQ = 60, SomaticEVS>12, total ALT count > 4) yielding 27,220 total unique variants across the 13 MRC5 clones. Reference and variant allelic read counts were assessed at each SNV site (via the R / Bioconductor Rsamtools package, version 3.6.1, http://www.r-project.org/) across all 13 clones. We then further required a >0.5 posterior probability of a variant being present in a sample, by assuming Binomial likelihood of variant read count and using the aggregated allele frequency in all samples as the prior, resulting in the final 14,970 unique mutations. The binary matrix of clones by SNV loci. was then used to derive a neighbor-joining phylogenetic tree using the R / Bioconductor package ape. Following tree construction, we associated each SNV with its most likely phylogenetic tree branch by comparing the binary incidence vector associated with each SNV with the binary incidence vector associated with each tree branch, and finding the closest branch using Jaccard distance, only linking SNV to branches when the SNV was within <0.1 Jaccard distance of the closest branch, thus producing the groupings of SNVs in Figure \ref*{fig:tc_phylo}.   

\subsection*{Parental SNP allelic phasing and imbalance}
Germline heterozygous sites in the parental MRC5 line were identified by computing allelic counts at HapMap sites (GATK human reference genome, hg19 data bundle, hapmap\_3.3.b37.vcf) and identifying loci with variant allele fraction >0.3 and <0.7.  Y11, a clone with loss of a single allele at 12p, was chosen to phase parental SNPs on 12p. At each locus, the allele (reference or alternate) with a 0 read count was assigned to the âLâ (lost) haplotype and the other allele was assigned to the âRâ (retained) haplotype. (All heterozygous SNP loci in the region contained exactly one allele with a 0 read count). L and R allelic counts were then computed at these sites across all 13 high pass WGS and 131 low pass WGS samples. These counts were divided by the genome wide mean of heterozygous SNP allele counts (in these 100\% pure and nearly diploid samples) to derive the absolute allelic copy number \cite{Carter2012-xo}.

\subsection*{SNV clustering}
Inter-SNV distances were computed for all pairs of reference adjacent acquired SNVs associated with each MRC5 clone and visualized as rainfall plots. Runs of two or more SNVs with inter SNV distances < 2 kbp were nominated as clusters. Two distinct SNV clusters were identified on chromosome 12p across the 13 clones.  

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Appendix D
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\chapter*{Appendix D: Chapter 5 Supplementary materials and methods} \label{app:d}
\subsection*{Sample selection and whole-genome sequencing}
Eighty-five samples were selected for WGS, among 118 previously whole-exome sequenced TCGA LUAD samples that were negative for 1) activating mutations in \textit{KRAS}, \textit{EGFR}, \textit{BRAF}, \textit{ERBB2}, \textit{MET}, \textit{RIT1}, \textit{NRAS}, \textit{RAF1}, \textit{HRAS}, \textit{ARAF}, \textit{MAP2K1} and \textit{SOS1}; 2) loss-of-function mutations in \textit{NF1} and \textit{RASA1}; 3) fusions in \textit{ALK}, \textit{ROS1}, \textit{RET}, \textit{MET} and \textit{NTRK2}; and 4) amplification in \textit{EGFR}, \textit{ERBB2}, \textit{KRAS}, \textit{MET}, \textit{FGFR1} and \textit{MAPK1}. The same criteria were applied to re-identify RPA(-) samples in the WGS analysis, except that over-expression (defined as a z-score greater than 1.96 for gene expression among the full TCGA LUAD samples) was additionally required to qualify an amplification as an oncogenic driving event.
DNA was sequenced using the Illumina HiSeq platform. Paired-end sequencing reads were aligned to hg19 using BWA (v0.6.2) \cite{Li2009-uz} aln and processed through NovoSort (v1.03.01) to mark PCR duplicates (http://www.novocraft.com/products/novosort/), then through GATK (v3.4) \cite{McKenna2010-tm} for indel realignment (jointly for the normal and tumor samples). Base quality scores were recalibrated with GATK.

\subsection*{Identification of somatic mutations and SCNAs}
Somatic SNVs were called by MuTect (v1.1.7) \cite{Cibulskis2013-ui}, Strelka (v1.0.14) \cite{Saunders2012-rm} and LoFreq (v2.1.3a) \cite{Wilm2012-rd}. Somatic indels were called by Strelka, Pindel (v0.2.5) \cite{Ye2009-tr} and Scalpel (v0.5.3) \cite{Narzisi2014-gy}. Variants called by only one of the three callers were filtered out. Final VCFs were formatted to pass the EBI validator (v 0.4.3) (https://vcftools.github.io/perl\_module.html). Variants were annotated for their effect (non-synonymous coding, nonsense, etc.) using snpEff \cite{Cingolani2012-bn} based on human genome annotations from ENSEMBL. We further annotated the variants using snpEff, snpSift \cite{Cingolani2012-bn} and GATK VariantAnnotator module with information from COSMIC \cite{Sondka2018-ad}, 1000 Genomes Project \cite{Consortium2015-dz} , ExAC \cite{Karczewski2017-ek}, CIViC \cite{Griffith2017-io}, and UniProt \cite{Consortium2019-bm}. Non-coding mutations were further annotated by Funseq2 \cite{Fu2014-so}. Mutational signatures were analyzed using SignatureAnalyzer \cite{Kim2016-vk}.

Data on genetic ancestry, genome double, aneuploidy, leukocyte fraction and other clinical features were downloaded from the Genomic Data Commons (https://portal.gdc.cancer.gov/). GISTIC 2.0 \cite{Mermel2011-gt} was used to identify significant SCNAs using copy number segments generated by Titan \cite{Ha2014-ai}. High-level amplification was defined by log2-transformed copy number ratios $>1$. To calculate the allelic fraction of \textit{KRAS} mutations in WGS and WES, we applied a custom script counting reads supporting the altered alleles and the reference alleles, respectively \cite{Carrot-Zhang2017-lo}. Reads with based quality and mapping quality lower than 30 were removed.

\subsection*{Identification of Structural variants and genome graph reconstructions}
Somatic aberrant junctions (i.e. pairs of strand-specific disconnected loci that form neo-adjacencies in the cancer genome) were identified by SvABA v1.1.3 \cite{Wala2018-qa}, using the default setting for tumor-normal pairs. We then used JaBbA, a junction balance analysis \cite{Hadi2020-um} to reconstruct a genome graph for each sample through the application of a maximum likelihood model to high-resolution binned (200bp) normalized read depth and unfiltered SvABA junctions (after exclusion of small (<1kbp) deletion-like junctions). The read depth input to JaBbA was calculated as the ratio between tumor and normal sampleâs WGS read counts in all 200bp genomic bins, corrected for guanine/cytosine content and 100mer mappability. Subsequently, we used Circular Binary Segmentation \cite{Olshen2004-cl} with \textalpha parameter at $1 \times 10^{-5}$ to derive a primary segmentation for each sample. The segmentation was later combined with SvABA-identified aberrant junctions to build the genome graph. The affine mapping between read depth signal and integer copy number is dictated by the hyperparameters ploidy and purity. We used the published purity and ploidy values from the GDC (https://portal.gdc.cancer.gov/). Any sample missing from that resource was supplemented by Sequenza (Favero et al., 2015), based on the allelic read counts at germline heterozygous sites using Samtools \cite{Li2009-uz}. A total of 13 types of simple and complex SV events were annotated and visualized using gGnome (https://github.com/mskilab/gGnome.git), with the same default parameters as described in Hadi et al. 2020. SV burden per sample was defined by the number of junctions of simple SVs in that sample. Genome graphs drawn in copy number over genomic coordinates plots are made with gTrack (https://www.github.com/mskilab/gTrack) and gGnome (https://www.github.com/mskilab/gGnome).

\subsection*{Oncoprints, mutation barplots, and expression quantiles}
Genomic alterations affecting genes in the cohort were plotted with ComplexHeatmap \cite{Gu2016-ft} with the aforementioned definitions for SNVs and CNAs. Tumor mutation burden was calculated by dividing the total number of SNVs in the eligible mutation calling region proposed in \cite{Li2014-mj} by the total width of these regions (2429.397 Mbp). Expression quantiles and density plots are made with the geneâs RSEM (RNA-seq by Expectation Maximization) values of the full set of 507 LUAD RNA-seq in TCGA. Mutation barplots (âloliplotâ) are made with trackViewer package (http://bioconductor.org/packages/release/bioc/html/trackViewer.html).

\clearpage
\subsection*{Differential alteration frequencies between RPA(-)\textsubscript{G} and RPA(+)\textsubscript{G}}
The frequency of genomic alterations in various genes were compared between RPA(-)\textsubscript{G} and RPA(+)\textsubscript{G} using Fisherâs exact tests with false discovery rate (FDR) threshold below 0.1. To maximize statistical power, we only considered the variant types that can be detected both through WES and WGS, and compared the frequency of alteration in the RPA(-)\textsubscript{G} group (N=57) to the rest of all the TCGA LUAD samples with WES-based variant calls from the PanCanAtlas (N=411, http://api.gdc.cancer.gov/data/1c8cfe5f-e52d-41ba-94da-f15ea1337efc). Same method was used for clinical or molecular features, including smoking history, age of diagnosis, leukocyte fraction, genome doubling, degree of aneuploidy, genetic ancestry, primary disease stage (data available through GDC) using Fisherâs exact test or Wilcoxonâs Rank test. TMB comparison was performed based on WES-defined TMB values using linear regression controlling for tumor purity.

% SCISSOR analysis
% RNA-seq BAM files for TCGA LUAD samples were used as input to SCISSOR (Choi et al. 2020) for analysis of structural changes in RNA transcripts. Briefly, SCISSOR is a statistical method for unsupervised screening of a range of structural alterations in RNA-seq data including alternative splicing, intron retention, de novo splice sites, intra-/intergenic deletions, and alternative transcription start/termination. For each gene under consideration, SCISSOR aims to identify anomalous shapes in expression profile by considering aligned short read data through a base-level read pileup. To identify such profile shape changes, it quantifies the level of abnormality of each sample using a projection depth approach (Dang and Serfling, 2010; Donoho and Gasko, 1992) and then uses the statistics from the cohort to detect and rank shape outliers. Using reference exon coordinates derived from UCSC hg19 known genes, we constructed base-resolution read coverage data including intronic parts from the BAM files. For TP53, STK11, and SMARCA4, SCISSOR identified shape changes where the aligned coverage shape is significantly different from the majority of samples with a significance level 1x10-4. This process confirmed the effects of focal, exon deletions on RNA transcripts.

\subsection*{Recurrence analysis of genomic alterations}
A gamma-Poisson regression framework (fishHook) that takes into account different confounders that affect the mutation count in a population was used \cite{Imielinski2017-nt}. fishHook allows for defining the genomic region of interest, called the hypotheses set, to be used for recurrence analyses. For coding mutation analysis, models were fitted with gene bodies as the hypotheses set. Non-synonymous, missense and truncating mutations were tested separately. We also corrected for multiple hypotheses on 47 genes identified by prior genomic studies on LUAD \cite{Campbell2016-xv,Cancer_Genome_Atlas_Research_Network2014-ju,Imielinski2012-vv}. For non-coding mutation analysis, lung-specific ATAC-seq peaks (https://gdc.cancer.gov/about-data/publications/ATACseq-AWG) defining open chromatin regions \cite{Corces2018-nl} were used as the hypotheses set. Lift over was used to map hg38 coordinates to hg19. ATAC peaks occurring at least 2 of 44 LUAD samples were used, and peaks within 100bp distance were merged. The model was fitted with the following covariates of the neutral mutation density: 1) Fraction of heterochromatic regions in each query interval. Heterochromatin annotation obtained from chromHMM on A549 cell line from Epigenomics Roadmap \cite{Roadmap_Epigenomics_Consortium2015-mg}; 2) Gene expression data for LUAD for A549 cell line; 3) GC content in reference genome; 4) Replication timing from normal human epidermal keratinocytes; 5) DNA accessibility annotation from DNase-seq for A549 cell line. Besides genome-wide hypotheses, we also tested within the subset of ATAC peaks that overlaps recurrently amplified regions \cite{Campbell2016-xv} and with putative target gene median RSEM$>$10 among LUAD samples.

Associations between non-coding mutation and target gene expression are evaluated through fitting an ordinary linear model to log RSEM values with the non-coding mutation presence and amplification status of \textit{ILF2} gene.

\subsection*{Quantification and Statistical Analysis}
All statistical tests are carried out using R (v3.6.1) and Bioconductor (v3.10) and listed within the figure legends and Results. Fisherâs exact tests are executed with âfisher.testâ function, Wilcoxonâs Rank test with âwilcox.testâ, ordinary linear models with âlmâ.

\bibliography{sampleThesis}

\end{document}

% Intuitively, a walk along on a genome graph maps to a DNA sequence and a genome graph is a compilation of all possible linear sequences that could result from rearranging the reference genome assuming the junctions are completely known. This idea is similar to the assembly graph in \textit{de novo} assembly, except we are pre-defined within a linear reference genome and adding further rearrangements.


% Following the above definitions, except for whole genome/chromosome/contig gain or losses or foreign sequence insertions, every structural change of a genome can be effectively seen as generating new adjacencies (ALT edges) between previously non-adjacent vertices.

% Extending the properties of $r$ to any function of vertices or edges, we call a function $f(v), v \in V(G)$ or $f(e), e \in E(G)$ as skew-symmetric if $f(v) = f(\bar{v})$, $f(e)=f(\bar{e})$. One such function is copy number $\kappa : G \rightarrow \mathbb{N}$, as for double stranded DNA the copy number of one strand should always be equal to its reverse complement.

%% For testing mutational status, only those DNA damage genes which intersected  between Cancer Gene Census (CGC) \cite{Sondka2018-ad} genes and genes in DNA damage repair pathways as annotated in \cite{pearl2015} were used. Final lists of genes that were nominated significant and mutational status within those genes per sample are found in \textbf{Table \ref{tab:S2}}.

%The inference of JBGGs is a computationally challenging combinatorial optimization problem that others have approached using probabilistic graphical models (PGM) \cite{Li2016-qa,McPherson2017-ry} and mixed-integer programming (MIP) \cite{Medvedev:2010bm, Greenman:2012eg, Oesper2012-vw, Dzamba2017-wo, Aganezov2020-cu,Lee2021-rl}. A serious impediment to applying these approaches to the study of complex SVs is their inability to produce CN profiles of comparable quality to non-graph based CNA callers (e.g. CONSERTING \cite{Chen2015-sw}, BIC-Seq \cite{Xi2011-oa}) while also fitting JCN, across a wide range of tumor purities.

% JaBbA method, need to adpat
% We also validated JaBbA's estimates of JCN from short-read WGS with 10X Chromium linked-read WGS, which provides a more direct readout of JCN with its high number of long junction-spanning fragments. In the breast cancer cell line HCC1954, JaBbA-derived JCN estimates closely correlated with the coverage of junction-spanning linked-read barcodes ($R^2 = 0.91$, \textbf{Fig. \ref{fig:Fig1}C}). This included examples of low JCN (e.g. JCN = 1) junctions connecting both low and high CN intervals, as well as high JCN (JCN = 10)  junctions (\textbf{Fig. \ref{fig:Fig1}D}). These results show that JCN is a property that can be robustly inferred from short-read WGS and is independent from (interval) CN. 

% from telomere intro
% Structural variant is a hallmark of cancer genomes. 
% Despite advances in the identification and classification of Structural variants, a mechanistic understanding of the underlying causes is often still lacking. SV mutational processes may have a more complex etiology than those driving the formation of SNVs and generate a more complex spectrum of patterns: layers of simple SVs can reshape a locus gradually and across multiple alleles, and complex SVs can rapidly rewire many genomic regions. In addition, multiple underlying causes can lead to the same type of rearrangement, and diverse outcomes can originate from a single cause. Further, it has been expensive and technically challenging to delineate specific mechanisms, although some progress has been made \cite{Zhang2015-np,Willis2017-bw,Maciejowski2015-wx,Ly2019-bt,Ghezraoui2014-dt,Umbreit2020-kr}.

% Telomere crisis, which is thought to occur at an early stage of carcinogenesis before a telomere maintenance mechanism is activated \cite{Shay2005-fb}, has been suggested as a cause of cancer genome SVs. A priori, the genomic consequences of telomere crisis are predicted to be profound: critically short telomeres in human cells can trigger a DNA damage response, and inappropriately engage DNA repair pathways resulting in telomere to telomere fusions \cite{Artandi2000-ab,Maciejowski2017-kx}. Subsequent cell divisions in the presence of fused dicentric chromosomes has long been considered a mechanism driving complex chromosomal rearrangements such as BFB cycles in tumors \cite{McClintock1939-oi,Gisselsson2001-hq}. The characteristic fold-back inversions of BFB cycles are known to contribute to tumorigenesis in acute lymphocytic leukemia (ALL) \cite{Li2014-mj} as well as squamous cell cancers and esophageal adenocarcinoma \cite{Hadi2020-um}. 

% Studies using  sequencing in cultured human cells have also illuminated the genomic consequences of telomere dysfunction. Even a single artificially deprotected telomere can fuse with multiple intra- and inter-chromosomal loci leading to complex fusion products \cite{Liddiard2016-ay} but it is unclear whether these complex rearrangements are compatible with viability and escape from telomere crisis. The resolution of dicentric chromosomes induced by over-expression of a dominant negative allele of the telomere binding protein TRF2 can lead to the dramatic chromosome shattering phenomenon of chromothripsis\cite{maciejowski2015,Umbreit2020-kr,Maciejowski2020-bw}. % However, to date the only study directly investigating the consequences of a sustained period of telomere dysfunction failed to identify any complex rearrangements in HCT116 colon carcinoma cells \cite{Cleal2019-dk}. This may be because these cells readily escaped from the telomere dysfunction that was induced by expression of a dominant-negative hTERT (telomerase reverse transcriptase) allele. In genetically unstable HCT116 cells deficient for Non-Homologous End Joining (NHEJ) factors, complex chained SVs were observed after telomere dysfunction, but the relevance of these types of rearrangements to human cancer remains unclear \cite{Cleal2019-dk}.
% We approached this problem in two ways. First, we performed whole genome sequencing (WGS) on a panel of nine previously isolated cell lines that had escaped telomere crisis spontaneously through telomerase activation. In the post-crisis immortalized cell line panel, the consequences of telomere crisis were varied, ranging from relatively unperturbed to highly rearranged genomes. Importantly, neither BFB cycles nor chromothripsis were universally observed. Secondly, we created a controlled \textit{in vitro} telomere crisis system by engineering an MRC5-derived cell line in which telomerase could be activated during telomere crisis and analyed the resulting post-crisis clones by WGS.  

% In this system, telomere crisis often engendered structures reminiscent of BFB cycles and chromothripsis. Together these data establish that the genomic consequences of telomere crisis are not readily predictable and do not invariably include BFBs and chromothripsis. Therefore, it is currently not possible to infer whether telomere crisis occurred in the proliferative history of cancers based on the pattern of SVs.

% A key question is whether these RPA(-) LUADs represent a distinct RTK/RAS/RAF-independent entity that is associated with a unique evolutionary path and therapeutic sensitivities, or whether they have been mislabeled as RPA(-) due to technical factors (e.g., sample quality or limitations in the genomic profiling technologies). % Specifically, apparent RPA(-) LUADs may harbor pathogenic variants in RTK/RAS/RAF pathway genes that are not adequately detected by WES, RNA-seq, or targeted gene panels such as those commonly used in large research studies or in clinical laboratories \cite{Vinagre2013-tg,Weischenfeldt2017-jr}.

% We postulated that a more comprehensive analysis of candidate RPA(-) LUADs using whole-genome sequencing (WGS) in addition to WES and RNA-seq might illuminate more precisely the basic biology and also influence clinical management.  Since WGS is particularly effective in the identification of non-coding and structural genomic alterations, we hypothesized that analysis of those types of genome variation might reveal key features of RPA(-) LUAD biology.

% \chapter{Old Introduction}
% Simple SVs (e.g. deletions, duplications, translocations, and inversions) arise through the breakage and fusion of 1 or 2 genomic locations, complex SVs can cause multiple ($>2$) DNA junctions in distinct topologies within the reference genome that yield one or more copies of complex rearranged alleles \cite{maciejowski2017}.

%1) the discordant read pairing or observed (either directly within one read or assembled from a group of overlapping reads) sequences indicate a variant junction -- two disconnected genomic loci in reference genome that are adjacent in the sample ; and 2) the read depth accrued within a genomic interval informs its copy number (CN). Leveraging these two readouts of the underlying rearranged genome, various cancer WGS studies have describe a plethora of complex SV patterns.

% adpated from TC
% More recently, pan-cancer WGS studies have systematically categorized the spectrum of structural variants (SVs) found in cancer genomes, ranging from simple deletions, duplications and translocations to complex and often multi-chromosomal rearrangements\cite{Drier2013-oe,Yang2013-fl,Hadi2020-um}. 

% In \cite{Hadi2020-um} and this dissertation, the analysis of genome graphs provides a rigorous and unified framework to classify simple and complex SVs (including chromothripsis, breakage-fusion-bridge (BFB) cycles, and double minutes), identify novel event classes, and study the rearranged structure of aneuploid alleles\cite{Hadi2020-um}.

% (chromothripsis \cite{stephens2011}, chromoplexy \cite{baca2013}, templated insertion chains (TIC) \cite{spies2017,Li2020-ds}, breakage fusion bridge cycles (BFB cycless) \cite{Zakov:2013cm}), 
%Though many mechanisms have been proposed to explain complex SV patterns, 

% \section{Complexity of SVs in the cancer genomes}
% Despite these extraordinary advancements, the discoveries of SNV and INDEL patterns have been outpacing that of SVs, exemplified by the distillation of mutation signatures from large-scale mutation profiles represented by local sequence context of the substitution \cite{alexandrov2013} to high-throughput experimental designs \cite{Zou2021-je} to associate mutation signatures with exogenous mutagens or defective repair pathways. Yet, it still remains challenging to characterize the full spectrum of SVs in cancer. There are many reasons to this, but a fundamental one is that the SVs in cancer genomes are complex and there lacks a suitable framework to account for it.

% Since each simple SV only contains one (tandem duplications and deletions) or two junctions (inversions and balanced translocations), it is convenient and sufficient to adopt the substitution model that has long been the standard practice in representing and storing small genomic variants \cite{Li2008-su}, such that one variant is represented as an edit of a fixed location of the reference genome and in most cases each variant can be interepreted independently from the other variants \cite{Li2009-fv}.

% In practice, The detection of variant \textit{junctions} (each a pair of oriented genomic locations, or \textit{breakends}, made adjacent in one of four configurations: deletion-like (DEL-like), duplication-like (DUP-like), inversion-like (INV-like), or translocation-like (TRA-like, Figure \ref{fig:junction_orientations}) in whole genome sequencing (WGS) is routine. However, the classification of junctions into events (e.g. deletion, duplication, inversion, translocation) becomes difficult when two or more junctions are near each other.  Furthermore, rearrangements and copy number alterations (CNA) are usually analyzed and interpreted separately, despite being facets of a single genome structure. As a result, quantitative details of the copy number state are only loosely integrated with the junction topology when loci are classified into events. 

% Lung adenocarcinoma (LUAD) is the most common lung malignancy and a leading cause of cancer death in the United States \cite{Siegel2019-gb}. Most LUADs are driven by constitutive activation of MAPK signaling, which is, in turn, a consequence of alterations in receptor tyrosine kinases (RTK), downstream RAS/RAF/MEK cascade proteins, and their regulators \cite{Desai2014-qe}. Comprehensive studies using whole-exome (WES) and/or transcriptome sequencing (RNA-seq) have identified RTK/RAS/RAF pathway driver alterations in 70-80\% of LUADs \cite{Campbell2016-xv,Cancer_Genome_Atlas_Research_Network2014-ju,Imielinski2012-vv}. Because driver alterations in RTK/RAS/RAF pathway genes such as EGFR, BRAF, ALK, RET, ROS1, and KRAS, can be targeted by small molecule inhibitors that significantly prolong survival, therapeutic decision making in LUADs is routinely directed by testing for many, but not all RTK/RAS/RAF alterations to direct \cite{Herbst2018-oo}.


%\subsection*{Cell lines}
%MRC5 human lung fibroblasts (CCL-171), Phoenix-ampho (CRL-3213), RPE-1 hTERT (CRL-4000), HCT-116 (CCL-247) and U2-OS (HTB-96) cells were obtained from ATCC for this study. 293-FT cells were obtained from ThermoFisher. MRC5 cells and derivatives thereof were grown in EMEM media (ATCC) supplemented with 15\% fetal bovine serum (FBS; Gibco) and 100 U/mL of penicillin and 100 $\mu$g/mL streptomycin (PenStrep, Gibco) at 37ÂºC, 5\% CO2. hTERT RPE-1 cells were grown in DMEM:F12 media (Gibco) with 10\% FBS and PenStrep at 37ÂºC, 5\% CO2. HCT-116 colorectal carcinoma cells and U2OS cells were grown in DMEM with 10\% FBS and PenStrep at 37ÂºC, 5\% CO2.

%\subsection*{Immortalized cell line panel }
%Details of the post-crisis immortalized cell line panel are provided in Supplementary Table 1. HA-1M cells were a kind gift of Silvia Bacchetti29, SW13/26/39 cells were a kind gift of Jerry Shay28, and Bet-3B/3K and BFT-3B/G/K cells were a kind gift of Roger Reddel27. 

%\subsection*{Cloning and plasmids}
%A dual-shRNA vector LM2PshRB.698-p21.890-PURO was used to knockdown Rb and p2149. The inducible dCas9-VPR (pCW57-dCas9-VPR) construct was created by Gibson assembly of the dCas9-VPR insert from SP-dCas9-VPR (Addgene\#63798) into pCW57-MCS1-P2A-MCS2-Neo (Addgene\#89180). 

%Retroviral pLVX-hTERT was a kind gift of Teresa Davoli. Activating TERT gRNAs were targeted up to 1000 bp upstream of the TERT promoter transcriptional start site, and designed using online software from the Broad Institute (portals.broadinstitute.org/gpp/public/analysis-tools/sgrna-design). gRNA sequences were cloned into a modified version of lentiGuide-Puro (Addgene\#52963) in which the selection cassette had been swapped for Zeocin resistance. Activating TERT gRNA sequences are shown in Supplementary Table 3. TTN gRNA sequences were used as described32.

%\subsection*{Viral gene delivery}
%Retroviral constructs were transfected into Phoenix amphitropic cells using calcium phosphate precipitation. Lentiviral constructs were transfected with appropriate packaging vectors using calcium phosphate precipitation into 293-FT cells. Viral supernatants were collected and filtered before addition to target cells, supplemented with 4 $\mu$g/ml polybrene. For activating gRNA constructs, multiple viral supernatants were collected and concentrated using PEG-it Virus Precipitation Solution (System Biosciences LV810A-1). Cells were infected 2-3 times at 12-hour intervals before selection in the appropriate antibiotic. 

%\subsection*{Immunoblotting}
%For immunoblotting, cell pellets were directly lysed in 1X Laemmli buffer (2\% SDS, 5\% \textbeta-mercaptoethanol, 10\% glycerol, 0.002\% bromophenol blue and 62.5 mM Tris-HCl pH 6.8) at a concentration of 107 cells/ml. Lysates were denatured at 100\textdegree C, and DNA was sheared with a 28Â½ gauge insulin needle. Lysates were resolved on SDS/PAGE gels (Life Technologies), transferred to nitrocellulose membranes and blocked with 5\% milk in TBS with 0.1\% Tween-20. Primary antibodies (anti-Cas9 7A9-3A3, Cell Signaling Technology \#14697S, anti-Î³-tubulin Sigma\#T5326, anti- Human Retinoblastoma protein BD Pharmigen \#554136, anti-p21 F-5 Santa Cruz sc-6246) were incubated overnight, before membrane washing and incubation with appropriate HRP-conjugated secondary antibodies (Amersham) and detection with SuperSignal ECL West Pico PLUS chemiluminescence (ThermoFisher).

%\subsection*{Immunofluorescence}
%Cells were grown on glass coverslips and fixed in 3\% paraformaldehyde and 2\% sucrose. Coverslips were permeabilized in 0.5\% Triton-X-100/PBS, and blocked in goat block (0.1\% BSA, 3\% goat serum, 0.1\% Triton-X-100, 2mM EDTA) in PBS. Primary and secondary antibodies (Rabbit anti-53BP1 Abcam \#ab-175933, F(ab')2-Goat anti-Rabbit IgG (H+L) Cross-Adsorbed Alexa Fluor 488 ThermoFisher A-11070) were diluted in goat block. Slides were counter-stained with DAPI and mounted using Prolong gold anti-fade medium. Images were acquired on a DeltaVision microscope (Applied Precision) equipped with a cooled charge-coupled device camera (DV Elite CMOS Camera), with a PlanApo 60Ã 1.42 NA objective (Olympus America), and SoftWoRx software. Images were analyzed for foci numbers using a custom-made algorithm written for FIJI, courtesy of Leonid Timashev50.

%\subsection*{Metaphase spread preparation and staining}
%Metaphase spreads were prepared by treatment of cells with 0.1 $\mu$g/ml colcemid (Roche) for 3 hours, before trypsinization and swelling at 37\textdegree C for 5-10 mins in 0.075 M KCl. Cells were fixed in a freshly prepared 3:1 mixture of methanol to acetic acid and stored at 4 \textdegree C at least overnight. Spreads were prepared by dropping cell solution onto cold glass slides exposed to steam from a 75\textdegree C water bath, flooding slide with acetic acid, before exposure of the dropped cells for 3-5 secs in steam. Slides were dried overnight before storage in 100\% Ethanol at -20 \textdegree C. For visualization of fusions, slides were rinsed in PBS, fixed in 4\% formaldehyde/PBS for 5 mins, and dehydrated in an ethanol series before co-denaturation of slide and PNA probes (TelG-Cy3 PNA Bio F1006, CENPB-AF488 PNA Bio F3004) for 3 mins at 80 \textdegree C in hybridization solution (10 mM Tris-HCl pH 7.2, 70\% formamide, 0.5\% Roche 11096176001 blocking reagent). Hybridization was carried out for 2 hours at RT in the dark, before washing twice in 10 mM Tris-HCl pH 7.2, 70\% formamide and 0.1\% BSA, then washing three times in 0.1 M Tris-HCl pH 7.2, 0.15M NaCl, 0.08\% Tween-20. DAPI was included in the second wash. Slides were dehydrated through an ethanol series before mounting with Prolong Gold antifade medium (Invitrogen). 

%For chromosome painting, slides were prepared as above for chromosome fusions, and co-denaturation of chromosome specific paints (XCP-12 Metasystems D-0312-050-FI, XCP-21 Metasystems D-0321-050-OR) was carried out at 75\textdegree C for 2 mins, before hybridization overnight at 37 \textdegree C. Post hybridization washes were 0.4X SSC for 2 mins at 72\textdegree C, 2X SSC, 0.05\% Tween-20 for 30 secs, followed by counterstaining in DAPI for 15 min, and a rinse in ddH2O before mounting in Prolong Gold antifade medium (Invitrogen). For karyotyping, slides were prepared as above, and analysis was carried out on DAPI stained chromosomes.

%\subsection*{BAC probes}
%To identify individual chromosomes on metaphase spreads, BAC probes were ordered from BACPAC Genomics (Chr.12p11.2 RP11-90H7, Chr.18q12.3~21.1 RP11-91K12, Chr.6p21.2~21.3 RP11-79J17). Probe DNA was nick-translated with either Digoxigenin-11-UTP or Biotin-16-UTP (Roche) using DNase I (Roche) and DNA polymerase I (NEB) overnight at 15\textdegree C. Probes were precipitated with Cot1 Human DNA (Invitrogen) and salmon sperm DNA (Invitrogen) and resuspended in 50\% formamide, 2X SSC and 10\% dextran sulfate before denaturation for 8 mins at 80\textdegree C. Metaphase spreads were prepared as above, and slides were denatured with 70\% formamide, 2X SSC for 2 mins at 80\textdegree C before dehydration through an ethanol series. Slides were co-denatured for 2 mins at 80\textdegree C with TelG-647 (PNA Bio F1014) in hybridization solution (10mM Tris-HCl pH 7.2, 70\% formamide, 0.5\% Roche 11096176001 blocking reagent) followed by a 2-hour hybridization at RT. Denatured BAC probes were then applied and hybridized overnight at 37 \textdegree C. Slides were washed for 3 x 5 mins in 1X SSC at 60 \textdegree C, followed by a blocking step in 30 \textmu g/ml BSA, 4X SSC and 0.1\% Tween-20 for 30 mins at 37\textdegree C. BAC probes were detected with anti-Digoxenin-Rhodamine (Roche 11207750910) and Avadin-FITC antibodies (VWR CAP21221) in 10 \textmu g/ml BSA, 1X SSC and 0.1\% Tween-20 by incubating for 30 mins at 37\textdegree C, before washing twice for 5 mins in 4X SSC and 0.1\% Tween-20 at 42\textdegree C. Counterstaining with DAPI was carried out for 15 mins at RT, before a further wash at 42 \textdegree C in 4X SSC, 0.1\% Tween-20 and mounting in Prolong-Gold antifade (Invitrogen). Images were acquired on the Deltavision microscope equipment detailed above. Images were analyzed using FIJI51. Briefly, individual chromosomes were detected with the BAC probes. Intensity measurements of the Tel-G signal were quantified for each p-arm and q-arm of identified chromosomes. Measurements were also taken for all other telomeres in the same spread. Background subtracted measurements for all telomeres were compared to the shortest (lowest intensity) of the Chr.12p (or Chr.18p or Chr.6p) telomeres on each spread, and the results expressed as a ratio. For identification of fusions containing chromosome 12, the same protocol was carried out using a centromere probe (CENPB-AF488, PNA Bio F3004).

%\subsection*{qPCR}
%RNA was isolated from cell pellets using a Qiagen RNeasy kit, according to the manufacturerâs instructions. cDNA was synthesized using Superscript IV first-strand synthesis (ThermoFisher). qPCR was carried with SYBER Green reagents (ThermoFisher) and run on a Life Technologies QuantStudio 12K machine. qPCR primer sequences are shown in Supplementary Table 3. Expression was quantified using the standard \textdelta \textdelta CT method relative to \textbeta-actin.

%\subsection*{TRAP assay}
%Telomerase activity was assessed using the TRAPeze kit (EMD Millipore S7700) according to the manufacturerâs instructions. Amplification products were resolved on 12\% PAGE gels and visualized with EtBr staining. 

%\subsection*{STELA, Fusion PCR and telomeric blots}
%High molecular weight DNA was extracted from cell pellets using a MagAttract HMW DNA kit (Qiagen) and solubilized by overnight digestion with EcoRI (for STELA and Fusion PCR) or a combination of AluI and MboI (for telomeric blots). STELA was carried out essentially as described34. Briefly, 10 ng of DNA was annealed to a mixture of  six telorette linkers (Supplementary Table 3) overnight at 35\textdegree C, before dilution with water to a concentration of 200 pg/\textmu l. Multiple PCR reactions for each sample were carried out with 200 pg of annealed DNA using the XpYpE2 and teltail primers (Supplementary Table 3) and FailSafe PCR reagents (Epicentre). PCR conditions were as follows: 94 \textdegree C for 15s, 27 cycles of 95\textdegree C for 15s, 58\textdegree C for 20s, 68\textdegree C for 10 mins, and a final extension at 68\textdegree C for 9 mins. PCR products were resolved on 0.8\% TAE gels, denatured and transferred to Hybond membrane via Southern blotting. Products were detected with a randomly primed \textalpha-$^{32}$P DNA probe created by amplification of the telomere-adjacent region of the XpYp telomere (using XpYpE2 and XpYpB2 primers, Supplementary Table 3). For quantification, FIJI was used to measure relative signal between indicated molecular weight markers relative to background signal for each sample. 

%Fusion PCR was carried out essentially as described11,35. Subtelomeric primers (Supplementary Table 3) used for amplification of telomeric fusions were XpYpM, 17p6 and 21q4. The control primer XpYpc2tr was included for control amplification of XpYp subtelomeric DNA and detected using EtBr. Fusion products were detected with a random primed \textalpha-$^{32}$P labelled DNA probe (21q probe) specific for the TelBam11 telomere subfamily36,52, which was created with the 21q4 primer and 21q-seq-rev2. The number of fusions per haploid genome (6 pg) is calculated based on the amount of input DNA in each PCR reaction. 

%Telomere length was assessed using telomeric restriction fragment analysis. Briefly, AluI/MboI-digested genomic DNA was run on 0.8\% TAE gels, before denaturation, neutralization, and transfer onto a Hybond membrane according to standard Southern blotting procedures. Telomeric DNA was detected using a \textalpha-$^{32}$P labelled Sty11 telomeric repeat probe45.

%\subsection*{WGS Library preparation}
%Genomic DNA was extracted from cell pellets using a QIAGEN QIAamp DNA mini kit and sheared using a Covaris Ultrasonicator (E220) to approximately 300 bp fragments. DNA concentration was measured using Qbit 4.0 reagents (ThermoFisher) and 200 ng of fragmented DNA was used for library preparation. End repair and A-tailing was carried out with NEBNext End repair reaction enzyme mix and buffer (E7442), and KAPA Dual-Indexed Adapters (Roche) were ligated using the T4 DNA ligase kit from NEB (M0202). Post-ligation size selection was performed with AMPure XP beads (Beckman Coulter) before washing two times in 80\% ethanol. Libraries were amplified using KAPA HiFi HotStart ready mix (Roche) and P5 and P7 primers (IDT). PCR program was as follows: 98\textdegree C for 45 s, 5 cycles of 98\textdegree C for 15 s, 60\textdegree C for 30 s, 72\textdegree C for 30 s and a final extension at 72\textdegree C for 5 mins. A further size selection and washing step was carried out after library amplification, and library quality was confirmed on Bioanalyzer chips (Agilent) and using a KAPA Library Quantification kit (Roche). Libraries were pooled and submitted for sequencing on NovaSeq 6000 at the New York Genome Center. 